{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "099c50f6-93aa-4c54-988d-90cfadde27b7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "library(ArchR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "66d2a6a8-51b8-4787-b257-3468d8379508",
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "caf5245a-27ea-48a9-838c-9b73fbd1c995",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting default number of Parallel threads to 1.\n",
      "\n",
      "Setting default genome to Hg38.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "set.seed(2022)\n",
    "addArchRThreads(threads = 1) \n",
    "addArchRGenome(\"hg38\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "d87f9fb3-ffb4-442b-a0d0-5a2ca4367402",
   "metadata": {},
   "outputs": [],
   "source": [
    "inputFiles = \"/project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/01_prepare_fragments_file/final_fragment_files/GSM4138889_scATAC_BMMC_D6T1_liftOverhg38_no_chrMt_sorted_fragments.tsv.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "84df57d8-0603-45d5-b47d-8fe24cbc4e13",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-createArrows-b94d6db1d353-Date-2024-05-12_Time-22-00-19.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "Cleaning Temporary Files\n",
      "\n",
      "2024-05-12 22:00:22 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "(BMMC_D6T1 : 1 of 1) Determining Arrow Method to use!\n",
      "\n",
      "2024-05-12 22:00:22 : (BMMC_D6T1 : 1 of 1) Reading In Fragments from inputFiles (readMethod = tabix), 0.004 mins elapsed.\n",
      "\n",
      "2024-05-12 22:00:22 : (BMMC_D6T1 : 1 of 1) Tabix Bed To Temporary File, 0.005 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:04:07 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 8 Percent, 3.748 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:06:11 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 17 Percent, 5.827 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:08:00 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 25 Percent, 7.633 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:09:36 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 33 Percent, 9.233 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:11:08 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 42 Percent, 10.776 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:12:49 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 50 Percent, 12.46 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:13:55 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 58 Percent, 13.557 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:15:08 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 67 Percent, 14.771 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:16:23 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 75 Percent, 16.024 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:17:37 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 83 Percent, 17.253 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:18:14 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 92 Percent, 17.869 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-12 22:18:58 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 100 Percent, 18.605 mins elapsed.\n",
      "\n",
      "2024-05-12 22:18:59 : (BMMC_D6T1 : 1 of 1) Successful creation of Temporary File, 18.627 mins elapsed.\n",
      "\n",
      "2024-05-12 22:18:59 : (BMMC_D6T1 : 1 of 1) Creating ArrowFile From Temporary File, 18.628 mins elapsed.\n",
      "\n",
      "2024-05-12 22:24:12 : (BMMC_D6T1 : 1 of 1) Successful creation of Arrow File, 23.846 mins elapsed.\n",
      "\n",
      "2024-05-12 22:28:25 : (BMMC_D6T1 : 1 of 1) CellStats : Number of Cells Pass Filter = 13745 , 28.052 mins elapsed.\n",
      "\n",
      "2024-05-12 22:28:25 : (BMMC_D6T1 : 1 of 1) CellStats : Median Frags = 9853 , 28.052 mins elapsed.\n",
      "\n",
      "2024-05-12 22:28:25 : (BMMC_D6T1 : 1 of 1) CellStats : Median TSS Enrichment = 14.317 , 28.053 mins elapsed.\n",
      "\n",
      "2024-05-12 22:28:30 : (BMMC_D6T1 : 1 of 1) Adding Additional Feature Counts!, 28.143 mins elapsed.\n",
      "\n",
      "2024-05-12 22:31:40 : (BMMC_D6T1 : 1 of 1) Removing Fragments from Filtered Cells, 31.307 mins elapsed.\n",
      "\n",
      "2024-05-12 22:31:40 : (BMMC_D6T1 : 1 of 1) Creating Filtered Arrow File, 31.308 mins elapsed.\n",
      "\n",
      "2024-05-12 22:33:30 : (BMMC_D6T1 : 1 of 1) Finished Constructing Filtered Arrow File!, 33.144 mins elapsed.\n",
      "\n",
      "2024-05-12 22:33:31 : (BMMC_D6T1 : 1 of 1) Adding TileMatrix!, 33.159 mins elapsed.\n",
      "\n",
      "2024-05-12 22:37:30 : (BMMC_D6T1 : 1 of 1) Adding GeneScoreMatrix!, 37.146 mins elapsed.\n",
      "\n",
      "2024-05-12 22:42:14 : (BMMC_D6T1 : 1 of 1) Finished Creating Arrow File, 41.87 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-createArrows-b94d6db1d353-Date-2024-05-12_Time-22-00-19.log\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ArrowFiles <- createArrowFiles(\n",
    "  inputFiles = inputFiles,\n",
    "  sampleNames = \"BMMC_D6T1\",\n",
    "  minTSS = 4, #Dont set this too high because you can always increase later\n",
    "  minFrags = 1000, \n",
    "  addTileMat = TRUE,\n",
    "  addGeneScoreMat = TRUE\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d04cef22-5dc5-4a6d-9f5a-2c8582ef8fbb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Validating Arrows...\n",
      "\n",
      "Getting SampleNames...\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Copying ArrowFiles to Ouptut Directory! If you want to save disk space set copyArrows = FALSE\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Getting Cell Metadata...\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Merging Cell Metadata...\n",
      "\n",
      "Initializing ArchRProject...\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "proj1 <- ArchRProject(\n",
    "  ArrowFiles = ArrowFiles, \n",
    "  outputDirectory = \"BMMC_D6T1\",\n",
    "  copyArrows = TRUE #This is recommened so that you maintain an unaltered copy for later usage.\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "07439ed6-b2c7-4e40-b180-687d1b5dee3e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneScoreMatrix'</li><li>'TileMatrix'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneScoreMatrix'\n",
       "\\item 'TileMatrix'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneScoreMatrix'\n",
       "2. 'TileMatrix'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GeneScoreMatrix\" \"TileMatrix\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getAvailableMatrices(proj1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "1152ad17-f507-430d-809d-ee101f15d704",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-b94d14e36d76-Date-2024-05-12_Time-23-12-27.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-12 23:13:07 : Organizing colData, 0.665 mins elapsed.\n",
      "\n",
      "2024-05-12 23:13:07 : Organizing rowData, 0.666 mins elapsed.\n",
      "\n",
      "2024-05-12 23:13:07 : Organizing rowRanges, 0.666 mins elapsed.\n",
      "\n",
      "2024-05-12 23:13:07 : Organizing Assays (1 of 1), 0.667 mins elapsed.\n",
      "\n",
      "2024-05-12 23:13:07 : Constructing SummarizedExperiment, 0.668 mins elapsed.\n",
      "\n",
      "2024-05-12 23:13:08 : Finished Matrix Creation, 0.686 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>24919</li><li>13745</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 24919\n",
       "\\item 13745\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 24919\n",
       "2. 13745\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 24919 13745"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(getMatrixFromProject(proj1, useMatrix='GeneScoreMatrix'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "49e50b9c-94cf-4980-bf94-82c23686e57e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Checking Inputs...\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addIterativeLSI-b94d6b0677e8-Date-2024-05-12_Time-23-13-09.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-12 23:13:11 : Computing Total Across All Features, 0.001 mins elapsed.\n",
      "\n",
      "2024-05-12 23:13:12 : Computing Top Features, 0.017 mins elapsed.\n",
      "\n",
      "###########\n",
      "2024-05-12 23:13:14 : Running LSI (1 of 2) on Top Features, 0.048 mins elapsed.\n",
      "###########\n",
      "\n",
      "2024-05-12 23:13:14 : Sampling Cells (N = 10000) for Estimated LSI, 0.049 mins elapsed.\n",
      "\n",
      "2024-05-12 23:13:14 : Creating Sampled Partial Matrix, 0.049 mins elapsed.\n",
      "\n",
      "2024-05-12 23:13:36 : Computing Estimated LSI (projectAll = FALSE), 0.411 mins elapsed.\n",
      "\n",
      "2024-05-12 23:14:53 : Identifying Clusters, 1.694 mins elapsed.\n",
      "\n",
      "2024-05-12 23:15:25 : Identified 9 Clusters, 2.233 mins elapsed.\n",
      "\n",
      "2024-05-12 23:15:25 : Saving LSI Iteration, 2.235 mins elapsed.\n",
      "\n",
      "2024-05-12 23:15:58 : Creating Cluster Matrix on the total Group Features, 2.777 mins elapsed.\n",
      "\n",
      "2024-05-12 23:17:03 : Computing Variable Features, 3.857 mins elapsed.\n",
      "\n",
      "###########\n",
      "2024-05-12 23:17:03 : Running LSI (2 of 2) on Variable Features, 3.861 mins elapsed.\n",
      "###########\n",
      "\n",
      "2024-05-12 23:17:03 : Creating Partial Matrix, 3.862 mins elapsed.\n",
      "\n",
      "2024-05-12 23:17:48 : Computing LSI, 4.606 mins elapsed.\n",
      "\n",
      "2024-05-12 23:18:29 : Finished Running IterativeLSI, 5.293 mins elapsed.\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addClusters-b94d11afa1ec-Date-2024-05-12_Time-23-18-29.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-12 23:18:33 : Running Seurats FindClusters (Stuart et al. Cell 2019), 0.003 mins elapsed.\n",
      "\n",
      "Computing nearest neighbor graph\n",
      "\n",
      "Computing SNN\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck\n",
      "\n",
      "Number of nodes: 13745\n",
      "Number of edges: 549245\n",
      "\n",
      "Running Louvain algorithm...\n",
      "Maximum modularity in 10 random starts: 0.8934\n",
      "Number of communities: 27\n",
      "Elapsed time: 4 seconds\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-12 23:18:54 : Testing Biased Clusters, 0.365 mins elapsed.\n",
      "\n",
      "2024-05-12 23:18:55 : Testing Outlier Clusters, 0.372 mins elapsed.\n",
      "\n",
      "2024-05-12 23:18:55 : Identified more clusters than maxClusters allowed (n = 0). Merging clusters to maxClusters (n = 25).\n",
      "If this is not desired set maxClusters = NULL!, 0.373 mins elapsed.\n",
      "\n",
      "2024-05-12 23:18:55 : Assigning Cluster Names to 25 Clusters, 0.373 mins elapsed.\n",
      "\n",
      "2024-05-12 23:18:55 : Finished addClusters, 0.375 mins elapsed.\n",
      "\n",
      "23:18:55 UMAP embedding parameters a = 0.583 b = 1.334\n",
      "\n",
      "23:18:55 Read 13745 rows and found 30 numeric columns\n",
      "\n",
      "23:18:55 Using Annoy for neighbor search, n_neighbors = 30\n",
      "\n",
      "23:18:55 Building Annoy index with metric = cosine, n_trees = 50\n",
      "\n",
      "0%   10   20   30   40   50   60   70   80   90   100%\n",
      "\n",
      "[----|----|----|----|----|----|----|----|----|----|\n",
      "\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "|\n",
      "\n",
      "23:18:58 Writing NN index file to temp file /scratch/local/Rtmpl78HX8/fileb94d67745526\n",
      "\n",
      "23:18:58 Searching Annoy index using 64 threads, search_k = 3000\n",
      "\n",
      "23:18:58 Annoy recall = 100%\n",
      "\n",
      "23:18:59 Commencing smooth kNN distance calibration using 64 threads\n",
      " with target n_neighbors = 30\n",
      "\n",
      "23:19:01 Initializing from normalized Laplacian + noise (using irlba)\n",
      "\n",
      "23:19:02 Commencing optimization for 200 epochs, with 607738 positive edges\n",
      "\n",
      "23:19:09 Optimization finished\n",
      "\n",
      "23:19:09 Creating temp model dir /scratch/local/Rtmpl78HX8/dirb94d168cef7f\n",
      "\n",
      "23:19:09 Creating dir /scratch/local/Rtmpl78HX8/dirb94d168cef7f\n",
      "\n",
      "23:19:10 Changing to /scratch/local/Rtmpl78HX8/dirb94d168cef7f\n",
      "\n",
      "23:19:10 Creating /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/Embeddings/Save-Uwot-UMAP-Params-IterativeLSI-b94d3a906f60-Date-2024-05-12_Time-23-19-09.tar\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addGroupCoverages-b94d787433ed-Date-2024-05-12_Time-23-19-11.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "C1 (1 of 25) : CellGroups N = 2\n",
      "\n",
      "C2 (2 of 25) : CellGroups N = 2\n",
      "\n",
      "C3 (3 of 25) : CellGroups N = 2\n",
      "\n",
      "C4 (4 of 25) : CellGroups N = 2\n",
      "\n",
      "C5 (5 of 25) : CellGroups N = 2\n",
      "\n",
      "C6 (6 of 25) : CellGroups N = 2\n",
      "\n",
      "C7 (7 of 25) : CellGroups N = 2\n",
      "\n",
      "C8 (8 of 25) : CellGroups N = 2\n",
      "\n",
      "C9 (9 of 25) : CellGroups N = 2\n",
      "\n",
      "C10 (10 of 25) : CellGroups N = 2\n",
      "\n",
      "C11 (11 of 25) : CellGroups N = 2\n",
      "\n",
      "C12 (12 of 25) : CellGroups N = 2\n",
      "\n",
      "C13 (13 of 25) : CellGroups N = 2\n",
      "\n",
      "C14 (14 of 25) : CellGroups N = 2\n",
      "\n",
      "C15 (15 of 25) : CellGroups N = 2\n",
      "\n",
      "C16 (16 of 25) : CellGroups N = 2\n",
      "\n",
      "C17 (17 of 25) : CellGroups N = 2\n",
      "\n",
      "C18 (18 of 25) : CellGroups N = 2\n",
      "\n",
      "C19 (19 of 25) : CellGroups N = 2\n",
      "\n",
      "C20 (20 of 25) : CellGroups N = 2\n",
      "\n",
      "C21 (21 of 25) : CellGroups N = 2\n",
      "\n",
      "C22 (22 of 25) : CellGroups N = 2\n",
      "\n",
      "C23 (23 of 25) : CellGroups N = 2\n",
      "\n",
      "C24 (24 of 25) : CellGroups N = 2\n",
      "\n",
      "C25 (25 of 25) : CellGroups N = 2\n",
      "\n",
      "2024-05-12 23:19:16 : Creating Coverage Files!, 0.089 mins elapsed.\n",
      "\n",
      "2024-05-12 23:19:16 : Batch Execution w/ safelapply!, 0.089 mins elapsed.\n",
      "\n",
      "2024-05-12 23:19:16 : Group C1._.Rep1 (1 of 50) : Creating Group Coverage File : C1._.Rep1.insertions.coverage.h5, 0.091 mins elapsed.\n",
      "\n",
      "Number of Cells = 52\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:20:10 : Group C1._.Rep2 (2 of 50) : Creating Group Coverage File : C1._.Rep2.insertions.coverage.h5, 0.987 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:21:02 : Group C2._.Rep1 (3 of 50) : Creating Group Coverage File : C2._.Rep1.insertions.coverage.h5, 1.845 mins elapsed.\n",
      "\n",
      "Number of Cells = 59\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:21:54 : Group C2._.Rep2 (4 of 50) : Creating Group Coverage File : C2._.Rep2.insertions.coverage.h5, 2.714 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:22:46 : Group C3._.Rep1 (5 of 50) : Creating Group Coverage File : C3._.Rep1.insertions.coverage.h5, 3.581 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:23:49 : Group C3._.Rep2 (6 of 50) : Creating Group Coverage File : C3._.Rep2.insertions.coverage.h5, 4.637 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:24:41 : Group C4._.Rep1 (7 of 50) : Creating Group Coverage File : C4._.Rep1.insertions.coverage.h5, 5.501 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:25:40 : Group C4._.Rep2 (8 of 50) : Creating Group Coverage File : C4._.Rep2.insertions.coverage.h5, 6.48 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:26:31 : Group C5._.Rep1 (9 of 50) : Creating Group Coverage File : C5._.Rep1.insertions.coverage.h5, 7.335 mins elapsed.\n",
      "\n",
      "Number of Cells = 97\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:27:24 : Group C5._.Rep2 (10 of 50) : Creating Group Coverage File : C5._.Rep2.insertions.coverage.h5, 8.213 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:28:15 : Group C6._.Rep1 (11 of 50) : Creating Group Coverage File : C6._.Rep1.insertions.coverage.h5, 9.062 mins elapsed.\n",
      "\n",
      "Number of Cells = 267\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:29:11 : Group C6._.Rep2 (12 of 50) : Creating Group Coverage File : C6._.Rep2.insertions.coverage.h5, 10.01 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:30:03 : Group C7._.Rep1 (13 of 50) : Creating Group Coverage File : C7._.Rep1.insertions.coverage.h5, 10.867 mins elapsed.\n",
      "\n",
      "Number of Cells = 405\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:31:05 : Group C7._.Rep2 (14 of 50) : Creating Group Coverage File : C7._.Rep2.insertions.coverage.h5, 11.902 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:31:56 : Group C8._.Rep1 (15 of 50) : Creating Group Coverage File : C8._.Rep1.insertions.coverage.h5, 12.757 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:32:48 : Group C8._.Rep2 (16 of 50) : Creating Group Coverage File : C8._.Rep2.insertions.coverage.h5, 13.614 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:33:39 : Group C9._.Rep1 (17 of 50) : Creating Group Coverage File : C9._.Rep1.insertions.coverage.h5, 14.466 mins elapsed.\n",
      "\n",
      "Number of Cells = 343\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:34:35 : Group C9._.Rep2 (18 of 50) : Creating Group Coverage File : C9._.Rep2.insertions.coverage.h5, 15.402 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:35:27 : Group C10._.Rep1 (19 of 50) : Creating Group Coverage File : C10._.Rep1.insertions.coverage.h5, 16.263 mins elapsed.\n",
      "\n",
      "Number of Cells = 495\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:36:24 : Group C10._.Rep2 (20 of 50) : Creating Group Coverage File : C10._.Rep2.insertions.coverage.h5, 17.216 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:37:15 : Group C11._.Rep1 (21 of 50) : Creating Group Coverage File : C11._.Rep1.insertions.coverage.h5, 18.075 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:38:06 : Group C11._.Rep2 (22 of 50) : Creating Group Coverage File : C11._.Rep2.insertions.coverage.h5, 18.913 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:38:57 : Group C12._.Rep1 (23 of 50) : Creating Group Coverage File : C12._.Rep1.insertions.coverage.h5, 19.777 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:39:54 : Group C12._.Rep2 (24 of 50) : Creating Group Coverage File : C12._.Rep2.insertions.coverage.h5, 20.718 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:40:45 : Group C13._.Rep1 (25 of 50) : Creating Group Coverage File : C13._.Rep1.insertions.coverage.h5, 21.562 mins elapsed.\n",
      "\n",
      "Number of Cells = 124\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:41:36 : Group C13._.Rep2 (26 of 50) : Creating Group Coverage File : C13._.Rep2.insertions.coverage.h5, 22.414 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:42:26 : Group C14._.Rep1 (27 of 50) : Creating Group Coverage File : C14._.Rep1.insertions.coverage.h5, 23.251 mins elapsed.\n",
      "\n",
      "Number of Cells = 357\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:43:25 : Group C14._.Rep2 (28 of 50) : Creating Group Coverage File : C14._.Rep2.insertions.coverage.h5, 24.244 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:44:15 : Group C15._.Rep1 (29 of 50) : Creating Group Coverage File : C15._.Rep1.insertions.coverage.h5, 25.074 mins elapsed.\n",
      "\n",
      "Number of Cells = 190\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:45:07 : Group C15._.Rep2 (30 of 50) : Creating Group Coverage File : C15._.Rep2.insertions.coverage.h5, 25.933 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:45:57 : Group C16._.Rep1 (31 of 50) : Creating Group Coverage File : C16._.Rep1.insertions.coverage.h5, 26.767 mins elapsed.\n",
      "\n",
      "Number of Cells = 446\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:46:53 : Group C16._.Rep2 (32 of 50) : Creating Group Coverage File : C16._.Rep2.insertions.coverage.h5, 27.696 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:47:42 : Group C17._.Rep1 (33 of 50) : Creating Group Coverage File : C17._.Rep1.insertions.coverage.h5, 28.511 mins elapsed.\n",
      "\n",
      "Number of Cells = 420\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:48:37 : Group C17._.Rep2 (34 of 50) : Creating Group Coverage File : C17._.Rep2.insertions.coverage.h5, 29.433 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:49:28 : Group C18._.Rep1 (35 of 50) : Creating Group Coverage File : C18._.Rep1.insertions.coverage.h5, 30.282 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:50:24 : Group C18._.Rep2 (36 of 50) : Creating Group Coverage File : C18._.Rep2.insertions.coverage.h5, 31.217 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:51:14 : Group C19._.Rep1 (37 of 50) : Creating Group Coverage File : C19._.Rep1.insertions.coverage.h5, 32.06 mins elapsed.\n",
      "\n",
      "Number of Cells = 22\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:52:04 : Group C19._.Rep2 (38 of 50) : Creating Group Coverage File : C19._.Rep2.insertions.coverage.h5, 32.884 mins elapsed.\n",
      "\n",
      "Number of Cells = 17\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:52:55 : Group C20._.Rep1 (39 of 50) : Creating Group Coverage File : C20._.Rep1.insertions.coverage.h5, 33.738 mins elapsed.\n",
      "\n",
      "Number of Cells = 224\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:53:50 : Group C20._.Rep2 (40 of 50) : Creating Group Coverage File : C20._.Rep2.insertions.coverage.h5, 34.653 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:54:42 : Group C21._.Rep1 (41 of 50) : Creating Group Coverage File : C21._.Rep1.insertions.coverage.h5, 35.511 mins elapsed.\n",
      "\n",
      "Number of Cells = 49\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:55:33 : Group C21._.Rep2 (42 of 50) : Creating Group Coverage File : C21._.Rep2.insertions.coverage.h5, 36.366 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:56:24 : Group C22._.Rep1 (43 of 50) : Creating Group Coverage File : C22._.Rep1.insertions.coverage.h5, 37.219 mins elapsed.\n",
      "\n",
      "Number of Cells = 312\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:57:20 : Group C22._.Rep2 (44 of 50) : Creating Group Coverage File : C22._.Rep2.insertions.coverage.h5, 38.158 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:58:10 : Group C23._.Rep1 (45 of 50) : Creating Group Coverage File : C23._.Rep1.insertions.coverage.h5, 38.989 mins elapsed.\n",
      "\n",
      "Number of Cells = 247\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:59:03 : Group C23._.Rep2 (46 of 50) : Creating Group Coverage File : C23._.Rep2.insertions.coverage.h5, 39.87 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 23:59:54 : Group C24._.Rep1 (47 of 50) : Creating Group Coverage File : C24._.Rep1.insertions.coverage.h5, 40.715 mins elapsed.\n",
      "\n",
      "Number of Cells = 395\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-13 00:00:50 : Group C24._.Rep2 (48 of 50) : Creating Group Coverage File : C24._.Rep2.insertions.coverage.h5, 41.658 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-13 00:01:41 : Group C25._.Rep1 (49 of 50) : Creating Group Coverage File : C25._.Rep1.insertions.coverage.h5, 42.51 mins elapsed.\n",
      "\n",
      "Number of Cells = 427\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-13 00:02:36 : Group C25._.Rep2 (50 of 50) : Creating Group Coverage File : C25._.Rep2.insertions.coverage.h5, 43.419 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-13 00:03:27 : Adding Kmer Bias to Coverage Files!, 44.27 mins elapsed.\n",
      "\n",
      "Kmer Bias chr1 (1 of 24)\n",
      "\n",
      "chr1 \n",
      "Coverage File chr1 (1 of 50)\n",
      "\n",
      "Coverage File chr1 (2 of 50)\n",
      "\n",
      "Coverage File chr1 (3 of 50)\n",
      "\n",
      "Coverage File chr1 (4 of 50)\n",
      "\n",
      "Coverage File chr1 (5 of 50)\n",
      "\n",
      "Coverage File chr1 (6 of 50)\n",
      "\n",
      "Coverage File chr1 (7 of 50)\n",
      "\n",
      "Coverage File chr1 (8 of 50)\n",
      "\n",
      "Coverage File chr1 (9 of 50)\n",
      "\n",
      "Coverage File chr1 (10 of 50)\n",
      "\n",
      "Coverage File chr1 (11 of 50)\n",
      "\n",
      "Coverage File chr1 (12 of 50)\n",
      "\n",
      "Coverage File chr1 (13 of 50)\n",
      "\n",
      "Coverage File chr1 (14 of 50)\n",
      "\n",
      "Coverage File chr1 (15 of 50)\n",
      "\n",
      "Coverage File chr1 (16 of 50)\n",
      "\n",
      "Coverage File chr1 (17 of 50)\n",
      "\n",
      "Coverage File chr1 (18 of 50)\n",
      "\n",
      "Coverage File chr1 (19 of 50)\n",
      "\n",
      "Coverage File chr1 (20 of 50)\n",
      "\n",
      "Coverage File chr1 (21 of 50)\n",
      "\n",
      "Coverage File chr1 (22 of 50)\n",
      "\n",
      "Coverage File chr1 (23 of 50)\n",
      "\n",
      "Coverage File chr1 (24 of 50)\n",
      "\n",
      "Coverage File chr1 (25 of 50)\n",
      "\n",
      "Coverage File chr1 (26 of 50)\n",
      "\n",
      "Coverage File chr1 (27 of 50)\n",
      "\n",
      "Coverage File chr1 (28 of 50)\n",
      "\n",
      "Coverage File chr1 (29 of 50)\n",
      "\n",
      "Coverage File chr1 (30 of 50)\n",
      "\n",
      "Coverage File chr1 (31 of 50)\n",
      "\n",
      "Coverage File chr1 (32 of 50)\n",
      "\n",
      "Coverage File chr1 (33 of 50)\n",
      "\n",
      "Coverage File chr1 (34 of 50)\n",
      "\n",
      "Coverage File chr1 (35 of 50)\n",
      "\n",
      "Coverage File chr1 (36 of 50)\n",
      "\n",
      "Coverage File chr1 (37 of 50)\n",
      "\n",
      "Coverage File chr1 (38 of 50)\n",
      "\n",
      "Coverage File chr1 (39 of 50)\n",
      "\n",
      "Coverage File chr1 (40 of 50)\n",
      "\n",
      "Coverage File chr1 (41 of 50)\n",
      "\n",
      "Coverage File chr1 (42 of 50)\n",
      "\n",
      "Coverage File chr1 (43 of 50)\n",
      "\n",
      "Coverage File chr1 (44 of 50)\n",
      "\n",
      "Coverage File chr1 (45 of 50)\n",
      "\n",
      "Coverage File chr1 (46 of 50)\n",
      "\n",
      "Coverage File chr1 (47 of 50)\n",
      "\n",
      "Coverage File chr1 (48 of 50)\n",
      "\n",
      "Coverage File chr1 (49 of 50)\n",
      "\n",
      "Coverage File chr1 (50 of 50)\n",
      "\n",
      "Kmer Bias chr10 (2 of 24)\n",
      "\n",
      "chr10 \n",
      "Coverage File chr10 (1 of 50)\n",
      "\n",
      "Coverage File chr10 (2 of 50)\n",
      "\n",
      "Coverage File chr10 (3 of 50)\n",
      "\n",
      "Coverage File chr10 (4 of 50)\n",
      "\n",
      "Coverage File chr10 (5 of 50)\n",
      "\n",
      "Coverage File chr10 (6 of 50)\n",
      "\n",
      "Coverage File chr10 (7 of 50)\n",
      "\n",
      "Coverage File chr10 (8 of 50)\n",
      "\n",
      "Coverage File chr10 (9 of 50)\n",
      "\n",
      "Coverage File chr10 (10 of 50)\n",
      "\n",
      "Coverage File chr10 (11 of 50)\n",
      "\n",
      "Coverage File chr10 (12 of 50)\n",
      "\n",
      "Coverage File chr10 (13 of 50)\n",
      "\n",
      "Coverage File chr10 (14 of 50)\n",
      "\n",
      "Coverage File chr10 (15 of 50)\n",
      "\n",
      "Coverage File chr10 (16 of 50)\n",
      "\n",
      "Coverage File chr10 (17 of 50)\n",
      "\n",
      "Coverage File chr10 (18 of 50)\n",
      "\n",
      "Coverage File chr10 (19 of 50)\n",
      "\n",
      "Coverage File chr10 (20 of 50)\n",
      "\n",
      "Coverage File chr10 (21 of 50)\n",
      "\n",
      "Coverage File chr10 (22 of 50)\n",
      "\n",
      "Coverage File chr10 (23 of 50)\n",
      "\n",
      "Coverage File chr10 (24 of 50)\n",
      "\n",
      "Coverage File chr10 (25 of 50)\n",
      "\n",
      "Coverage File chr10 (26 of 50)\n",
      "\n",
      "Coverage File chr10 (27 of 50)\n",
      "\n",
      "Coverage File chr10 (28 of 50)\n",
      "\n",
      "Coverage File chr10 (29 of 50)\n",
      "\n",
      "Coverage File chr10 (30 of 50)\n",
      "\n",
      "Coverage File chr10 (31 of 50)\n",
      "\n",
      "Coverage File chr10 (32 of 50)\n",
      "\n",
      "Coverage File chr10 (33 of 50)\n",
      "\n",
      "Coverage File chr10 (34 of 50)\n",
      "\n",
      "Coverage File chr10 (35 of 50)\n",
      "\n",
      "Coverage File chr10 (36 of 50)\n",
      "\n",
      "Coverage File chr10 (37 of 50)\n",
      "\n",
      "Coverage File chr10 (38 of 50)\n",
      "\n",
      "Coverage File chr10 (39 of 50)\n",
      "\n",
      "Coverage File chr10 (40 of 50)\n",
      "\n",
      "Coverage File chr10 (41 of 50)\n",
      "\n",
      "Coverage File chr10 (42 of 50)\n",
      "\n",
      "Coverage File chr10 (43 of 50)\n",
      "\n",
      "Coverage File chr10 (44 of 50)\n",
      "\n",
      "Coverage File chr10 (45 of 50)\n",
      "\n",
      "Coverage File chr10 (46 of 50)\n",
      "\n",
      "Coverage File chr10 (47 of 50)\n",
      "\n",
      "Coverage File chr10 (48 of 50)\n",
      "\n",
      "Coverage File chr10 (49 of 50)\n",
      "\n",
      "Coverage File chr10 (50 of 50)\n",
      "\n",
      "Kmer Bias chr11 (3 of 24)\n",
      "\n",
      "chr11 \n",
      "Coverage File chr11 (1 of 50)\n",
      "\n",
      "Coverage File chr11 (2 of 50)\n",
      "\n",
      "Coverage File chr11 (3 of 50)\n",
      "\n",
      "Coverage File chr11 (4 of 50)\n",
      "\n",
      "Coverage File chr11 (5 of 50)\n",
      "\n",
      "Coverage File chr11 (6 of 50)\n",
      "\n",
      "Coverage File chr11 (7 of 50)\n",
      "\n",
      "Coverage File chr11 (8 of 50)\n",
      "\n",
      "Coverage File chr11 (9 of 50)\n",
      "\n",
      "Coverage File chr11 (10 of 50)\n",
      "\n",
      "Coverage File chr11 (11 of 50)\n",
      "\n",
      "Coverage File chr11 (12 of 50)\n",
      "\n",
      "Coverage File chr11 (13 of 50)\n",
      "\n",
      "Coverage File chr11 (14 of 50)\n",
      "\n",
      "Coverage File chr11 (15 of 50)\n",
      "\n",
      "Coverage File chr11 (16 of 50)\n",
      "\n",
      "Coverage File chr11 (17 of 50)\n",
      "\n",
      "Coverage File chr11 (18 of 50)\n",
      "\n",
      "Coverage File chr11 (19 of 50)\n",
      "\n",
      "Coverage File chr11 (20 of 50)\n",
      "\n",
      "Coverage File chr11 (21 of 50)\n",
      "\n",
      "Coverage File chr11 (22 of 50)\n",
      "\n",
      "Coverage File chr11 (23 of 50)\n",
      "\n",
      "Coverage File chr11 (24 of 50)\n",
      "\n",
      "Coverage File chr11 (25 of 50)\n",
      "\n",
      "Coverage File chr11 (26 of 50)\n",
      "\n",
      "Coverage File chr11 (27 of 50)\n",
      "\n",
      "Coverage File chr11 (28 of 50)\n",
      "\n",
      "Coverage File chr11 (29 of 50)\n",
      "\n",
      "Coverage File chr11 (30 of 50)\n",
      "\n",
      "Coverage File chr11 (31 of 50)\n",
      "\n",
      "Coverage File chr11 (32 of 50)\n",
      "\n",
      "Coverage File chr11 (33 of 50)\n",
      "\n",
      "Coverage File chr11 (34 of 50)\n",
      "\n",
      "Coverage File chr11 (35 of 50)\n",
      "\n",
      "Coverage File chr11 (36 of 50)\n",
      "\n",
      "Coverage File chr11 (37 of 50)\n",
      "\n",
      "Coverage File chr11 (38 of 50)\n",
      "\n",
      "Coverage File chr11 (39 of 50)\n",
      "\n",
      "Coverage File chr11 (40 of 50)\n",
      "\n",
      "Coverage File chr11 (41 of 50)\n",
      "\n",
      "Coverage File chr11 (42 of 50)\n",
      "\n",
      "Coverage File chr11 (43 of 50)\n",
      "\n",
      "Coverage File chr11 (44 of 50)\n",
      "\n",
      "Coverage File chr11 (45 of 50)\n",
      "\n",
      "Coverage File chr11 (46 of 50)\n",
      "\n",
      "Coverage File chr11 (47 of 50)\n",
      "\n",
      "Coverage File chr11 (48 of 50)\n",
      "\n",
      "Coverage File chr11 (49 of 50)\n",
      "\n",
      "Coverage File chr11 (50 of 50)\n",
      "\n",
      "Kmer Bias chr12 (4 of 24)\n",
      "\n",
      "chr12 \n",
      "Coverage File chr12 (1 of 50)\n",
      "\n",
      "Coverage File chr12 (2 of 50)\n",
      "\n",
      "Coverage File chr12 (3 of 50)\n",
      "\n",
      "Coverage File chr12 (4 of 50)\n",
      "\n",
      "Coverage File chr12 (5 of 50)\n",
      "\n",
      "Coverage File chr12 (6 of 50)\n",
      "\n",
      "Coverage File chr12 (7 of 50)\n",
      "\n",
      "Coverage File chr12 (8 of 50)\n",
      "\n",
      "Coverage File chr12 (9 of 50)\n",
      "\n",
      "Coverage File chr12 (10 of 50)\n",
      "\n",
      "Coverage File chr12 (11 of 50)\n",
      "\n",
      "Coverage File chr12 (12 of 50)\n",
      "\n",
      "Coverage File chr12 (13 of 50)\n",
      "\n",
      "Coverage File chr12 (14 of 50)\n",
      "\n",
      "Coverage File chr12 (15 of 50)\n",
      "\n",
      "Coverage File chr12 (16 of 50)\n",
      "\n",
      "Coverage File chr12 (17 of 50)\n",
      "\n",
      "Coverage File chr12 (18 of 50)\n",
      "\n",
      "Coverage File chr12 (19 of 50)\n",
      "\n",
      "Coverage File chr12 (20 of 50)\n",
      "\n",
      "Coverage File chr12 (21 of 50)\n",
      "\n",
      "Coverage File chr12 (22 of 50)\n",
      "\n",
      "Coverage File chr12 (23 of 50)\n",
      "\n",
      "Coverage File chr12 (24 of 50)\n",
      "\n",
      "Coverage File chr12 (25 of 50)\n",
      "\n",
      "Coverage File chr12 (26 of 50)\n",
      "\n",
      "Coverage File chr12 (27 of 50)\n",
      "\n",
      "Coverage File chr12 (28 of 50)\n",
      "\n",
      "Coverage File chr12 (29 of 50)\n",
      "\n",
      "Coverage File chr12 (30 of 50)\n",
      "\n",
      "Coverage File chr12 (31 of 50)\n",
      "\n",
      "Coverage File chr12 (32 of 50)\n",
      "\n",
      "Coverage File chr12 (33 of 50)\n",
      "\n",
      "Coverage File chr12 (34 of 50)\n",
      "\n",
      "Coverage File chr12 (35 of 50)\n",
      "\n",
      "Coverage File chr12 (36 of 50)\n",
      "\n",
      "Coverage File chr12 (37 of 50)\n",
      "\n",
      "Coverage File chr12 (38 of 50)\n",
      "\n",
      "Coverage File chr12 (39 of 50)\n",
      "\n",
      "Coverage File chr12 (40 of 50)\n",
      "\n",
      "Coverage File chr12 (41 of 50)\n",
      "\n",
      "Coverage File chr12 (42 of 50)\n",
      "\n",
      "Coverage File chr12 (43 of 50)\n",
      "\n",
      "Coverage File chr12 (44 of 50)\n",
      "\n",
      "Coverage File chr12 (45 of 50)\n",
      "\n",
      "Coverage File chr12 (46 of 50)\n",
      "\n",
      "Coverage File chr12 (47 of 50)\n",
      "\n",
      "Coverage File chr12 (48 of 50)\n",
      "\n",
      "Coverage File chr12 (49 of 50)\n",
      "\n",
      "Coverage File chr12 (50 of 50)\n",
      "\n",
      "Kmer Bias chr13 (5 of 24)\n",
      "\n",
      "chr13 \n",
      "Coverage File chr13 (1 of 50)\n",
      "\n",
      "Coverage File chr13 (2 of 50)\n",
      "\n",
      "Coverage File chr13 (3 of 50)\n",
      "\n",
      "Coverage File chr13 (4 of 50)\n",
      "\n",
      "Coverage File chr13 (5 of 50)\n",
      "\n",
      "Coverage File chr13 (6 of 50)\n",
      "\n",
      "Coverage File chr13 (7 of 50)\n",
      "\n",
      "Coverage File chr13 (8 of 50)\n",
      "\n",
      "Coverage File chr13 (9 of 50)\n",
      "\n",
      "Coverage File chr13 (10 of 50)\n",
      "\n",
      "Coverage File chr13 (11 of 50)\n",
      "\n",
      "Coverage File chr13 (12 of 50)\n",
      "\n",
      "Coverage File chr13 (13 of 50)\n",
      "\n",
      "Coverage File chr13 (14 of 50)\n",
      "\n",
      "Coverage File chr13 (15 of 50)\n",
      "\n",
      "Coverage File chr13 (16 of 50)\n",
      "\n",
      "Coverage File chr13 (17 of 50)\n",
      "\n",
      "Coverage File chr13 (18 of 50)\n",
      "\n",
      "Coverage File chr13 (19 of 50)\n",
      "\n",
      "Coverage File chr13 (20 of 50)\n",
      "\n",
      "Coverage File chr13 (21 of 50)\n",
      "\n",
      "Coverage File chr13 (22 of 50)\n",
      "\n",
      "Coverage File chr13 (23 of 50)\n",
      "\n",
      "Coverage File chr13 (24 of 50)\n",
      "\n",
      "Coverage File chr13 (25 of 50)\n",
      "\n",
      "Coverage File chr13 (26 of 50)\n",
      "\n",
      "Coverage File chr13 (27 of 50)\n",
      "\n",
      "Coverage File chr13 (28 of 50)\n",
      "\n",
      "Coverage File chr13 (29 of 50)\n",
      "\n",
      "Coverage File chr13 (30 of 50)\n",
      "\n",
      "Coverage File chr13 (31 of 50)\n",
      "\n",
      "Coverage File chr13 (32 of 50)\n",
      "\n",
      "Coverage File chr13 (33 of 50)\n",
      "\n",
      "Coverage File chr13 (34 of 50)\n",
      "\n",
      "Coverage File chr13 (35 of 50)\n",
      "\n",
      "Coverage File chr13 (36 of 50)\n",
      "\n",
      "Coverage File chr13 (37 of 50)\n",
      "\n",
      "Coverage File chr13 (38 of 50)\n",
      "\n",
      "Coverage File chr13 (39 of 50)\n",
      "\n",
      "Coverage File chr13 (40 of 50)\n",
      "\n",
      "Coverage File chr13 (41 of 50)\n",
      "\n",
      "Coverage File chr13 (42 of 50)\n",
      "\n",
      "Coverage File chr13 (43 of 50)\n",
      "\n",
      "Coverage File chr13 (44 of 50)\n",
      "\n",
      "Coverage File chr13 (45 of 50)\n",
      "\n",
      "Coverage File chr13 (46 of 50)\n",
      "\n",
      "Coverage File chr13 (47 of 50)\n",
      "\n",
      "Coverage File chr13 (48 of 50)\n",
      "\n",
      "Coverage File chr13 (49 of 50)\n",
      "\n",
      "Coverage File chr13 (50 of 50)\n",
      "\n",
      "Kmer Bias chr14 (6 of 24)\n",
      "\n",
      "chr14 \n",
      "Coverage File chr14 (1 of 50)\n",
      "\n",
      "Coverage File chr14 (2 of 50)\n",
      "\n",
      "Coverage File chr14 (3 of 50)\n",
      "\n",
      "Coverage File chr14 (4 of 50)\n",
      "\n",
      "Coverage File chr14 (5 of 50)\n",
      "\n",
      "Coverage File chr14 (6 of 50)\n",
      "\n",
      "Coverage File chr14 (7 of 50)\n",
      "\n",
      "Coverage File chr14 (8 of 50)\n",
      "\n",
      "Coverage File chr14 (9 of 50)\n",
      "\n",
      "Coverage File chr14 (10 of 50)\n",
      "\n",
      "Coverage File chr14 (11 of 50)\n",
      "\n",
      "Coverage File chr14 (12 of 50)\n",
      "\n",
      "Coverage File chr14 (13 of 50)\n",
      "\n",
      "Coverage File chr14 (14 of 50)\n",
      "\n",
      "Coverage File chr14 (15 of 50)\n",
      "\n",
      "Coverage File chr14 (16 of 50)\n",
      "\n",
      "Coverage File chr14 (17 of 50)\n",
      "\n",
      "Coverage File chr14 (18 of 50)\n",
      "\n",
      "Coverage File chr14 (19 of 50)\n",
      "\n",
      "Coverage File chr14 (20 of 50)\n",
      "\n",
      "Coverage File chr14 (21 of 50)\n",
      "\n",
      "Coverage File chr14 (22 of 50)\n",
      "\n",
      "Coverage File chr14 (23 of 50)\n",
      "\n",
      "Coverage File chr14 (24 of 50)\n",
      "\n",
      "Coverage File chr14 (25 of 50)\n",
      "\n",
      "Coverage File chr14 (26 of 50)\n",
      "\n",
      "Coverage File chr14 (27 of 50)\n",
      "\n",
      "Coverage File chr14 (28 of 50)\n",
      "\n",
      "Coverage File chr14 (29 of 50)\n",
      "\n",
      "Coverage File chr14 (30 of 50)\n",
      "\n",
      "Coverage File chr14 (31 of 50)\n",
      "\n",
      "Coverage File chr14 (32 of 50)\n",
      "\n",
      "Coverage File chr14 (33 of 50)\n",
      "\n",
      "Coverage File chr14 (34 of 50)\n",
      "\n",
      "Coverage File chr14 (35 of 50)\n",
      "\n",
      "Coverage File chr14 (36 of 50)\n",
      "\n",
      "Coverage File chr14 (37 of 50)\n",
      "\n",
      "Coverage File chr14 (38 of 50)\n",
      "\n",
      "Coverage File chr14 (39 of 50)\n",
      "\n",
      "Coverage File chr14 (40 of 50)\n",
      "\n",
      "Coverage File chr14 (41 of 50)\n",
      "\n",
      "Coverage File chr14 (42 of 50)\n",
      "\n",
      "Coverage File chr14 (43 of 50)\n",
      "\n",
      "Coverage File chr14 (44 of 50)\n",
      "\n",
      "Coverage File chr14 (45 of 50)\n",
      "\n",
      "Coverage File chr14 (46 of 50)\n",
      "\n",
      "Coverage File chr14 (47 of 50)\n",
      "\n",
      "Coverage File chr14 (48 of 50)\n",
      "\n",
      "Coverage File chr14 (49 of 50)\n",
      "\n",
      "Coverage File chr14 (50 of 50)\n",
      "\n",
      "Kmer Bias chr15 (7 of 24)\n",
      "\n",
      "chr15 \n",
      "Coverage File chr15 (1 of 50)\n",
      "\n",
      "Coverage File chr15 (2 of 50)\n",
      "\n",
      "Coverage File chr15 (3 of 50)\n",
      "\n",
      "Coverage File chr15 (4 of 50)\n",
      "\n",
      "Coverage File chr15 (5 of 50)\n",
      "\n",
      "Coverage File chr15 (6 of 50)\n",
      "\n",
      "Coverage File chr15 (7 of 50)\n",
      "\n",
      "Coverage File chr15 (8 of 50)\n",
      "\n",
      "Coverage File chr15 (9 of 50)\n",
      "\n",
      "Coverage File chr15 (10 of 50)\n",
      "\n",
      "Coverage File chr15 (11 of 50)\n",
      "\n",
      "Coverage File chr15 (12 of 50)\n",
      "\n",
      "Coverage File chr15 (13 of 50)\n",
      "\n",
      "Coverage File chr15 (14 of 50)\n",
      "\n",
      "Coverage File chr15 (15 of 50)\n",
      "\n",
      "Coverage File chr15 (16 of 50)\n",
      "\n",
      "Coverage File chr15 (17 of 50)\n",
      "\n",
      "Coverage File chr15 (18 of 50)\n",
      "\n",
      "Coverage File chr15 (19 of 50)\n",
      "\n",
      "Coverage File chr15 (20 of 50)\n",
      "\n",
      "Coverage File chr15 (21 of 50)\n",
      "\n",
      "Coverage File chr15 (22 of 50)\n",
      "\n",
      "Coverage File chr15 (23 of 50)\n",
      "\n",
      "Coverage File chr15 (24 of 50)\n",
      "\n",
      "Coverage File chr15 (25 of 50)\n",
      "\n",
      "Coverage File chr15 (26 of 50)\n",
      "\n",
      "Coverage File chr15 (27 of 50)\n",
      "\n",
      "Coverage File chr15 (28 of 50)\n",
      "\n",
      "Coverage File chr15 (29 of 50)\n",
      "\n",
      "Coverage File chr15 (30 of 50)\n",
      "\n",
      "Coverage File chr15 (31 of 50)\n",
      "\n",
      "Coverage File chr15 (32 of 50)\n",
      "\n",
      "Coverage File chr15 (33 of 50)\n",
      "\n",
      "Coverage File chr15 (34 of 50)\n",
      "\n",
      "Coverage File chr15 (35 of 50)\n",
      "\n",
      "Coverage File chr15 (36 of 50)\n",
      "\n",
      "Coverage File chr15 (37 of 50)\n",
      "\n",
      "Coverage File chr15 (38 of 50)\n",
      "\n",
      "Coverage File chr15 (39 of 50)\n",
      "\n",
      "Coverage File chr15 (40 of 50)\n",
      "\n",
      "Coverage File chr15 (41 of 50)\n",
      "\n",
      "Coverage File chr15 (42 of 50)\n",
      "\n",
      "Coverage File chr15 (43 of 50)\n",
      "\n",
      "Coverage File chr15 (44 of 50)\n",
      "\n",
      "Coverage File chr15 (45 of 50)\n",
      "\n",
      "Coverage File chr15 (46 of 50)\n",
      "\n",
      "Coverage File chr15 (47 of 50)\n",
      "\n",
      "Coverage File chr15 (48 of 50)\n",
      "\n",
      "Coverage File chr15 (49 of 50)\n",
      "\n",
      "Coverage File chr15 (50 of 50)\n",
      "\n",
      "Kmer Bias chr16 (8 of 24)\n",
      "\n",
      "chr16 \n",
      "Coverage File chr16 (1 of 50)\n",
      "\n",
      "Coverage File chr16 (2 of 50)\n",
      "\n",
      "Coverage File chr16 (3 of 50)\n",
      "\n",
      "Coverage File chr16 (4 of 50)\n",
      "\n",
      "Coverage File chr16 (5 of 50)\n",
      "\n",
      "Coverage File chr16 (6 of 50)\n",
      "\n",
      "Coverage File chr16 (7 of 50)\n",
      "\n",
      "Coverage File chr16 (8 of 50)\n",
      "\n",
      "Coverage File chr16 (9 of 50)\n",
      "\n",
      "Coverage File chr16 (10 of 50)\n",
      "\n",
      "Coverage File chr16 (11 of 50)\n",
      "\n",
      "Coverage File chr16 (12 of 50)\n",
      "\n",
      "Coverage File chr16 (13 of 50)\n",
      "\n",
      "Coverage File chr16 (14 of 50)\n",
      "\n",
      "Coverage File chr16 (15 of 50)\n",
      "\n",
      "Coverage File chr16 (16 of 50)\n",
      "\n",
      "Coverage File chr16 (17 of 50)\n",
      "\n",
      "Coverage File chr16 (18 of 50)\n",
      "\n",
      "Coverage File chr16 (19 of 50)\n",
      "\n",
      "Coverage File chr16 (20 of 50)\n",
      "\n",
      "Coverage File chr16 (21 of 50)\n",
      "\n",
      "Coverage File chr16 (22 of 50)\n",
      "\n",
      "Coverage File chr16 (23 of 50)\n",
      "\n",
      "Coverage File chr16 (24 of 50)\n",
      "\n",
      "Coverage File chr16 (25 of 50)\n",
      "\n",
      "Coverage File chr16 (26 of 50)\n",
      "\n",
      "Coverage File chr16 (27 of 50)\n",
      "\n",
      "Coverage File chr16 (28 of 50)\n",
      "\n",
      "Coverage File chr16 (29 of 50)\n",
      "\n",
      "Coverage File chr16 (30 of 50)\n",
      "\n",
      "Coverage File chr16 (31 of 50)\n",
      "\n",
      "Coverage File chr16 (32 of 50)\n",
      "\n",
      "Coverage File chr16 (33 of 50)\n",
      "\n",
      "Coverage File chr16 (34 of 50)\n",
      "\n",
      "Coverage File chr16 (35 of 50)\n",
      "\n",
      "Coverage File chr16 (36 of 50)\n",
      "\n",
      "Coverage File chr16 (37 of 50)\n",
      "\n",
      "Coverage File chr16 (38 of 50)\n",
      "\n",
      "Coverage File chr16 (39 of 50)\n",
      "\n",
      "Coverage File chr16 (40 of 50)\n",
      "\n",
      "Coverage File chr16 (41 of 50)\n",
      "\n",
      "Coverage File chr16 (42 of 50)\n",
      "\n",
      "Coverage File chr16 (43 of 50)\n",
      "\n",
      "Coverage File chr16 (44 of 50)\n",
      "\n",
      "Coverage File chr16 (45 of 50)\n",
      "\n",
      "Coverage File chr16 (46 of 50)\n",
      "\n",
      "Coverage File chr16 (47 of 50)\n",
      "\n",
      "Coverage File chr16 (48 of 50)\n",
      "\n",
      "Coverage File chr16 (49 of 50)\n",
      "\n",
      "Coverage File chr16 (50 of 50)\n",
      "\n",
      "Kmer Bias chr17 (9 of 24)\n",
      "\n",
      "chr17 \n",
      "Coverage File chr17 (1 of 50)\n",
      "\n",
      "Coverage File chr17 (2 of 50)\n",
      "\n",
      "Coverage File chr17 (3 of 50)\n",
      "\n",
      "Coverage File chr17 (4 of 50)\n",
      "\n",
      "Coverage File chr17 (5 of 50)\n",
      "\n",
      "Coverage File chr17 (6 of 50)\n",
      "\n",
      "Coverage File chr17 (7 of 50)\n",
      "\n",
      "Coverage File chr17 (8 of 50)\n",
      "\n",
      "Coverage File chr17 (9 of 50)\n",
      "\n",
      "Coverage File chr17 (10 of 50)\n",
      "\n",
      "Coverage File chr17 (11 of 50)\n",
      "\n",
      "Coverage File chr17 (12 of 50)\n",
      "\n",
      "Coverage File chr17 (13 of 50)\n",
      "\n",
      "Coverage File chr17 (14 of 50)\n",
      "\n",
      "Coverage File chr17 (15 of 50)\n",
      "\n",
      "Coverage File chr17 (16 of 50)\n",
      "\n",
      "Coverage File chr17 (17 of 50)\n",
      "\n",
      "Coverage File chr17 (18 of 50)\n",
      "\n",
      "Coverage File chr17 (19 of 50)\n",
      "\n",
      "Coverage File chr17 (20 of 50)\n",
      "\n",
      "Coverage File chr17 (21 of 50)\n",
      "\n",
      "Coverage File chr17 (22 of 50)\n",
      "\n",
      "Coverage File chr17 (23 of 50)\n",
      "\n",
      "Coverage File chr17 (24 of 50)\n",
      "\n",
      "Coverage File chr17 (25 of 50)\n",
      "\n",
      "Coverage File chr17 (26 of 50)\n",
      "\n",
      "Coverage File chr17 (27 of 50)\n",
      "\n",
      "Coverage File chr17 (28 of 50)\n",
      "\n",
      "Coverage File chr17 (29 of 50)\n",
      "\n",
      "Coverage File chr17 (30 of 50)\n",
      "\n",
      "Coverage File chr17 (31 of 50)\n",
      "\n",
      "Coverage File chr17 (32 of 50)\n",
      "\n",
      "Coverage File chr17 (33 of 50)\n",
      "\n",
      "Coverage File chr17 (34 of 50)\n",
      "\n",
      "Coverage File chr17 (35 of 50)\n",
      "\n",
      "Coverage File chr17 (36 of 50)\n",
      "\n",
      "Coverage File chr17 (37 of 50)\n",
      "\n",
      "Coverage File chr17 (38 of 50)\n",
      "\n",
      "Coverage File chr17 (39 of 50)\n",
      "\n",
      "Coverage File chr17 (40 of 50)\n",
      "\n",
      "Coverage File chr17 (41 of 50)\n",
      "\n",
      "Coverage File chr17 (42 of 50)\n",
      "\n",
      "Coverage File chr17 (43 of 50)\n",
      "\n",
      "Coverage File chr17 (44 of 50)\n",
      "\n",
      "Coverage File chr17 (45 of 50)\n",
      "\n",
      "Coverage File chr17 (46 of 50)\n",
      "\n",
      "Coverage File chr17 (47 of 50)\n",
      "\n",
      "Coverage File chr17 (48 of 50)\n",
      "\n",
      "Coverage File chr17 (49 of 50)\n",
      "\n",
      "Coverage File chr17 (50 of 50)\n",
      "\n",
      "Kmer Bias chr18 (10 of 24)\n",
      "\n",
      "chr18 \n",
      "Coverage File chr18 (1 of 50)\n",
      "\n",
      "Coverage File chr18 (2 of 50)\n",
      "\n",
      "Coverage File chr18 (3 of 50)\n",
      "\n",
      "Coverage File chr18 (4 of 50)\n",
      "\n",
      "Coverage File chr18 (5 of 50)\n",
      "\n",
      "Coverage File chr18 (6 of 50)\n",
      "\n",
      "Coverage File chr18 (7 of 50)\n",
      "\n",
      "Coverage File chr18 (8 of 50)\n",
      "\n",
      "Coverage File chr18 (9 of 50)\n",
      "\n",
      "Coverage File chr18 (10 of 50)\n",
      "\n",
      "Coverage File chr18 (11 of 50)\n",
      "\n",
      "Coverage File chr18 (12 of 50)\n",
      "\n",
      "Coverage File chr18 (13 of 50)\n",
      "\n",
      "Coverage File chr18 (14 of 50)\n",
      "\n",
      "Coverage File chr18 (15 of 50)\n",
      "\n",
      "Coverage File chr18 (16 of 50)\n",
      "\n",
      "Coverage File chr18 (17 of 50)\n",
      "\n",
      "Coverage File chr18 (18 of 50)\n",
      "\n",
      "Coverage File chr18 (19 of 50)\n",
      "\n",
      "Coverage File chr18 (20 of 50)\n",
      "\n",
      "Coverage File chr18 (21 of 50)\n",
      "\n",
      "Coverage File chr18 (22 of 50)\n",
      "\n",
      "Coverage File chr18 (23 of 50)\n",
      "\n",
      "Coverage File chr18 (24 of 50)\n",
      "\n",
      "Coverage File chr18 (25 of 50)\n",
      "\n",
      "Coverage File chr18 (26 of 50)\n",
      "\n",
      "Coverage File chr18 (27 of 50)\n",
      "\n",
      "Coverage File chr18 (28 of 50)\n",
      "\n",
      "Coverage File chr18 (29 of 50)\n",
      "\n",
      "Coverage File chr18 (30 of 50)\n",
      "\n",
      "Coverage File chr18 (31 of 50)\n",
      "\n",
      "Coverage File chr18 (32 of 50)\n",
      "\n",
      "Coverage File chr18 (33 of 50)\n",
      "\n",
      "Coverage File chr18 (34 of 50)\n",
      "\n",
      "Coverage File chr18 (35 of 50)\n",
      "\n",
      "Coverage File chr18 (36 of 50)\n",
      "\n",
      "Coverage File chr18 (37 of 50)\n",
      "\n",
      "Coverage File chr18 (38 of 50)\n",
      "\n",
      "Coverage File chr18 (39 of 50)\n",
      "\n",
      "Coverage File chr18 (40 of 50)\n",
      "\n",
      "Coverage File chr18 (41 of 50)\n",
      "\n",
      "Coverage File chr18 (42 of 50)\n",
      "\n",
      "Coverage File chr18 (43 of 50)\n",
      "\n",
      "Coverage File chr18 (44 of 50)\n",
      "\n",
      "Coverage File chr18 (45 of 50)\n",
      "\n",
      "Coverage File chr18 (46 of 50)\n",
      "\n",
      "Coverage File chr18 (47 of 50)\n",
      "\n",
      "Coverage File chr18 (48 of 50)\n",
      "\n",
      "Coverage File chr18 (49 of 50)\n",
      "\n",
      "Coverage File chr18 (50 of 50)\n",
      "\n",
      "Kmer Bias chr19 (11 of 24)\n",
      "\n",
      "chr19 \n",
      "Coverage File chr19 (1 of 50)\n",
      "\n",
      "Coverage File chr19 (2 of 50)\n",
      "\n",
      "Coverage File chr19 (3 of 50)\n",
      "\n",
      "Coverage File chr19 (4 of 50)\n",
      "\n",
      "Coverage File chr19 (5 of 50)\n",
      "\n",
      "Coverage File chr19 (6 of 50)\n",
      "\n",
      "Coverage File chr19 (7 of 50)\n",
      "\n",
      "Coverage File chr19 (8 of 50)\n",
      "\n",
      "Coverage File chr19 (9 of 50)\n",
      "\n",
      "Coverage File chr19 (10 of 50)\n",
      "\n",
      "Coverage File chr19 (11 of 50)\n",
      "\n",
      "Coverage File chr19 (12 of 50)\n",
      "\n",
      "Coverage File chr19 (13 of 50)\n",
      "\n",
      "Coverage File chr19 (14 of 50)\n",
      "\n",
      "Coverage File chr19 (15 of 50)\n",
      "\n",
      "Coverage File chr19 (16 of 50)\n",
      "\n",
      "Coverage File chr19 (17 of 50)\n",
      "\n",
      "Coverage File chr19 (18 of 50)\n",
      "\n",
      "Coverage File chr19 (19 of 50)\n",
      "\n",
      "Coverage File chr19 (20 of 50)\n",
      "\n",
      "Coverage File chr19 (21 of 50)\n",
      "\n",
      "Coverage File chr19 (22 of 50)\n",
      "\n",
      "Coverage File chr19 (23 of 50)\n",
      "\n",
      "Coverage File chr19 (24 of 50)\n",
      "\n",
      "Coverage File chr19 (25 of 50)\n",
      "\n",
      "Coverage File chr19 (26 of 50)\n",
      "\n",
      "Coverage File chr19 (27 of 50)\n",
      "\n",
      "Coverage File chr19 (28 of 50)\n",
      "\n",
      "Coverage File chr19 (29 of 50)\n",
      "\n",
      "Coverage File chr19 (30 of 50)\n",
      "\n",
      "Coverage File chr19 (31 of 50)\n",
      "\n",
      "Coverage File chr19 (32 of 50)\n",
      "\n",
      "Coverage File chr19 (33 of 50)\n",
      "\n",
      "Coverage File chr19 (34 of 50)\n",
      "\n",
      "Coverage File chr19 (35 of 50)\n",
      "\n",
      "Coverage File chr19 (36 of 50)\n",
      "\n",
      "Coverage File chr19 (37 of 50)\n",
      "\n",
      "Coverage File chr19 (38 of 50)\n",
      "\n",
      "Coverage File chr19 (39 of 50)\n",
      "\n",
      "Coverage File chr19 (40 of 50)\n",
      "\n",
      "Coverage File chr19 (41 of 50)\n",
      "\n",
      "Coverage File chr19 (42 of 50)\n",
      "\n",
      "Coverage File chr19 (43 of 50)\n",
      "\n",
      "Coverage File chr19 (44 of 50)\n",
      "\n",
      "Coverage File chr19 (45 of 50)\n",
      "\n",
      "Coverage File chr19 (46 of 50)\n",
      "\n",
      "Coverage File chr19 (47 of 50)\n",
      "\n",
      "Coverage File chr19 (48 of 50)\n",
      "\n",
      "Coverage File chr19 (49 of 50)\n",
      "\n",
      "Coverage File chr19 (50 of 50)\n",
      "\n",
      "Kmer Bias chr2 (12 of 24)\n",
      "\n",
      "chr2 \n",
      "Coverage File chr2 (1 of 50)\n",
      "\n",
      "Coverage File chr2 (2 of 50)\n",
      "\n",
      "Coverage File chr2 (3 of 50)\n",
      "\n",
      "Coverage File chr2 (4 of 50)\n",
      "\n",
      "Coverage File chr2 (5 of 50)\n",
      "\n",
      "Coverage File chr2 (6 of 50)\n",
      "\n",
      "Coverage File chr2 (7 of 50)\n",
      "\n",
      "Coverage File chr2 (8 of 50)\n",
      "\n",
      "Coverage File chr2 (9 of 50)\n",
      "\n",
      "Coverage File chr2 (10 of 50)\n",
      "\n",
      "Coverage File chr2 (11 of 50)\n",
      "\n",
      "Coverage File chr2 (12 of 50)\n",
      "\n",
      "Coverage File chr2 (13 of 50)\n",
      "\n",
      "Coverage File chr2 (14 of 50)\n",
      "\n",
      "Coverage File chr2 (15 of 50)\n",
      "\n",
      "Coverage File chr2 (16 of 50)\n",
      "\n",
      "Coverage File chr2 (17 of 50)\n",
      "\n",
      "Coverage File chr2 (18 of 50)\n",
      "\n",
      "Coverage File chr2 (19 of 50)\n",
      "\n",
      "Coverage File chr2 (20 of 50)\n",
      "\n",
      "Coverage File chr2 (21 of 50)\n",
      "\n",
      "Coverage File chr2 (22 of 50)\n",
      "\n",
      "Coverage File chr2 (23 of 50)\n",
      "\n",
      "Coverage File chr2 (24 of 50)\n",
      "\n",
      "Coverage File chr2 (25 of 50)\n",
      "\n",
      "Coverage File chr2 (26 of 50)\n",
      "\n",
      "Coverage File chr2 (27 of 50)\n",
      "\n",
      "Coverage File chr2 (28 of 50)\n",
      "\n",
      "Coverage File chr2 (29 of 50)\n",
      "\n",
      "Coverage File chr2 (30 of 50)\n",
      "\n",
      "Coverage File chr2 (31 of 50)\n",
      "\n",
      "Coverage File chr2 (32 of 50)\n",
      "\n",
      "Coverage File chr2 (33 of 50)\n",
      "\n",
      "Coverage File chr2 (34 of 50)\n",
      "\n",
      "Coverage File chr2 (35 of 50)\n",
      "\n",
      "Coverage File chr2 (36 of 50)\n",
      "\n",
      "Coverage File chr2 (37 of 50)\n",
      "\n",
      "Coverage File chr2 (38 of 50)\n",
      "\n",
      "Coverage File chr2 (39 of 50)\n",
      "\n",
      "Coverage File chr2 (40 of 50)\n",
      "\n",
      "Coverage File chr2 (41 of 50)\n",
      "\n",
      "Coverage File chr2 (42 of 50)\n",
      "\n",
      "Coverage File chr2 (43 of 50)\n",
      "\n",
      "Coverage File chr2 (44 of 50)\n",
      "\n",
      "Coverage File chr2 (45 of 50)\n",
      "\n",
      "Coverage File chr2 (46 of 50)\n",
      "\n",
      "Coverage File chr2 (47 of 50)\n",
      "\n",
      "Coverage File chr2 (48 of 50)\n",
      "\n",
      "Coverage File chr2 (49 of 50)\n",
      "\n",
      "Coverage File chr2 (50 of 50)\n",
      "\n",
      "Kmer Bias chr20 (13 of 24)\n",
      "\n",
      "chr20 \n",
      "Coverage File chr20 (1 of 50)\n",
      "\n",
      "Coverage File chr20 (2 of 50)\n",
      "\n",
      "Coverage File chr20 (3 of 50)\n",
      "\n",
      "Coverage File chr20 (4 of 50)\n",
      "\n",
      "Coverage File chr20 (5 of 50)\n",
      "\n",
      "Coverage File chr20 (6 of 50)\n",
      "\n",
      "Coverage File chr20 (7 of 50)\n",
      "\n",
      "Coverage File chr20 (8 of 50)\n",
      "\n",
      "Coverage File chr20 (9 of 50)\n",
      "\n",
      "Coverage File chr20 (10 of 50)\n",
      "\n",
      "Coverage File chr20 (11 of 50)\n",
      "\n",
      "Coverage File chr20 (12 of 50)\n",
      "\n",
      "Coverage File chr20 (13 of 50)\n",
      "\n",
      "Coverage File chr20 (14 of 50)\n",
      "\n",
      "Coverage File chr20 (15 of 50)\n",
      "\n",
      "Coverage File chr20 (16 of 50)\n",
      "\n",
      "Coverage File chr20 (17 of 50)\n",
      "\n",
      "Coverage File chr20 (18 of 50)\n",
      "\n",
      "Coverage File chr20 (19 of 50)\n",
      "\n",
      "Coverage File chr20 (20 of 50)\n",
      "\n",
      "Coverage File chr20 (21 of 50)\n",
      "\n",
      "Coverage File chr20 (22 of 50)\n",
      "\n",
      "Coverage File chr20 (23 of 50)\n",
      "\n",
      "Coverage File chr20 (24 of 50)\n",
      "\n",
      "Coverage File chr20 (25 of 50)\n",
      "\n",
      "Coverage File chr20 (26 of 50)\n",
      "\n",
      "Coverage File chr20 (27 of 50)\n",
      "\n",
      "Coverage File chr20 (28 of 50)\n",
      "\n",
      "Coverage File chr20 (29 of 50)\n",
      "\n",
      "Coverage File chr20 (30 of 50)\n",
      "\n",
      "Coverage File chr20 (31 of 50)\n",
      "\n",
      "Coverage File chr20 (32 of 50)\n",
      "\n",
      "Coverage File chr20 (33 of 50)\n",
      "\n",
      "Coverage File chr20 (34 of 50)\n",
      "\n",
      "Coverage File chr20 (35 of 50)\n",
      "\n",
      "Coverage File chr20 (36 of 50)\n",
      "\n",
      "Coverage File chr20 (37 of 50)\n",
      "\n",
      "Coverage File chr20 (38 of 50)\n",
      "\n",
      "Coverage File chr20 (39 of 50)\n",
      "\n",
      "Coverage File chr20 (40 of 50)\n",
      "\n",
      "Coverage File chr20 (41 of 50)\n",
      "\n",
      "Coverage File chr20 (42 of 50)\n",
      "\n",
      "Coverage File chr20 (43 of 50)\n",
      "\n",
      "Coverage File chr20 (44 of 50)\n",
      "\n",
      "Coverage File chr20 (45 of 50)\n",
      "\n",
      "Coverage File chr20 (46 of 50)\n",
      "\n",
      "Coverage File chr20 (47 of 50)\n",
      "\n",
      "Coverage File chr20 (48 of 50)\n",
      "\n",
      "Coverage File chr20 (49 of 50)\n",
      "\n",
      "Coverage File chr20 (50 of 50)\n",
      "\n",
      "Kmer Bias chr21 (14 of 24)\n",
      "\n",
      "chr21 \n",
      "Coverage File chr21 (1 of 50)\n",
      "\n",
      "Coverage File chr21 (2 of 50)\n",
      "\n",
      "Coverage File chr21 (3 of 50)\n",
      "\n",
      "Coverage File chr21 (4 of 50)\n",
      "\n",
      "Coverage File chr21 (5 of 50)\n",
      "\n",
      "Coverage File chr21 (6 of 50)\n",
      "\n",
      "Coverage File chr21 (7 of 50)\n",
      "\n",
      "Coverage File chr21 (8 of 50)\n",
      "\n",
      "Coverage File chr21 (9 of 50)\n",
      "\n",
      "Coverage File chr21 (10 of 50)\n",
      "\n",
      "Coverage File chr21 (11 of 50)\n",
      "\n",
      "Coverage File chr21 (12 of 50)\n",
      "\n",
      "Coverage File chr21 (13 of 50)\n",
      "\n",
      "Coverage File chr21 (14 of 50)\n",
      "\n",
      "Coverage File chr21 (15 of 50)\n",
      "\n",
      "Coverage File chr21 (16 of 50)\n",
      "\n",
      "Coverage File chr21 (17 of 50)\n",
      "\n",
      "Coverage File chr21 (18 of 50)\n",
      "\n",
      "Coverage File chr21 (19 of 50)\n",
      "\n",
      "Coverage File chr21 (20 of 50)\n",
      "\n",
      "Coverage File chr21 (21 of 50)\n",
      "\n",
      "Coverage File chr21 (22 of 50)\n",
      "\n",
      "Coverage File chr21 (23 of 50)\n",
      "\n",
      "Coverage File chr21 (24 of 50)\n",
      "\n",
      "Coverage File chr21 (25 of 50)\n",
      "\n",
      "Coverage File chr21 (26 of 50)\n",
      "\n",
      "Coverage File chr21 (27 of 50)\n",
      "\n",
      "Coverage File chr21 (28 of 50)\n",
      "\n",
      "Coverage File chr21 (29 of 50)\n",
      "\n",
      "Coverage File chr21 (30 of 50)\n",
      "\n",
      "Coverage File chr21 (31 of 50)\n",
      "\n",
      "Coverage File chr21 (32 of 50)\n",
      "\n",
      "Coverage File chr21 (33 of 50)\n",
      "\n",
      "Coverage File chr21 (34 of 50)\n",
      "\n",
      "Coverage File chr21 (35 of 50)\n",
      "\n",
      "Coverage File chr21 (36 of 50)\n",
      "\n",
      "Coverage File chr21 (37 of 50)\n",
      "\n",
      "Coverage File chr21 (38 of 50)\n",
      "\n",
      "Coverage File chr21 (39 of 50)\n",
      "\n",
      "Coverage File chr21 (40 of 50)\n",
      "\n",
      "Coverage File chr21 (41 of 50)\n",
      "\n",
      "Coverage File chr21 (42 of 50)\n",
      "\n",
      "Coverage File chr21 (43 of 50)\n",
      "\n",
      "Coverage File chr21 (44 of 50)\n",
      "\n",
      "Coverage File chr21 (45 of 50)\n",
      "\n",
      "Coverage File chr21 (46 of 50)\n",
      "\n",
      "Coverage File chr21 (47 of 50)\n",
      "\n",
      "Coverage File chr21 (48 of 50)\n",
      "\n",
      "Coverage File chr21 (49 of 50)\n",
      "\n",
      "Coverage File chr21 (50 of 50)\n",
      "\n",
      "Kmer Bias chr22 (15 of 24)\n",
      "\n",
      "chr22 \n",
      "Coverage File chr22 (1 of 50)\n",
      "\n",
      "Coverage File chr22 (2 of 50)\n",
      "\n",
      "Coverage File chr22 (3 of 50)\n",
      "\n",
      "Coverage File chr22 (4 of 50)\n",
      "\n",
      "Coverage File chr22 (5 of 50)\n",
      "\n",
      "Coverage File chr22 (6 of 50)\n",
      "\n",
      "Coverage File chr22 (7 of 50)\n",
      "\n",
      "Coverage File chr22 (8 of 50)\n",
      "\n",
      "Coverage File chr22 (9 of 50)\n",
      "\n",
      "Coverage File chr22 (10 of 50)\n",
      "\n",
      "Coverage File chr22 (11 of 50)\n",
      "\n",
      "Coverage File chr22 (12 of 50)\n",
      "\n",
      "Coverage File chr22 (13 of 50)\n",
      "\n",
      "Coverage File chr22 (14 of 50)\n",
      "\n",
      "Coverage File chr22 (15 of 50)\n",
      "\n",
      "Coverage File chr22 (16 of 50)\n",
      "\n",
      "Coverage File chr22 (17 of 50)\n",
      "\n",
      "Coverage File chr22 (18 of 50)\n",
      "\n",
      "Coverage File chr22 (19 of 50)\n",
      "\n",
      "Coverage File chr22 (20 of 50)\n",
      "\n",
      "Coverage File chr22 (21 of 50)\n",
      "\n",
      "Coverage File chr22 (22 of 50)\n",
      "\n",
      "Coverage File chr22 (23 of 50)\n",
      "\n",
      "Coverage File chr22 (24 of 50)\n",
      "\n",
      "Coverage File chr22 (25 of 50)\n",
      "\n",
      "Coverage File chr22 (26 of 50)\n",
      "\n",
      "Coverage File chr22 (27 of 50)\n",
      "\n",
      "Coverage File chr22 (28 of 50)\n",
      "\n",
      "Coverage File chr22 (29 of 50)\n",
      "\n",
      "Coverage File chr22 (30 of 50)\n",
      "\n",
      "Coverage File chr22 (31 of 50)\n",
      "\n",
      "Coverage File chr22 (32 of 50)\n",
      "\n",
      "Coverage File chr22 (33 of 50)\n",
      "\n",
      "Coverage File chr22 (34 of 50)\n",
      "\n",
      "Coverage File chr22 (35 of 50)\n",
      "\n",
      "Coverage File chr22 (36 of 50)\n",
      "\n",
      "Coverage File chr22 (37 of 50)\n",
      "\n",
      "Coverage File chr22 (38 of 50)\n",
      "\n",
      "Coverage File chr22 (39 of 50)\n",
      "\n",
      "Coverage File chr22 (40 of 50)\n",
      "\n",
      "Coverage File chr22 (41 of 50)\n",
      "\n",
      "Coverage File chr22 (42 of 50)\n",
      "\n",
      "Coverage File chr22 (43 of 50)\n",
      "\n",
      "Coverage File chr22 (44 of 50)\n",
      "\n",
      "Coverage File chr22 (45 of 50)\n",
      "\n",
      "Coverage File chr22 (46 of 50)\n",
      "\n",
      "Coverage File chr22 (47 of 50)\n",
      "\n",
      "Coverage File chr22 (48 of 50)\n",
      "\n",
      "Coverage File chr22 (49 of 50)\n",
      "\n",
      "Coverage File chr22 (50 of 50)\n",
      "\n",
      "Kmer Bias chr3 (16 of 24)\n",
      "\n",
      "chr3 \n",
      "Coverage File chr3 (1 of 50)\n",
      "\n",
      "Coverage File chr3 (2 of 50)\n",
      "\n",
      "Coverage File chr3 (3 of 50)\n",
      "\n",
      "Coverage File chr3 (4 of 50)\n",
      "\n",
      "Coverage File chr3 (5 of 50)\n",
      "\n",
      "Coverage File chr3 (6 of 50)\n",
      "\n",
      "Coverage File chr3 (7 of 50)\n",
      "\n",
      "Coverage File chr3 (8 of 50)\n",
      "\n",
      "Coverage File chr3 (9 of 50)\n",
      "\n",
      "Coverage File chr3 (10 of 50)\n",
      "\n",
      "Coverage File chr3 (11 of 50)\n",
      "\n",
      "Coverage File chr3 (12 of 50)\n",
      "\n",
      "Coverage File chr3 (13 of 50)\n",
      "\n",
      "Coverage File chr3 (14 of 50)\n",
      "\n",
      "Coverage File chr3 (15 of 50)\n",
      "\n",
      "Coverage File chr3 (16 of 50)\n",
      "\n",
      "Coverage File chr3 (17 of 50)\n",
      "\n",
      "Coverage File chr3 (18 of 50)\n",
      "\n",
      "Coverage File chr3 (19 of 50)\n",
      "\n",
      "Coverage File chr3 (20 of 50)\n",
      "\n",
      "Coverage File chr3 (21 of 50)\n",
      "\n",
      "Coverage File chr3 (22 of 50)\n",
      "\n",
      "Coverage File chr3 (23 of 50)\n",
      "\n",
      "Coverage File chr3 (24 of 50)\n",
      "\n",
      "Coverage File chr3 (25 of 50)\n",
      "\n",
      "Coverage File chr3 (26 of 50)\n",
      "\n",
      "Coverage File chr3 (27 of 50)\n",
      "\n",
      "Coverage File chr3 (28 of 50)\n",
      "\n",
      "Coverage File chr3 (29 of 50)\n",
      "\n",
      "Coverage File chr3 (30 of 50)\n",
      "\n",
      "Coverage File chr3 (31 of 50)\n",
      "\n",
      "Coverage File chr3 (32 of 50)\n",
      "\n",
      "Coverage File chr3 (33 of 50)\n",
      "\n",
      "Coverage File chr3 (34 of 50)\n",
      "\n",
      "Coverage File chr3 (35 of 50)\n",
      "\n",
      "Coverage File chr3 (36 of 50)\n",
      "\n",
      "Coverage File chr3 (37 of 50)\n",
      "\n",
      "Coverage File chr3 (38 of 50)\n",
      "\n",
      "Coverage File chr3 (39 of 50)\n",
      "\n",
      "Coverage File chr3 (40 of 50)\n",
      "\n",
      "Coverage File chr3 (41 of 50)\n",
      "\n",
      "Coverage File chr3 (42 of 50)\n",
      "\n",
      "Coverage File chr3 (43 of 50)\n",
      "\n",
      "Coverage File chr3 (44 of 50)\n",
      "\n",
      "Coverage File chr3 (45 of 50)\n",
      "\n",
      "Coverage File chr3 (46 of 50)\n",
      "\n",
      "Coverage File chr3 (47 of 50)\n",
      "\n",
      "Coverage File chr3 (48 of 50)\n",
      "\n",
      "Coverage File chr3 (49 of 50)\n",
      "\n",
      "Coverage File chr3 (50 of 50)\n",
      "\n",
      "Kmer Bias chr4 (17 of 24)\n",
      "\n",
      "chr4 \n",
      "Coverage File chr4 (1 of 50)\n",
      "\n",
      "Coverage File chr4 (2 of 50)\n",
      "\n",
      "Coverage File chr4 (3 of 50)\n",
      "\n",
      "Coverage File chr4 (4 of 50)\n",
      "\n",
      "Coverage File chr4 (5 of 50)\n",
      "\n",
      "Coverage File chr4 (6 of 50)\n",
      "\n",
      "Coverage File chr4 (7 of 50)\n",
      "\n",
      "Coverage File chr4 (8 of 50)\n",
      "\n",
      "Coverage File chr4 (9 of 50)\n",
      "\n",
      "Coverage File chr4 (10 of 50)\n",
      "\n",
      "Coverage File chr4 (11 of 50)\n",
      "\n",
      "Coverage File chr4 (12 of 50)\n",
      "\n",
      "Coverage File chr4 (13 of 50)\n",
      "\n",
      "Coverage File chr4 (14 of 50)\n",
      "\n",
      "Coverage File chr4 (15 of 50)\n",
      "\n",
      "Coverage File chr4 (16 of 50)\n",
      "\n",
      "Coverage File chr4 (17 of 50)\n",
      "\n",
      "Coverage File chr4 (18 of 50)\n",
      "\n",
      "Coverage File chr4 (19 of 50)\n",
      "\n",
      "Coverage File chr4 (20 of 50)\n",
      "\n",
      "Coverage File chr4 (21 of 50)\n",
      "\n",
      "Coverage File chr4 (22 of 50)\n",
      "\n",
      "Coverage File chr4 (23 of 50)\n",
      "\n",
      "Coverage File chr4 (24 of 50)\n",
      "\n",
      "Coverage File chr4 (25 of 50)\n",
      "\n",
      "Coverage File chr4 (26 of 50)\n",
      "\n",
      "Coverage File chr4 (27 of 50)\n",
      "\n",
      "Coverage File chr4 (28 of 50)\n",
      "\n",
      "Coverage File chr4 (29 of 50)\n",
      "\n",
      "Coverage File chr4 (30 of 50)\n",
      "\n",
      "Coverage File chr4 (31 of 50)\n",
      "\n",
      "Coverage File chr4 (32 of 50)\n",
      "\n",
      "Coverage File chr4 (33 of 50)\n",
      "\n",
      "Coverage File chr4 (34 of 50)\n",
      "\n",
      "Coverage File chr4 (35 of 50)\n",
      "\n",
      "Coverage File chr4 (36 of 50)\n",
      "\n",
      "Coverage File chr4 (37 of 50)\n",
      "\n",
      "Coverage File chr4 (38 of 50)\n",
      "\n",
      "Coverage File chr4 (39 of 50)\n",
      "\n",
      "Coverage File chr4 (40 of 50)\n",
      "\n",
      "Coverage File chr4 (41 of 50)\n",
      "\n",
      "Coverage File chr4 (42 of 50)\n",
      "\n",
      "Coverage File chr4 (43 of 50)\n",
      "\n",
      "Coverage File chr4 (44 of 50)\n",
      "\n",
      "Coverage File chr4 (45 of 50)\n",
      "\n",
      "Coverage File chr4 (46 of 50)\n",
      "\n",
      "Coverage File chr4 (47 of 50)\n",
      "\n",
      "Coverage File chr4 (48 of 50)\n",
      "\n",
      "Coverage File chr4 (49 of 50)\n",
      "\n",
      "Coverage File chr4 (50 of 50)\n",
      "\n",
      "Kmer Bias chr5 (18 of 24)\n",
      "\n",
      "chr5 \n",
      "Coverage File chr5 (1 of 50)\n",
      "\n",
      "Coverage File chr5 (2 of 50)\n",
      "\n",
      "Coverage File chr5 (3 of 50)\n",
      "\n",
      "Coverage File chr5 (4 of 50)\n",
      "\n",
      "Coverage File chr5 (5 of 50)\n",
      "\n",
      "Coverage File chr5 (6 of 50)\n",
      "\n",
      "Coverage File chr5 (7 of 50)\n",
      "\n",
      "Coverage File chr5 (8 of 50)\n",
      "\n",
      "Coverage File chr5 (9 of 50)\n",
      "\n",
      "Coverage File chr5 (10 of 50)\n",
      "\n",
      "Coverage File chr5 (11 of 50)\n",
      "\n",
      "Coverage File chr5 (12 of 50)\n",
      "\n",
      "Coverage File chr5 (13 of 50)\n",
      "\n",
      "Coverage File chr5 (14 of 50)\n",
      "\n",
      "Coverage File chr5 (15 of 50)\n",
      "\n",
      "Coverage File chr5 (16 of 50)\n",
      "\n",
      "Coverage File chr5 (17 of 50)\n",
      "\n",
      "Coverage File chr5 (18 of 50)\n",
      "\n",
      "Coverage File chr5 (19 of 50)\n",
      "\n",
      "Coverage File chr5 (20 of 50)\n",
      "\n",
      "Coverage File chr5 (21 of 50)\n",
      "\n",
      "Coverage File chr5 (22 of 50)\n",
      "\n",
      "Coverage File chr5 (23 of 50)\n",
      "\n",
      "Coverage File chr5 (24 of 50)\n",
      "\n",
      "Coverage File chr5 (25 of 50)\n",
      "\n",
      "Coverage File chr5 (26 of 50)\n",
      "\n",
      "Coverage File chr5 (27 of 50)\n",
      "\n",
      "Coverage File chr5 (28 of 50)\n",
      "\n",
      "Coverage File chr5 (29 of 50)\n",
      "\n",
      "Coverage File chr5 (30 of 50)\n",
      "\n",
      "Coverage File chr5 (31 of 50)\n",
      "\n",
      "Coverage File chr5 (32 of 50)\n",
      "\n",
      "Coverage File chr5 (33 of 50)\n",
      "\n",
      "Coverage File chr5 (34 of 50)\n",
      "\n",
      "Coverage File chr5 (35 of 50)\n",
      "\n",
      "Coverage File chr5 (36 of 50)\n",
      "\n",
      "Coverage File chr5 (37 of 50)\n",
      "\n",
      "Coverage File chr5 (38 of 50)\n",
      "\n",
      "Coverage File chr5 (39 of 50)\n",
      "\n",
      "Coverage File chr5 (40 of 50)\n",
      "\n",
      "Coverage File chr5 (41 of 50)\n",
      "\n",
      "Coverage File chr5 (42 of 50)\n",
      "\n",
      "Coverage File chr5 (43 of 50)\n",
      "\n",
      "Coverage File chr5 (44 of 50)\n",
      "\n",
      "Coverage File chr5 (45 of 50)\n",
      "\n",
      "Coverage File chr5 (46 of 50)\n",
      "\n",
      "Coverage File chr5 (47 of 50)\n",
      "\n",
      "Coverage File chr5 (48 of 50)\n",
      "\n",
      "Coverage File chr5 (49 of 50)\n",
      "\n",
      "Coverage File chr5 (50 of 50)\n",
      "\n",
      "Kmer Bias chr6 (19 of 24)\n",
      "\n",
      "chr6 \n",
      "Coverage File chr6 (1 of 50)\n",
      "\n",
      "Coverage File chr6 (2 of 50)\n",
      "\n",
      "Coverage File chr6 (3 of 50)\n",
      "\n",
      "Coverage File chr6 (4 of 50)\n",
      "\n",
      "Coverage File chr6 (5 of 50)\n",
      "\n",
      "Coverage File chr6 (6 of 50)\n",
      "\n",
      "Coverage File chr6 (7 of 50)\n",
      "\n",
      "Coverage File chr6 (8 of 50)\n",
      "\n",
      "Coverage File chr6 (9 of 50)\n",
      "\n",
      "Coverage File chr6 (10 of 50)\n",
      "\n",
      "Coverage File chr6 (11 of 50)\n",
      "\n",
      "Coverage File chr6 (12 of 50)\n",
      "\n",
      "Coverage File chr6 (13 of 50)\n",
      "\n",
      "Coverage File chr6 (14 of 50)\n",
      "\n",
      "Coverage File chr6 (15 of 50)\n",
      "\n",
      "Coverage File chr6 (16 of 50)\n",
      "\n",
      "Coverage File chr6 (17 of 50)\n",
      "\n",
      "Coverage File chr6 (18 of 50)\n",
      "\n",
      "Coverage File chr6 (19 of 50)\n",
      "\n",
      "Coverage File chr6 (20 of 50)\n",
      "\n",
      "Coverage File chr6 (21 of 50)\n",
      "\n",
      "Coverage File chr6 (22 of 50)\n",
      "\n",
      "Coverage File chr6 (23 of 50)\n",
      "\n",
      "Coverage File chr6 (24 of 50)\n",
      "\n",
      "Coverage File chr6 (25 of 50)\n",
      "\n",
      "Coverage File chr6 (26 of 50)\n",
      "\n",
      "Coverage File chr6 (27 of 50)\n",
      "\n",
      "Coverage File chr6 (28 of 50)\n",
      "\n",
      "Coverage File chr6 (29 of 50)\n",
      "\n",
      "Coverage File chr6 (30 of 50)\n",
      "\n",
      "Coverage File chr6 (31 of 50)\n",
      "\n",
      "Coverage File chr6 (32 of 50)\n",
      "\n",
      "Coverage File chr6 (33 of 50)\n",
      "\n",
      "Coverage File chr6 (34 of 50)\n",
      "\n",
      "Coverage File chr6 (35 of 50)\n",
      "\n",
      "Coverage File chr6 (36 of 50)\n",
      "\n",
      "Coverage File chr6 (37 of 50)\n",
      "\n",
      "Coverage File chr6 (38 of 50)\n",
      "\n",
      "Coverage File chr6 (39 of 50)\n",
      "\n",
      "Coverage File chr6 (40 of 50)\n",
      "\n",
      "Coverage File chr6 (41 of 50)\n",
      "\n",
      "Coverage File chr6 (42 of 50)\n",
      "\n",
      "Coverage File chr6 (43 of 50)\n",
      "\n",
      "Coverage File chr6 (44 of 50)\n",
      "\n",
      "Coverage File chr6 (45 of 50)\n",
      "\n",
      "Coverage File chr6 (46 of 50)\n",
      "\n",
      "Coverage File chr6 (47 of 50)\n",
      "\n",
      "Coverage File chr6 (48 of 50)\n",
      "\n",
      "Coverage File chr6 (49 of 50)\n",
      "\n",
      "Coverage File chr6 (50 of 50)\n",
      "\n",
      "Kmer Bias chr7 (20 of 24)\n",
      "\n",
      "chr7 \n",
      "Coverage File chr7 (1 of 50)\n",
      "\n",
      "Coverage File chr7 (2 of 50)\n",
      "\n",
      "Coverage File chr7 (3 of 50)\n",
      "\n",
      "Coverage File chr7 (4 of 50)\n",
      "\n",
      "Coverage File chr7 (5 of 50)\n",
      "\n",
      "Coverage File chr7 (6 of 50)\n",
      "\n",
      "Coverage File chr7 (7 of 50)\n",
      "\n",
      "Coverage File chr7 (8 of 50)\n",
      "\n",
      "Coverage File chr7 (9 of 50)\n",
      "\n",
      "Coverage File chr7 (10 of 50)\n",
      "\n",
      "Coverage File chr7 (11 of 50)\n",
      "\n",
      "Coverage File chr7 (12 of 50)\n",
      "\n",
      "Coverage File chr7 (13 of 50)\n",
      "\n",
      "Coverage File chr7 (14 of 50)\n",
      "\n",
      "Coverage File chr7 (15 of 50)\n",
      "\n",
      "Coverage File chr7 (16 of 50)\n",
      "\n",
      "Coverage File chr7 (17 of 50)\n",
      "\n",
      "Coverage File chr7 (18 of 50)\n",
      "\n",
      "Coverage File chr7 (19 of 50)\n",
      "\n",
      "Coverage File chr7 (20 of 50)\n",
      "\n",
      "Coverage File chr7 (21 of 50)\n",
      "\n",
      "Coverage File chr7 (22 of 50)\n",
      "\n",
      "Coverage File chr7 (23 of 50)\n",
      "\n",
      "Coverage File chr7 (24 of 50)\n",
      "\n",
      "Coverage File chr7 (25 of 50)\n",
      "\n",
      "Coverage File chr7 (26 of 50)\n",
      "\n",
      "Coverage File chr7 (27 of 50)\n",
      "\n",
      "Coverage File chr7 (28 of 50)\n",
      "\n",
      "Coverage File chr7 (29 of 50)\n",
      "\n",
      "Coverage File chr7 (30 of 50)\n",
      "\n",
      "Coverage File chr7 (31 of 50)\n",
      "\n",
      "Coverage File chr7 (32 of 50)\n",
      "\n",
      "Coverage File chr7 (33 of 50)\n",
      "\n",
      "Coverage File chr7 (34 of 50)\n",
      "\n",
      "Coverage File chr7 (35 of 50)\n",
      "\n",
      "Coverage File chr7 (36 of 50)\n",
      "\n",
      "Coverage File chr7 (37 of 50)\n",
      "\n",
      "Coverage File chr7 (38 of 50)\n",
      "\n",
      "Coverage File chr7 (39 of 50)\n",
      "\n",
      "Coverage File chr7 (40 of 50)\n",
      "\n",
      "Coverage File chr7 (41 of 50)\n",
      "\n",
      "Coverage File chr7 (42 of 50)\n",
      "\n",
      "Coverage File chr7 (43 of 50)\n",
      "\n",
      "Coverage File chr7 (44 of 50)\n",
      "\n",
      "Coverage File chr7 (45 of 50)\n",
      "\n",
      "Coverage File chr7 (46 of 50)\n",
      "\n",
      "Coverage File chr7 (47 of 50)\n",
      "\n",
      "Coverage File chr7 (48 of 50)\n",
      "\n",
      "Coverage File chr7 (49 of 50)\n",
      "\n",
      "Coverage File chr7 (50 of 50)\n",
      "\n",
      "Kmer Bias chr8 (21 of 24)\n",
      "\n",
      "chr8 \n",
      "Coverage File chr8 (1 of 50)\n",
      "\n",
      "Coverage File chr8 (2 of 50)\n",
      "\n",
      "Coverage File chr8 (3 of 50)\n",
      "\n",
      "Coverage File chr8 (4 of 50)\n",
      "\n",
      "Coverage File chr8 (5 of 50)\n",
      "\n",
      "Coverage File chr8 (6 of 50)\n",
      "\n",
      "Coverage File chr8 (7 of 50)\n",
      "\n",
      "Coverage File chr8 (8 of 50)\n",
      "\n",
      "Coverage File chr8 (9 of 50)\n",
      "\n",
      "Coverage File chr8 (10 of 50)\n",
      "\n",
      "Coverage File chr8 (11 of 50)\n",
      "\n",
      "Coverage File chr8 (12 of 50)\n",
      "\n",
      "Coverage File chr8 (13 of 50)\n",
      "\n",
      "Coverage File chr8 (14 of 50)\n",
      "\n",
      "Coverage File chr8 (15 of 50)\n",
      "\n",
      "Coverage File chr8 (16 of 50)\n",
      "\n",
      "Coverage File chr8 (17 of 50)\n",
      "\n",
      "Coverage File chr8 (18 of 50)\n",
      "\n",
      "Coverage File chr8 (19 of 50)\n",
      "\n",
      "Coverage File chr8 (20 of 50)\n",
      "\n",
      "Coverage File chr8 (21 of 50)\n",
      "\n",
      "Coverage File chr8 (22 of 50)\n",
      "\n",
      "Coverage File chr8 (23 of 50)\n",
      "\n",
      "Coverage File chr8 (24 of 50)\n",
      "\n",
      "Coverage File chr8 (25 of 50)\n",
      "\n",
      "Coverage File chr8 (26 of 50)\n",
      "\n",
      "Coverage File chr8 (27 of 50)\n",
      "\n",
      "Coverage File chr8 (28 of 50)\n",
      "\n",
      "Coverage File chr8 (29 of 50)\n",
      "\n",
      "Coverage File chr8 (30 of 50)\n",
      "\n",
      "Coverage File chr8 (31 of 50)\n",
      "\n",
      "Coverage File chr8 (32 of 50)\n",
      "\n",
      "Coverage File chr8 (33 of 50)\n",
      "\n",
      "Coverage File chr8 (34 of 50)\n",
      "\n",
      "Coverage File chr8 (35 of 50)\n",
      "\n",
      "Coverage File chr8 (36 of 50)\n",
      "\n",
      "Coverage File chr8 (37 of 50)\n",
      "\n",
      "Coverage File chr8 (38 of 50)\n",
      "\n",
      "Coverage File chr8 (39 of 50)\n",
      "\n",
      "Coverage File chr8 (40 of 50)\n",
      "\n",
      "Coverage File chr8 (41 of 50)\n",
      "\n",
      "Coverage File chr8 (42 of 50)\n",
      "\n",
      "Coverage File chr8 (43 of 50)\n",
      "\n",
      "Coverage File chr8 (44 of 50)\n",
      "\n",
      "Coverage File chr8 (45 of 50)\n",
      "\n",
      "Coverage File chr8 (46 of 50)\n",
      "\n",
      "Coverage File chr8 (47 of 50)\n",
      "\n",
      "Coverage File chr8 (48 of 50)\n",
      "\n",
      "Coverage File chr8 (49 of 50)\n",
      "\n",
      "Coverage File chr8 (50 of 50)\n",
      "\n",
      "Kmer Bias chr9 (22 of 24)\n",
      "\n",
      "chr9 \n",
      "Coverage File chr9 (1 of 50)\n",
      "\n",
      "Coverage File chr9 (2 of 50)\n",
      "\n",
      "Coverage File chr9 (3 of 50)\n",
      "\n",
      "Coverage File chr9 (4 of 50)\n",
      "\n",
      "Coverage File chr9 (5 of 50)\n",
      "\n",
      "Coverage File chr9 (6 of 50)\n",
      "\n",
      "Coverage File chr9 (7 of 50)\n",
      "\n",
      "Coverage File chr9 (8 of 50)\n",
      "\n",
      "Coverage File chr9 (9 of 50)\n",
      "\n",
      "Coverage File chr9 (10 of 50)\n",
      "\n",
      "Coverage File chr9 (11 of 50)\n",
      "\n",
      "Coverage File chr9 (12 of 50)\n",
      "\n",
      "Coverage File chr9 (13 of 50)\n",
      "\n",
      "Coverage File chr9 (14 of 50)\n",
      "\n",
      "Coverage File chr9 (15 of 50)\n",
      "\n",
      "Coverage File chr9 (16 of 50)\n",
      "\n",
      "Coverage File chr9 (17 of 50)\n",
      "\n",
      "Coverage File chr9 (18 of 50)\n",
      "\n",
      "Coverage File chr9 (19 of 50)\n",
      "\n",
      "Coverage File chr9 (20 of 50)\n",
      "\n",
      "Coverage File chr9 (21 of 50)\n",
      "\n",
      "Coverage File chr9 (22 of 50)\n",
      "\n",
      "Coverage File chr9 (23 of 50)\n",
      "\n",
      "Coverage File chr9 (24 of 50)\n",
      "\n",
      "Coverage File chr9 (25 of 50)\n",
      "\n",
      "Coverage File chr9 (26 of 50)\n",
      "\n",
      "Coverage File chr9 (27 of 50)\n",
      "\n",
      "Coverage File chr9 (28 of 50)\n",
      "\n",
      "Coverage File chr9 (29 of 50)\n",
      "\n",
      "Coverage File chr9 (30 of 50)\n",
      "\n",
      "Coverage File chr9 (31 of 50)\n",
      "\n",
      "Coverage File chr9 (32 of 50)\n",
      "\n",
      "Coverage File chr9 (33 of 50)\n",
      "\n",
      "Coverage File chr9 (34 of 50)\n",
      "\n",
      "Coverage File chr9 (35 of 50)\n",
      "\n",
      "Coverage File chr9 (36 of 50)\n",
      "\n",
      "Coverage File chr9 (37 of 50)\n",
      "\n",
      "Coverage File chr9 (38 of 50)\n",
      "\n",
      "Coverage File chr9 (39 of 50)\n",
      "\n",
      "Coverage File chr9 (40 of 50)\n",
      "\n",
      "Coverage File chr9 (41 of 50)\n",
      "\n",
      "Coverage File chr9 (42 of 50)\n",
      "\n",
      "Coverage File chr9 (43 of 50)\n",
      "\n",
      "Coverage File chr9 (44 of 50)\n",
      "\n",
      "Coverage File chr9 (45 of 50)\n",
      "\n",
      "Coverage File chr9 (46 of 50)\n",
      "\n",
      "Coverage File chr9 (47 of 50)\n",
      "\n",
      "Coverage File chr9 (48 of 50)\n",
      "\n",
      "Coverage File chr9 (49 of 50)\n",
      "\n",
      "Coverage File chr9 (50 of 50)\n",
      "\n",
      "Kmer Bias chrX (23 of 24)\n",
      "\n",
      "chrX \n",
      "Coverage File chrX (1 of 50)\n",
      "\n",
      "Coverage File chrX (2 of 50)\n",
      "\n",
      "Coverage File chrX (3 of 50)\n",
      "\n",
      "Coverage File chrX (4 of 50)\n",
      "\n",
      "Coverage File chrX (5 of 50)\n",
      "\n",
      "Coverage File chrX (6 of 50)\n",
      "\n",
      "Coverage File chrX (7 of 50)\n",
      "\n",
      "Coverage File chrX (8 of 50)\n",
      "\n",
      "Coverage File chrX (9 of 50)\n",
      "\n",
      "Coverage File chrX (10 of 50)\n",
      "\n",
      "Coverage File chrX (11 of 50)\n",
      "\n",
      "Coverage File chrX (12 of 50)\n",
      "\n",
      "Coverage File chrX (13 of 50)\n",
      "\n",
      "Coverage File chrX (14 of 50)\n",
      "\n",
      "Coverage File chrX (15 of 50)\n",
      "\n",
      "Coverage File chrX (16 of 50)\n",
      "\n",
      "Coverage File chrX (17 of 50)\n",
      "\n",
      "Coverage File chrX (18 of 50)\n",
      "\n",
      "Coverage File chrX (19 of 50)\n",
      "\n",
      "Coverage File chrX (20 of 50)\n",
      "\n",
      "Coverage File chrX (21 of 50)\n",
      "\n",
      "Coverage File chrX (22 of 50)\n",
      "\n",
      "Coverage File chrX (23 of 50)\n",
      "\n",
      "Coverage File chrX (24 of 50)\n",
      "\n",
      "Coverage File chrX (25 of 50)\n",
      "\n",
      "Coverage File chrX (26 of 50)\n",
      "\n",
      "Coverage File chrX (27 of 50)\n",
      "\n",
      "Coverage File chrX (28 of 50)\n",
      "\n",
      "Coverage File chrX (29 of 50)\n",
      "\n",
      "Coverage File chrX (30 of 50)\n",
      "\n",
      "Coverage File chrX (31 of 50)\n",
      "\n",
      "Coverage File chrX (32 of 50)\n",
      "\n",
      "Coverage File chrX (33 of 50)\n",
      "\n",
      "Coverage File chrX (34 of 50)\n",
      "\n",
      "Coverage File chrX (35 of 50)\n",
      "\n",
      "Coverage File chrX (36 of 50)\n",
      "\n",
      "Coverage File chrX (37 of 50)\n",
      "\n",
      "Coverage File chrX (38 of 50)\n",
      "\n",
      "Coverage File chrX (39 of 50)\n",
      "\n",
      "Coverage File chrX (40 of 50)\n",
      "\n",
      "Coverage File chrX (41 of 50)\n",
      "\n",
      "Coverage File chrX (42 of 50)\n",
      "\n",
      "Coverage File chrX (43 of 50)\n",
      "\n",
      "Coverage File chrX (44 of 50)\n",
      "\n",
      "Coverage File chrX (45 of 50)\n",
      "\n",
      "Coverage File chrX (46 of 50)\n",
      "\n",
      "Coverage File chrX (47 of 50)\n",
      "\n",
      "Coverage File chrX (48 of 50)\n",
      "\n",
      "Coverage File chrX (49 of 50)\n",
      "\n",
      "Coverage File chrX (50 of 50)\n",
      "\n",
      "Kmer Bias chrY (24 of 24)\n",
      "\n",
      "chrY \n",
      "Coverage File chrY (1 of 50)\n",
      "\n",
      "Coverage File chrY (2 of 50)\n",
      "\n",
      "Coverage File chrY (3 of 50)\n",
      "\n",
      "Coverage File chrY (4 of 50)\n",
      "\n",
      "Coverage File chrY (5 of 50)\n",
      "\n",
      "Coverage File chrY (6 of 50)\n",
      "\n",
      "Coverage File chrY (7 of 50)\n",
      "\n",
      "Coverage File chrY (8 of 50)\n",
      "\n",
      "Coverage File chrY (9 of 50)\n",
      "\n",
      "Coverage File chrY (10 of 50)\n",
      "\n",
      "Coverage File chrY (11 of 50)\n",
      "\n",
      "Coverage File chrY (12 of 50)\n",
      "\n",
      "Coverage File chrY (13 of 50)\n",
      "\n",
      "Coverage File chrY (14 of 50)\n",
      "\n",
      "Coverage File chrY (15 of 50)\n",
      "\n",
      "Coverage File chrY (16 of 50)\n",
      "\n",
      "Coverage File chrY (17 of 50)\n",
      "\n",
      "Coverage File chrY (18 of 50)\n",
      "\n",
      "Coverage File chrY (19 of 50)\n",
      "\n",
      "Coverage File chrY (20 of 50)\n",
      "\n",
      "Coverage File chrY (21 of 50)\n",
      "\n",
      "Coverage File chrY (22 of 50)\n",
      "\n",
      "Coverage File chrY (23 of 50)\n",
      "\n",
      "Coverage File chrY (24 of 50)\n",
      "\n",
      "Coverage File chrY (25 of 50)\n",
      "\n",
      "Coverage File chrY (26 of 50)\n",
      "\n",
      "Coverage File chrY (27 of 50)\n",
      "\n",
      "Coverage File chrY (28 of 50)\n",
      "\n",
      "Coverage File chrY (29 of 50)\n",
      "\n",
      "Coverage File chrY (30 of 50)\n",
      "\n",
      "Coverage File chrY (31 of 50)\n",
      "\n",
      "Coverage File chrY (32 of 50)\n",
      "\n",
      "Coverage File chrY (33 of 50)\n",
      "\n",
      "Coverage File chrY (34 of 50)\n",
      "\n",
      "Coverage File chrY (35 of 50)\n",
      "\n",
      "Coverage File chrY (36 of 50)\n",
      "\n",
      "Coverage File chrY (37 of 50)\n",
      "\n",
      "Coverage File chrY (38 of 50)\n",
      "\n",
      "Coverage File chrY (39 of 50)\n",
      "\n",
      "Coverage File chrY (40 of 50)\n",
      "\n",
      "Coverage File chrY (41 of 50)\n",
      "\n",
      "Coverage File chrY (42 of 50)\n",
      "\n",
      "Coverage File chrY (43 of 50)\n",
      "\n",
      "Coverage File chrY (44 of 50)\n",
      "\n",
      "Coverage File chrY (45 of 50)\n",
      "\n",
      "Coverage File chrY (46 of 50)\n",
      "\n",
      "Coverage File chrY (47 of 50)\n",
      "\n",
      "Coverage File chrY (48 of 50)\n",
      "\n",
      "Coverage File chrY (49 of 50)\n",
      "\n",
      "Coverage File chrY (50 of 50)\n",
      "\n",
      "Completed Kmer Bias Calculation\n",
      "\n",
      "Adding Kmer Bias (1 of 50)\n",
      "\n",
      "Adding Kmer Bias (2 of 50)\n",
      "\n",
      "Adding Kmer Bias (3 of 50)\n",
      "\n",
      "Adding Kmer Bias (4 of 50)\n",
      "\n",
      "Adding Kmer Bias (5 of 50)\n",
      "\n",
      "Adding Kmer Bias (6 of 50)\n",
      "\n",
      "Adding Kmer Bias (7 of 50)\n",
      "\n",
      "Adding Kmer Bias (8 of 50)\n",
      "\n",
      "Adding Kmer Bias (9 of 50)\n",
      "\n",
      "Adding Kmer Bias (10 of 50)\n",
      "\n",
      "Adding Kmer Bias (11 of 50)\n",
      "\n",
      "Adding Kmer Bias (12 of 50)\n",
      "\n",
      "Adding Kmer Bias (13 of 50)\n",
      "\n",
      "Adding Kmer Bias (14 of 50)\n",
      "\n",
      "Adding Kmer Bias (15 of 50)\n",
      "\n",
      "Adding Kmer Bias (16 of 50)\n",
      "\n",
      "Adding Kmer Bias (17 of 50)\n",
      "\n",
      "Adding Kmer Bias (18 of 50)\n",
      "\n",
      "Adding Kmer Bias (19 of 50)\n",
      "\n",
      "Adding Kmer Bias (20 of 50)\n",
      "\n",
      "Adding Kmer Bias (21 of 50)\n",
      "\n",
      "Adding Kmer Bias (22 of 50)\n",
      "\n",
      "Adding Kmer Bias (23 of 50)\n",
      "\n",
      "Adding Kmer Bias (24 of 50)\n",
      "\n",
      "Adding Kmer Bias (25 of 50)\n",
      "\n",
      "Adding Kmer Bias (26 of 50)\n",
      "\n",
      "Adding Kmer Bias (27 of 50)\n",
      "\n",
      "Adding Kmer Bias (28 of 50)\n",
      "\n",
      "Adding Kmer Bias (29 of 50)\n",
      "\n",
      "Adding Kmer Bias (30 of 50)\n",
      "\n",
      "Adding Kmer Bias (31 of 50)\n",
      "\n",
      "Adding Kmer Bias (32 of 50)\n",
      "\n",
      "Adding Kmer Bias (33 of 50)\n",
      "\n",
      "Adding Kmer Bias (34 of 50)\n",
      "\n",
      "Adding Kmer Bias (35 of 50)\n",
      "\n",
      "Adding Kmer Bias (36 of 50)\n",
      "\n",
      "Adding Kmer Bias (37 of 50)\n",
      "\n",
      "Adding Kmer Bias (38 of 50)\n",
      "\n",
      "Adding Kmer Bias (39 of 50)\n",
      "\n",
      "Adding Kmer Bias (40 of 50)\n",
      "\n",
      "Adding Kmer Bias (41 of 50)\n",
      "\n",
      "Adding Kmer Bias (42 of 50)\n",
      "\n",
      "Adding Kmer Bias (43 of 50)\n",
      "\n",
      "Adding Kmer Bias (44 of 50)\n",
      "\n",
      "Adding Kmer Bias (45 of 50)\n",
      "\n",
      "Adding Kmer Bias (46 of 50)\n",
      "\n",
      "Adding Kmer Bias (47 of 50)\n",
      "\n",
      "Adding Kmer Bias (48 of 50)\n",
      "\n",
      "Adding Kmer Bias (49 of 50)\n",
      "\n",
      "Adding Kmer Bias (50 of 50)\n",
      "\n",
      "2024-05-13 00:31:17 : Finished Creation of Coverage Files!, 72.095 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addGroupCoverages-b94d787433ed-Date-2024-05-12_Time-23-19-11.log\n",
      "\n",
      "Searching For MACS2..\n",
      "\n",
      "Found with $path!\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-b94d514e7a1b-Date-2024-05-13_Time-00-31-19.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "Calling Peaks with Macs2\n",
      "\n",
      "2024-05-13 00:31:21 : Peak Calling Parameters!, 0.046 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Group nCells nCellsUsed nReplicates nMin nMax maxPeaks\n",
      "C1     C1     92         92           2   40   52    46000\n",
      "C2     C2     99         99           2   40   59    49500\n",
      "C3     C3    997        540           2   40  500   150000\n",
      "C4     C4   2911        540           2   40  500   150000\n",
      "C5     C5    137        137           2   40   97    68500\n",
      "C6     C6    307        307           2   40  267   150000\n",
      "C7     C7    445        445           2   40  405   150000\n",
      "C8     C8     67         61           2   40   40    30500\n",
      "C9     C9    383        383           2   40  343   150000\n",
      "C10   C10    535        535           2   40  495   150000\n",
      "C11   C11     53         52           2   40   40    26000\n",
      "C12   C12   3461        540           2   40  500   150000\n",
      "C13   C13    164        164           2   40  124    82000\n",
      "C14   C14    397        397           2   40  357   150000\n",
      "C15   C15    230        230           2   40  190   115000\n",
      "C16   C16    486        486           2   40  446   150000\n",
      "C17   C17    460        460           2   40  420   150000\n",
      "C18   C18    598        540           2   40  500   150000\n",
      "C19   C19     29         28           2   17   22    14000\n",
      "C20   C20    264        264           2   40  224   132000\n",
      "C21   C21     89         89           2   40   49    44500\n",
      "C22   C22    352        352           2   40  312   150000\n",
      "C23   C23    287        287           2   40  247   143500\n",
      "C24   C24    435        435           2   40  395   150000\n",
      "C25   C25    467        467           2   40  427   150000\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-13 00:31:22 : Batching Peak Calls!, 0.052 mins elapsed.\n",
      "\n",
      "2024-05-13 00:31:22 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2024-05-13 00:31:22 : Group 1 of 48, Calling Peaks with MACS2!, 0 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C1._.Rep1-1 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C1._.Rep1-1.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:31:33 : Group 2 of 48, Calling Peaks with MACS2!, 0.173 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C1._.Rep2-2 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C1._.Rep2-2.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:31:42 : Group 3 of 48, Calling Peaks with MACS2!, 0.325 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C2._.Rep1-3 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C2._.Rep1-3.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:32:16 : Group 4 of 48, Calling Peaks with MACS2!, 0.89 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C2._.Rep2-4 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C2._.Rep2-4.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:32:38 : Group 5 of 48, Calling Peaks with MACS2!, 1.264 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C3._.Rep1-5 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C3._.Rep1-5.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:35:44 : Group 6 of 48, Calling Peaks with MACS2!, 4.354 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C3._.Rep2-6 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C3._.Rep2-6.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:36:38 : Group 7 of 48, Calling Peaks with MACS2!, 5.262 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C4._.Rep1-7 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C4._.Rep1-7.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:38:42 : Group 8 of 48, Calling Peaks with MACS2!, 7.331 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C4._.Rep2-8 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C4._.Rep2-8.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:39:11 : Group 9 of 48, Calling Peaks with MACS2!, 7.816 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C5._.Rep1-9 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C5._.Rep1-9.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:40:37 : Group 10 of 48, Calling Peaks with MACS2!, 9.251 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C5._.Rep2-10 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C5._.Rep2-10.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:41:23 : Group 11 of 48, Calling Peaks with MACS2!, 10.01 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C6._.Rep1-11 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C6._.Rep1-11.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:43:30 : Group 12 of 48, Calling Peaks with MACS2!, 12.129 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C6._.Rep2-12 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C6._.Rep2-12.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:44:10 : Group 13 of 48, Calling Peaks with MACS2!, 12.795 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C7._.Rep1-13 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C7._.Rep1-13.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:46:36 : Group 14 of 48, Calling Peaks with MACS2!, 15.223 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C7._.Rep2-14 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C7._.Rep2-14.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:47:21 : Group 15 of 48, Calling Peaks with MACS2!, 15.968 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C8._.Rep1-15 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C8._.Rep1-15.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:47:59 : Group 16 of 48, Calling Peaks with MACS2!, 16.608 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C8._.Rep2-16 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C8._.Rep2-16.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:48:40 : Group 17 of 48, Calling Peaks with MACS2!, 17.297 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C9._.Rep1-17 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C9._.Rep1-17.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:50:07 : Group 18 of 48, Calling Peaks with MACS2!, 18.744 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C9._.Rep2-18 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C9._.Rep2-18.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:50:34 : Group 19 of 48, Calling Peaks with MACS2!, 19.194 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C10._.Rep1-19 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C10._.Rep1-19.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:52:06 : Group 20 of 48, Calling Peaks with MACS2!, 20.719 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C10._.Rep2-20 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C10._.Rep2-20.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:52:30 : Group 21 of 48, Calling Peaks with MACS2!, 21.134 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C11._.Rep1-21 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C11._.Rep1-21.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:53:19 : Group 22 of 48, Calling Peaks with MACS2!, 21.94 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C11._.Rep2-22 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C11._.Rep2-22.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:54:08 : Group 23 of 48, Calling Peaks with MACS2!, 22.764 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C12._.Rep1-23 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C12._.Rep1-23.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:55:34 : Group 24 of 48, Calling Peaks with MACS2!, 24.187 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C12._.Rep2-24 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C12._.Rep2-24.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:56:48 : Group 25 of 48, Calling Peaks with MACS2!, 25.423 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C13._.Rep1-25 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C13._.Rep1-25.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:57:29 : Group 26 of 48, Calling Peaks with MACS2!, 26.114 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C13._.Rep2-26 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C13._.Rep2-26.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 00:57:45 : Group 27 of 48, Calling Peaks with MACS2!, 26.38 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C14._.Rep1-27 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C14._.Rep1-27.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:00:35 : Group 28 of 48, Calling Peaks with MACS2!, 29.216 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C14._.Rep2-28 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C14._.Rep2-28.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:01:26 : Group 29 of 48, Calling Peaks with MACS2!, 30.067 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C15._.Rep1-29 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C15._.Rep1-29.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:02:41 : Group 30 of 48, Calling Peaks with MACS2!, 31.302 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C15._.Rep2-30 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C15._.Rep2-30.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:03:10 : Group 31 of 48, Calling Peaks with MACS2!, 31.794 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C16._.Rep1-31 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C16._.Rep1-31.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:05:12 : Group 32 of 48, Calling Peaks with MACS2!, 33.826 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C16._.Rep2-32 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C16._.Rep2-32.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:05:36 : Group 33 of 48, Calling Peaks with MACS2!, 34.227 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C17._.Rep1-33 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C17._.Rep1-33.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:07:19 : Group 34 of 48, Calling Peaks with MACS2!, 35.945 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C17._.Rep2-34 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C17._.Rep2-34.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:07:43 : Group 35 of 48, Calling Peaks with MACS2!, 36.339 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C18._.Rep1-35 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C18._.Rep1-35.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:09:14 : Group 36 of 48, Calling Peaks with MACS2!, 37.855 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C18._.Rep2-36 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C18._.Rep2-36.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:09:34 : Group 37 of 48, Calling Peaks with MACS2!, 38.194 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C20._.Rep1-37 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C20._.Rep1-37.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:10:58 : Group 38 of 48, Calling Peaks with MACS2!, 39.597 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C20._.Rep2-38 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C20._.Rep2-38.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:11:37 : Group 39 of 48, Calling Peaks with MACS2!, 40.241 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C21._.Rep1-39 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C21._.Rep1-39.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:12:19 : Group 40 of 48, Calling Peaks with MACS2!, 40.946 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C21._.Rep2-40 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C21._.Rep2-40.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:12:54 : Group 41 of 48, Calling Peaks with MACS2!, 41.533 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C22._.Rep1-41 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C22._.Rep1-41.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:14:49 : Group 42 of 48, Calling Peaks with MACS2!, 43.447 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C22._.Rep2-42 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C22._.Rep2-42.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:15:24 : Group 43 of 48, Calling Peaks with MACS2!, 44.03 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C23._.Rep1-43 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C23._.Rep1-43.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:16:54 : Group 44 of 48, Calling Peaks with MACS2!, 45.518 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C23._.Rep2-44 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C23._.Rep2-44.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:18:34 : Group 45 of 48, Calling Peaks with MACS2!, 47.199 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C24._.Rep1-45 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C24._.Rep1-45.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:20:11 : Group 46 of 48, Calling Peaks with MACS2!, 48.816 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C24._.Rep2-46 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C24._.Rep2-46.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:20:38 : Group 47 of 48, Calling Peaks with MACS2!, 49.255 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C25._.Rep1-47 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C25._.Rep1-47.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:22:12 : Group 48 of 48, Calling Peaks with MACS2!, 50.832 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C25._.Rep2-48 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds/C25._.Rep2-48.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-13 01:22:40 : Identifying Reproducible Peaks!, 51.355 mins elapsed.\n",
      "\n",
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C1-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C10-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C11-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C12-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C13-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C14-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C15-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C16-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C17-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C18-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C2-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C20-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C21-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C22-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C23-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C24-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C25-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C3-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C4-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C5-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C6-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C7-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C8-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1/PeakCalls/C9-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-13 01:24:57 : Creating Union Peak Set!, 53.638 mins elapsed.\n",
      "\n",
      "Converged after 8 iterations!\n",
      "\n",
      "Plotting Ggplot!\n",
      "\n",
      "2024-05-13 01:25:06 : Finished Creating Union Peak Set (198362)!, 53.794 mins elapsed.\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-b94d53288b59-Date-2024-05-13_Time-01-25-06.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-13 01:25:07 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2024-05-13 01:25:09 : Adding BMMC_D6T1 to PeakMatrix for Chr (1 of 23)!, 0.024 mins elapsed.\n",
      "\n",
      "2024-05-13 01:25:26 : Adding BMMC_D6T1 to PeakMatrix for Chr (2 of 23)!, 0.314 mins elapsed.\n",
      "\n",
      "2024-05-13 01:25:39 : Adding BMMC_D6T1 to PeakMatrix for Chr (3 of 23)!, 0.531 mins elapsed.\n",
      "\n",
      "2024-05-13 01:25:49 : Adding BMMC_D6T1 to PeakMatrix for Chr (4 of 23)!, 0.699 mins elapsed.\n",
      "\n",
      "2024-05-13 01:25:57 : Adding BMMC_D6T1 to PeakMatrix for Chr (5 of 23)!, 0.83 mins elapsed.\n",
      "\n",
      "2024-05-13 01:26:06 : Adding BMMC_D6T1 to PeakMatrix for Chr (6 of 23)!, 0.979 mins elapsed.\n",
      "\n",
      "2024-05-13 01:26:16 : Adding BMMC_D6T1 to PeakMatrix for Chr (7 of 23)!, 1.152 mins elapsed.\n",
      "\n",
      "2024-05-13 01:26:25 : Adding BMMC_D6T1 to PeakMatrix for Chr (8 of 23)!, 1.303 mins elapsed.\n",
      "\n",
      "2024-05-13 01:26:33 : Adding BMMC_D6T1 to PeakMatrix for Chr (9 of 23)!, 1.435 mins elapsed.\n",
      "\n",
      "2024-05-13 01:26:41 : Adding BMMC_D6T1 to PeakMatrix for Chr (10 of 23)!, 1.569 mins elapsed.\n",
      "\n",
      "2024-05-13 01:26:51 : Adding BMMC_D6T1 to PeakMatrix for Chr (11 of 23)!, 1.726 mins elapsed.\n",
      "\n",
      "2024-05-13 01:27:00 : Adding BMMC_D6T1 to PeakMatrix for Chr (12 of 23)!, 1.885 mins elapsed.\n",
      "\n",
      "2024-05-13 01:27:10 : Adding BMMC_D6T1 to PeakMatrix for Chr (13 of 23)!, 2.045 mins elapsed.\n",
      "\n",
      "2024-05-13 01:27:16 : Adding BMMC_D6T1 to PeakMatrix for Chr (14 of 23)!, 2.143 mins elapsed.\n",
      "\n",
      "2024-05-13 01:27:23 : Adding BMMC_D6T1 to PeakMatrix for Chr (15 of 23)!, 2.262 mins elapsed.\n",
      "\n",
      "2024-05-13 01:27:30 : Adding BMMC_D6T1 to PeakMatrix for Chr (16 of 23)!, 2.382 mins elapsed.\n",
      "\n",
      "2024-05-13 01:27:38 : Adding BMMC_D6T1 to PeakMatrix for Chr (17 of 23)!, 2.514 mins elapsed.\n",
      "\n",
      "2024-05-13 01:27:49 : Adding BMMC_D6T1 to PeakMatrix for Chr (18 of 23)!, 2.695 mins elapsed.\n",
      "\n",
      "2024-05-13 01:27:54 : Adding BMMC_D6T1 to PeakMatrix for Chr (19 of 23)!, 2.786 mins elapsed.\n",
      "\n",
      "2024-05-13 01:28:04 : Adding BMMC_D6T1 to PeakMatrix for Chr (20 of 23)!, 2.95 mins elapsed.\n",
      "\n",
      "2024-05-13 01:28:11 : Adding BMMC_D6T1 to PeakMatrix for Chr (21 of 23)!, 3.055 mins elapsed.\n",
      "\n",
      "2024-05-13 01:28:15 : Adding BMMC_D6T1 to PeakMatrix for Chr (22 of 23)!, 3.135 mins elapsed.\n",
      "\n",
      "2024-05-13 01:28:21 : Adding BMMC_D6T1 to PeakMatrix for Chr (23 of 23)!, 3.235 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-b94d53288b59-Date-2024-05-13_Time-01-25-06.log\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-b94d92c6ad0-Date-2024-05-13_Time-01-28-28.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-13 01:29:16 : Organizing colData, 0.81 mins elapsed.\n",
      "\n",
      "2024-05-13 01:29:16 : Organizing rowData, 0.81 mins elapsed.\n",
      "\n",
      "2024-05-13 01:29:17 : Organizing rowRanges, 0.811 mins elapsed.\n",
      "\n",
      "2024-05-13 01:29:17 : Organizing Assays (1 of 1), 0.812 mins elapsed.\n",
      "\n",
      "2024-05-13 01:29:17 : Constructing SummarizedExperiment, 0.813 mins elapsed.\n",
      "\n",
      "2024-05-13 01:29:18 : Finished Matrix Creation, 0.835 mins elapsed.\n",
      "\n",
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 20.3 GiB”\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "out_path = \"/project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/05_peaks_and_genesScore_matrix_ArchR/input_for_EpiAnno/\" # where to save resulting matrices | with / at the end\n",
    "threads_=1 # how many threads to use\n",
    "\n",
    "proj1 <- addIterativeLSI(\n",
    "    ArchRProj = proj1,\n",
    "    useMatrix = \"TileMatrix\", \n",
    "    name = \"IterativeLSI\", \n",
    "    iterations = 2, \n",
    "    clusterParams = list( #See Seurat::FindClusters\n",
    "        resolution = c(0.2), \n",
    "        sampleCells = 10000, \n",
    "        n.start = 10\n",
    "    ), \n",
    "    varFeatures = 25000, \n",
    "    dimsToUse = 1:30\n",
    ")\n",
    "\n",
    "proj1 <- addClusters(\n",
    "    input = proj1,\n",
    "    reducedDims = \"IterativeLSI\",\n",
    "    method = \"Seurat\",\n",
    "    name = \"Clusters\",\n",
    "    resolution = 0.8\n",
    ")\n",
    "\n",
    "proj1 <- addUMAP(\n",
    "    ArchRProj = proj1, \n",
    "    reducedDims = \"IterativeLSI\", \n",
    "    name = \"UMAP\", \n",
    "    nNeighbors = 30, \n",
    "    minDist = 0.5, \n",
    "    metric = \"cosine\"\n",
    ")\n",
    "\n",
    "HDF5_USE_FILE_LOCKING=FALSE\n",
    "RHDF5_USE_FILE_LOCKING=FALSE\n",
    "\n",
    "proj2 <- addGroupCoverages(ArchRProj = proj1, groupBy = \"Clusters\")\n",
    "\n",
    "pathToMacs2 <- findMacs2()\n",
    "\n",
    "proj2 <- addReproduciblePeakSet(\n",
    "    ArchRProj = proj2, \n",
    "    groupBy = \"Clusters\", \n",
    "    pathToMacs2 = pathToMacs2\n",
    ")\n",
    "\n",
    "\n",
    "#saveArchRProject(ArchRProj = proj2, outputDirectory = \"Save-Proj2_afterPeakCalling\", load = FALSE)\n",
    "\n",
    "proj3 <- addPeakMatrix(proj2)\n",
    "\n",
    "peakMatrix <- getMatrixFromProject(proj3, useMatrix='PeakMatrix')\n",
    "\n",
    "PeakMatrix_proj3_dense =  as.matrix(peakMatrix@assays@data@listData$PeakMatrix)\n",
    "\n",
    "write.csv(PeakMatrix_proj3_dense, file = paste(out_path,\"PeakMatrix_dense.csv\",sep=\"\" )) \n",
    "\n",
    "# get the peak regions\n",
    "proj3_peak_metadata  = (getPeakSet(proj3))\n",
    "proj3_peak_regions = data.frame(seqnames = seqnames(proj3_peak_metadata),\n",
    "           start= start(proj3_peak_metadata),\n",
    "            end = end(proj3_peak_metadata))\n",
    "\n",
    "write.csv(proj3_peak_metadata, file =paste(out_path,\"peak_metadata.csv\",sep=\"\" ))\n",
    "write.csv(proj3_peak_regions, file =paste(out_path,\"peak_regions.csv\",sep=\"\" ))\n",
    "write.table(proj3_peak_regions, file =paste(out_path,\"peak_regions.tsv\",sep=\"\" ),col.names = F,row.names=FALSE,sep=\"\\t\", quote = FALSE)\n",
    "\n",
    "# save the peak matrix as sparse mtx\n",
    "\n",
    "\n",
    "# save sparse matrix\n",
    "sparse_matrix <- Matrix(PeakMatrix_proj3_dense , sparse = T )\n",
    "writeMM(obj = sparse_matrix, file = paste(out_path,\"ArchR_peakMatrix.mtx\",sep=\"\" ))\n",
    "\n",
    "# save genes and cells names\n",
    "combined_peak_features =  paste(proj3_peak_regions$seqnames,\"_\",proj3_peak_regions$start,\"_\",proj3_peak_regions$end, sep=\"\")\n",
    "\n",
    "write.table(as.data.frame(combined_peak_features), file = paste(out_path,\"ArchR_peaks.tsv\",sep=\"\" ) , row.names = FALSE, quote = FALSE, col.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "1c79f87d-0159-4d17-b32a-6c78ad52bcab",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "class: ArchRProject \n",
       "outputDirectory: /project/scATAC_analysis/scATAcat_review/preprocessing/Granja2019/BMMC_D6T1/05_get_peaks_and_genesScore_matrix_ArchR/BMMC_D6T1 \n",
       "samples(1): BMMC_D6T1\n",
       "sampleColData names(1): ArrowFiles\n",
       "cellColData names(14): Sample nMultiFrags ... BlacklistRatio Clusters\n",
       "numberOfCells(1): 13745\n",
       "medianTSS(1): 14.317\n",
       "medianFrags(1): 9853"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "proj1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "ec641abe-7a74-4eff-8955-7dd0fb133c68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneScoreMatrix'</li><li>'PeakMatrix'</li><li>'TileMatrix'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneScoreMatrix'\n",
       "\\item 'PeakMatrix'\n",
       "\\item 'TileMatrix'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneScoreMatrix'\n",
       "2. 'PeakMatrix'\n",
       "3. 'TileMatrix'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GeneScoreMatrix\" \"PeakMatrix\"      \"TileMatrix\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getAvailableMatrices(proj3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "634cea34-1618-4af5-a338-b8d1129febc5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-b94d155fe190-Date-2024-05-13_Time-02-08-45.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-13 02:10:54 : Organizing colData, 2.16 mins elapsed.\n",
      "\n",
      "2024-05-13 02:10:55 : Organizing rowData, 2.178 mins elapsed.\n",
      "\n",
      "2024-05-13 02:10:58 : Organizing rowRanges, 2.218 mins elapsed.\n",
      "\n",
      "2024-05-13 02:10:59 : Organizing Assays (1 of 1), 2.236 mins elapsed.\n",
      "\n",
      "2024-05-13 02:11:00 : Constructing SummarizedExperiment, 2.246 mins elapsed.\n",
      "\n",
      "2024-05-13 02:11:02 : Finished Matrix Creation, 2.295 mins elapsed.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# get the gene activity matrix \n",
    "\n",
    "geneScoreMatrix <- getMatrixFromProject(proj3, useMatrix='GeneScoreMatrix')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "776d96ef-b636-4b0f-9dc3-4dd7afa4cb53",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-b94d256cc192-Date-2024-05-13_Time-02-11-04.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-13 02:13:14 : Organizing colData, 2.168 mins elapsed.\n",
      "\n",
      "2024-05-13 02:13:15 : Organizing rowData, 2.183 mins elapsed.\n",
      "\n",
      "2024-05-13 02:13:17 : Organizing rowRanges, 2.217 mins elapsed.\n",
      "\n",
      "2024-05-13 02:13:18 : Organizing Assays (1 of 1), 2.236 mins elapsed.\n",
      "\n",
      "2024-05-13 02:13:19 : Constructing SummarizedExperiment, 2.252 mins elapsed.\n",
      "\n",
      "2024-05-13 02:13:23 : Finished Matrix Creation, 2.312 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>198362</li><li>13745</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 198362\n",
       "\\item 13745\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 198362\n",
       "2. 13745\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 198362  13745"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(getMatrixFromProject(proj3, useMatrix='PeakMatrix'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "0ff29c89-7d7f-4e73-8bf0-248222debbe6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>24919</li><li>13745</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 24919\n",
       "\\item 13745\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 24919\n",
       "2. 13745\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 24919 13745"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(geneScoreMatrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "67599673-dcde-44df-9050-ec01bd7afe57",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneScore_outDir = \"/project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/05_peaks_and_genesScore_matrix_ArchR/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "496d2733-3552-4ae8-b1db-adf8f6e9cb1c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "write.table(rowData(geneScoreMatrix)$name, paste(geneScore_outDir, \"ArchR_genescore_genes.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "4c9cc218-b028-4c12-a674-81ada3922ff6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "write.table(colnames(geneScoreMatrix), paste(geneScore_outDir, \"ArchR_genescore_barcodes.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "d6731a5a-e4a1-4705-b0bc-d7b4bb555241",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "geneScoreMatrix_sparse = geneScoreMatrix@assays@data@listData$GeneScoreMatrix\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "cbfdd6a6-c095-49d2-8fd7-c99b7db87d96",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "writeMM(obj = geneScoreMatrix_sparse, file=paste(geneScore_outDir, \"ArchR_genescore.mtx\", sep=\"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3bbb363a-a304-46eb-ba54-880d0eae5b93",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Copying ArchRProject to new outputDirectory : /project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/05_peaks_and_genesScore_matrix_ArchR/Save-Proj\n",
      "\n",
      "Copying Arrow Files...\n",
      "\n",
      "Copying Arrow Files (1 of 1)\n",
      "\n",
      "Getting ImputeWeights\n",
      "\n",
      "No imputeWeights found, returning NULL\n",
      "\n",
      "Copying Other Files...\n",
      "\n",
      "Copying Other Files (1 of 5): Embeddings\n",
      "\n",
      "Copying Other Files (2 of 5): GroupCoverages\n",
      "\n",
      "Copying Other Files (3 of 5): IterativeLSI\n",
      "\n",
      "Copying Other Files (4 of 5): PeakCalls\n",
      "\n",
      "Copying Other Files (5 of 5): Plots\n",
      "\n",
      "Saving ArchRProject...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "saveArchRProject(ArchRProj = proj3, outputDirectory = paste(geneScore_outDir,\"Save-Proj\", sep=\"\"), load = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "66a37710-4da9-4dc8-8982-2be6bbf0d1d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(stringr)\n",
    "\n",
    "write.table(as.data.frame(str_split_fixed(colnames(geneScoreMatrix),'#',2))$V2, paste(geneScore_outDir, \"cell_IDs.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "raw",
   "id": "ccf549b7-f3e6-4742-babd-0cc5fed30047",
   "metadata": {},
   "source": [
    "cd /project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/05_peaks_and_genesScore_matrix_ArchR/\n",
    "gzip -k ArchR_genescore.mtx"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.1.0",
   "language": "R",
   "name": "ir4.1.0"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
