{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "099c50f6-93aa-4c54-988d-90cfadde27b7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Loading required package: ggplot2\n",
      "\n",
      "Loading required package: SummarizedExperiment\n",
      "\n",
      "Loading required package: MatrixGenerics\n",
      "\n",
      "Loading required package: matrixStats\n",
      "\n",
      "\n",
      "Attaching package: ‘MatrixGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,\n",
      "    colCounts, colCummaxs, colCummins, colCumprods, colCumsums,\n",
      "    colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,\n",
      "    colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,\n",
      "    colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,\n",
      "    colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,\n",
      "    colWeightedMeans, colWeightedMedians, colWeightedSds,\n",
      "    colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,\n",
      "    rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,\n",
      "    rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,\n",
      "    rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,\n",
      "    rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,\n",
      "    rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,\n",
      "    rowWeightedMads, rowWeightedMeans, rowWeightedMedians,\n",
      "    rowWeightedSds, rowWeightedVars\n",
      "\n",
      "\n",
      "Loading required package: GenomicRanges\n",
      "\n",
      "Loading required package: stats4\n",
      "\n",
      "Loading required package: BiocGenerics\n",
      "\n",
      "Loading required package: parallel\n",
      "\n",
      "\n",
      "Attaching package: ‘BiocGenerics’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:parallel’:\n",
      "\n",
      "    clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,\n",
      "    clusterExport, clusterMap, parApply, parCapply, parLapply,\n",
      "    parLapplyLB, parRapply, parSapply, parSapplyLB\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:stats’:\n",
      "\n",
      "    IQR, mad, sd, var, xtabs\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    Filter, Find, Map, Position, Reduce, anyDuplicated, append,\n",
      "    as.data.frame, basename, cbind, colnames, dirname, do.call,\n",
      "    duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,\n",
      "    lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,\n",
      "    pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,\n",
      "    tapply, union, unique, unsplit, which.max, which.min\n",
      "\n",
      "\n",
      "Loading required package: S4Vectors\n",
      "\n",
      "\n",
      "Attaching package: ‘S4Vectors’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:base’:\n",
      "\n",
      "    I, expand.grid, unname\n",
      "\n",
      "\n",
      "Loading required package: IRanges\n",
      "\n",
      "Loading required package: GenomeInfoDb\n",
      "\n",
      "Loading required package: Biobase\n",
      "\n",
      "Welcome to Bioconductor\n",
      "\n",
      "    Vignettes contain introductory material; view with\n",
      "    'browseVignettes()'. To cite Bioconductor, see\n",
      "    'citation(\"Biobase\")', and for packages 'citation(\"pkgname\")'.\n",
      "\n",
      "\n",
      "\n",
      "Attaching package: ‘Biobase’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:MatrixGenerics’:\n",
      "\n",
      "    rowMedians\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:matrixStats’:\n",
      "\n",
      "    anyMissing, rowMedians\n",
      "\n",
      "\n",
      "Loading required package: data.table\n",
      "\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:SummarizedExperiment’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:GenomicRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:IRanges’:\n",
      "\n",
      "    shift\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:S4Vectors’:\n",
      "\n",
      "    first, second\n",
      "\n",
      "\n",
      "Loading required package: Matrix\n",
      "\n",
      "\n",
      "Attaching package: ‘Matrix’\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:S4Vectors’:\n",
      "\n",
      "    expand\n",
      "\n",
      "\n",
      "Loading required package: rhdf5\n",
      "\n",
      "Loading required package: magrittr\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(ArchR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "caf5245a-27ea-48a9-838c-9b73fbd1c995",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting default number of Parallel threads to 1.\n",
      "\n",
      "Warning message:\n",
      "“no function found corresponding to methods exports from ‘BSgenome’ for: ‘releaseName’”\n",
      "Setting default genome to Hg38.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "set.seed(2022)\n",
    "addArchRThreads(threads = 1) \n",
    "addArchRGenome(\"hg38\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "51633fb9-37fc-4dd7-b1c4-4ec8aae16c44",
   "metadata": {},
   "outputs": [],
   "source": [
    "inputFiles = \"/project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/01_prepare_fragments_file/final_fragment_files/GSM4138889_scATAC_BMMC_D6T1_liftOverhg38_no_chrMt_sorted_fragments.tsv.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "84df57d8-0603-45d5-b47d-8fe24cbc4e13",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-createArrows-11dc34864ebe7-Date-2024-05-11_Time-23-08-28.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "Cleaning Temporary Files\n",
      "\n",
      "2024-05-11 23:08:33 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "(BMMC_D6T1 : 1 of 1) Determining Arrow Method to use!\n",
      "\n",
      "2024-05-11 23:08:33 : (BMMC_D6T1 : 1 of 1) Reading In Fragments from inputFiles (readMethod = tabix), 0.008 mins elapsed.\n",
      "\n",
      "2024-05-11 23:08:33 : (BMMC_D6T1 : 1 of 1) Tabix Bed To Temporary File, 0.009 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:10:14 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 8 Percent, 1.693 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:12:01 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 17 Percent, 3.48 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:13:33 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 25 Percent, 5.011 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:14:55 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 33 Percent, 6.373 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:16:10 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 42 Percent, 7.618 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:17:25 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 50 Percent, 8.871 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:18:21 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 58 Percent, 9.814 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:19:17 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 67 Percent, 10.748 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:20:13 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 75 Percent, 11.668 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:21:03 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 83 Percent, 12.501 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:21:36 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 92 Percent, 13.064 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-11 23:22:13 : (BMMC_D6T1 : 1 of 1) Reading TabixFile 100 Percent, 13.67 mins elapsed.\n",
      "\n",
      "2024-05-11 23:22:15 : (BMMC_D6T1 : 1 of 1) Successful creation of Temporary File, 13.705 mins elapsed.\n",
      "\n",
      "2024-05-11 23:22:15 : (BMMC_D6T1 : 1 of 1) Creating ArrowFile From Temporary File, 13.707 mins elapsed.\n",
      "\n",
      "2024-05-11 23:25:42 : (BMMC_D6T1 : 1 of 1) Successful creation of Arrow File, 17.152 mins elapsed.\n",
      "\n",
      "2024-05-11 23:28:29 : (BMMC_D6T1 : 1 of 1) CellStats : Number of Cells Pass Filter = 3413 , 19.95 mins elapsed.\n",
      "\n",
      "2024-05-11 23:28:30 : (BMMC_D6T1 : 1 of 1) CellStats : Median Frags = 9199 , 19.95 mins elapsed.\n",
      "\n",
      "2024-05-11 23:28:30 : (BMMC_D6T1 : 1 of 1) CellStats : Median TSS Enrichment = 18.622 , 19.951 mins elapsed.\n",
      "\n",
      "2024-05-11 23:28:34 : (BMMC_D6T1 : 1 of 1) Adding Additional Feature Counts!, 20.032 mins elapsed.\n",
      "\n",
      "2024-05-11 23:30:52 : (BMMC_D6T1 : 1 of 1) Removing Fragments from Filtered Cells, 22.331 mins elapsed.\n",
      "\n",
      "2024-05-11 23:30:52 : (BMMC_D6T1 : 1 of 1) Creating Filtered Arrow File, 22.332 mins elapsed.\n",
      "\n",
      "2024-05-11 23:32:01 : (BMMC_D6T1 : 1 of 1) Finished Constructing Filtered Arrow File!, 23.48 mins elapsed.\n",
      "\n",
      "2024-05-11 23:32:02 : (BMMC_D6T1 : 1 of 1) Adding TileMatrix!, 23.497 mins elapsed.\n",
      "\n",
      "2024-05-11 23:35:14 : (BMMC_D6T1 : 1 of 1) Adding GeneScoreMatrix!, 26.686 mins elapsed.\n",
      "\n",
      "2024-05-11 23:39:12 : (BMMC_D6T1 : 1 of 1) Finished Creating Arrow File, 30.659 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-createArrows-11dc34864ebe7-Date-2024-05-11_Time-23-08-28.log\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ArrowFiles <- createArrowFiles(\n",
    "  inputFiles = inputFiles,\n",
    "  sampleNames = \"BMMC_D6T1\",\n",
    "  minTSS = 4, #Dont set this too high because you can always increase later\n",
    "  minFrags = 1000, \n",
    "  addTileMat = TRUE,\n",
    "  addGeneScoreMat = TRUE\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d04cef22-5dc5-4a6d-9f5a-2c8582ef8fbb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Validating Arrows...\n",
      "\n",
      "Getting SampleNames...\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Copying ArrowFiles to Ouptut Directory! If you want to save disk space set copyArrows = FALSE\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Getting Cell Metadata...\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Merging Cell Metadata...\n",
      "\n",
      "Initializing ArchRProject...\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "proj1 <- ArchRProject(\n",
    "  ArrowFiles = ArrowFiles, \n",
    "  outputDirectory = \"BMMC_D6T1\",\n",
    "  copyArrows = TRUE #This is recommened so that you maintain an unaltered copy for later usage.\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "07439ed6-b2c7-4e40-b180-687d1b5dee3e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneScoreMatrix'</li><li>'TileMatrix'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneScoreMatrix'\n",
       "\\item 'TileMatrix'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneScoreMatrix'\n",
       "2. 'TileMatrix'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GeneScoreMatrix\" \"TileMatrix\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getAvailableMatrices(proj1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "49e50b9c-94cf-4980-bf94-82c23686e57e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Checking Inputs...\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addIterativeLSI-11dc38524e87-Date-2024-05-11_Time-23-41-15.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-11 23:41:17 : Computing Total Across All Features, 0.001 mins elapsed.\n",
      "\n",
      "2024-05-11 23:41:21 : Computing Top Features, 0.052 mins elapsed.\n",
      "\n",
      "###########\n",
      "2024-05-11 23:41:23 : Running LSI (1 of 2) on Top Features, 0.092 mins elapsed.\n",
      "###########\n",
      "\n",
      "2024-05-11 23:41:23 : Creating Partial Matrix, 0.094 mins elapsed.\n",
      "\n",
      "2024-05-11 23:41:38 : Computing LSI, 0.347 mins elapsed.\n",
      "\n",
      "2024-05-11 23:42:13 : Identifying Clusters, 0.929 mins elapsed.\n",
      "\n",
      "2024-05-11 23:42:23 : Identified 7 Clusters, 1.1 mins elapsed.\n",
      "\n",
      "2024-05-11 23:42:23 : Saving LSI Iteration, 1.101 mins elapsed.\n",
      "\n",
      "2024-05-11 23:42:44 : Creating Cluster Matrix on the total Group Features, 1.451 mins elapsed.\n",
      "\n",
      "2024-05-11 23:43:39 : Computing Variable Features, 2.361 mins elapsed.\n",
      "\n",
      "###########\n",
      "2024-05-11 23:43:39 : Running LSI (2 of 2) on Variable Features, 2.365 mins elapsed.\n",
      "###########\n",
      "\n",
      "2024-05-11 23:43:39 : Creating Partial Matrix, 2.366 mins elapsed.\n",
      "\n",
      "2024-05-11 23:43:49 : Computing LSI, 2.531 mins elapsed.\n",
      "\n",
      "2024-05-11 23:44:21 : Finished Running IterativeLSI, 3.065 mins elapsed.\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addClusters-11dc353ae3cca-Date-2024-05-11_Time-23-44-21.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-11 23:44:26 : Running Seurats FindClusters (Stuart et al. Cell 2019), 0.003 mins elapsed.\n",
      "\n",
      "Computing nearest neighbor graph\n",
      "\n",
      "Computing SNN\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck\n",
      "\n",
      "Number of nodes: 3413\n",
      "Number of edges: 134360\n",
      "\n",
      "Running Louvain algorithm...\n",
      "Maximum modularity in 10 random starts: 0.8565\n",
      "Number of communities: 13\n",
      "Elapsed time: 0 seconds\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-11 23:44:31 : Testing Outlier Clusters, 0.082 mins elapsed.\n",
      "\n",
      "2024-05-11 23:44:31 : Assigning Cluster Names to 13 Clusters, 0.084 mins elapsed.\n",
      "\n",
      "2024-05-11 23:44:31 : Finished addClusters, 0.087 mins elapsed.\n",
      "\n",
      "23:44:32 UMAP embedding parameters a = 0.583 b = 1.334\n",
      "\n",
      "23:44:32 Read 3413 rows and found 30 numeric columns\n",
      "\n",
      "23:44:32 Using Annoy for neighbor search, n_neighbors = 30\n",
      "\n",
      "23:44:32 Building Annoy index with metric = cosine, n_trees = 50\n",
      "\n",
      "0%   10   20   30   40   50   60   70   80   90   100%\n",
      "\n",
      "[----|----|----|----|----|----|----|----|----|----|\n",
      "\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "|\n",
      "\n",
      "23:44:32 Writing NN index file to temp file /scratch/local/RtmpcNpL1Z/file11dc342aee87e\n",
      "\n",
      "23:44:32 Searching Annoy index using 64 threads, search_k = 3000\n",
      "\n",
      "23:44:33 Annoy recall = 100%\n",
      "\n",
      "23:44:35 Commencing smooth kNN distance calibration using 64 threads\n",
      " with target n_neighbors = 30\n",
      "\n",
      "23:44:36 Initializing from normalized Laplacian + noise (using irlba)\n",
      "\n",
      "23:44:36 Commencing optimization for 500 epochs, with 149508 positive edges\n",
      "\n",
      "23:44:42 Optimization finished\n",
      "\n",
      "23:44:42 Creating temp model dir /scratch/local/RtmpcNpL1Z/dir11dc3473bc76b\n",
      "\n",
      "23:44:42 Creating dir /scratch/local/RtmpcNpL1Z/dir11dc3473bc76b\n",
      "\n",
      "23:44:43 Changing to /scratch/local/RtmpcNpL1Z/dir11dc3473bc76b\n",
      "\n",
      "23:44:43 Creating /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/Embeddings/Save-Uwot-UMAP-Params-IterativeLSI-11dc35270ddc8-Date-2024-05-11_Time-23-44-42.tar\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addGroupCoverages-11dc32e13eac1-Date-2024-05-11_Time-23-44-43.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "C1 (1 of 13) : CellGroups N = 2\n",
      "\n",
      "C2 (2 of 13) : CellGroups N = 2\n",
      "\n",
      "C3 (3 of 13) : CellGroups N = 2\n",
      "\n",
      "C4 (4 of 13) : CellGroups N = 2\n",
      "\n",
      "C5 (5 of 13) : CellGroups N = 2\n",
      "\n",
      "C6 (6 of 13) : CellGroups N = 2\n",
      "\n",
      "C7 (7 of 13) : CellGroups N = 2\n",
      "\n",
      "C8 (8 of 13) : CellGroups N = 2\n",
      "\n",
      "C9 (9 of 13) : CellGroups N = 2\n",
      "\n",
      "C10 (10 of 13) : CellGroups N = 2\n",
      "\n",
      "C11 (11 of 13) : CellGroups N = 2\n",
      "\n",
      "C12 (12 of 13) : CellGroups N = 2\n",
      "\n",
      "C13 (13 of 13) : CellGroups N = 2\n",
      "\n",
      "2024-05-11 23:44:47 : Creating Coverage Files!, 0.07 mins elapsed.\n",
      "\n",
      "2024-05-11 23:44:47 : Batch Execution w/ safelapply!, 0.07 mins elapsed.\n",
      "\n",
      "2024-05-11 23:44:47 : Group C1._.Rep1 (1 of 26) : Creating Group Coverage File : C1._.Rep1.insertions.coverage.h5, 0.071 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:46:00 : Group C1._.Rep2 (2 of 26) : Creating Group Coverage File : C1._.Rep2.insertions.coverage.h5, 1.285 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:47:03 : Group C2._.Rep1 (3 of 26) : Creating Group Coverage File : C2._.Rep1.insertions.coverage.h5, 2.342 mins elapsed.\n",
      "\n",
      "Number of Cells = 202\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:48:09 : Group C2._.Rep2 (4 of 26) : Creating Group Coverage File : C2._.Rep2.insertions.coverage.h5, 3.426 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:49:10 : Group C3._.Rep1 (5 of 26) : Creating Group Coverage File : C3._.Rep1.insertions.coverage.h5, 4.447 mins elapsed.\n",
      "\n",
      "Number of Cells = 236\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:50:14 : Group C3._.Rep2 (6 of 26) : Creating Group Coverage File : C3._.Rep2.insertions.coverage.h5, 5.525 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:51:17 : Group C4._.Rep1 (7 of 26) : Creating Group Coverage File : C4._.Rep1.insertions.coverage.h5, 6.562 mins elapsed.\n",
      "\n",
      "Number of Cells = 89\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:52:21 : Group C4._.Rep2 (8 of 26) : Creating Group Coverage File : C4._.Rep2.insertions.coverage.h5, 7.637 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:53:24 : Group C5._.Rep1 (9 of 26) : Creating Group Coverage File : C5._.Rep1.insertions.coverage.h5, 8.685 mins elapsed.\n",
      "\n",
      "Number of Cells = 216\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:54:30 : Group C5._.Rep2 (10 of 26) : Creating Group Coverage File : C5._.Rep2.insertions.coverage.h5, 9.782 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:55:33 : Group C6._.Rep1 (11 of 26) : Creating Group Coverage File : C6._.Rep1.insertions.coverage.h5, 10.829 mins elapsed.\n",
      "\n",
      "Number of Cells = 24\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:56:35 : Group C6._.Rep2 (12 of 26) : Creating Group Coverage File : C6._.Rep2.insertions.coverage.h5, 11.862 mins elapsed.\n",
      "\n",
      "Number of Cells = 21\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:57:37 : Group C7._.Rep1 (13 of 26) : Creating Group Coverage File : C7._.Rep1.insertions.coverage.h5, 12.899 mins elapsed.\n",
      "\n",
      "Number of Cells = 167\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:58:42 : Group C7._.Rep2 (14 of 26) : Creating Group Coverage File : C7._.Rep2.insertions.coverage.h5, 13.98 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-11 23:59:46 : Group C8._.Rep1 (15 of 26) : Creating Group Coverage File : C8._.Rep1.insertions.coverage.h5, 15.05 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:00:54 : Group C8._.Rep2 (16 of 26) : Creating Group Coverage File : C8._.Rep2.insertions.coverage.h5, 16.185 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:01:56 : Group C9._.Rep1 (17 of 26) : Creating Group Coverage File : C9._.Rep1.insertions.coverage.h5, 17.212 mins elapsed.\n",
      "\n",
      "Number of Cells = 90\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:03:02 : Group C9._.Rep2 (18 of 26) : Creating Group Coverage File : C9._.Rep2.insertions.coverage.h5, 18.313 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:04:06 : Group C10._.Rep1 (19 of 26) : Creating Group Coverage File : C10._.Rep1.insertions.coverage.h5, 19.381 mins elapsed.\n",
      "\n",
      "Number of Cells = 105\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:05:11 : Group C10._.Rep2 (20 of 26) : Creating Group Coverage File : C10._.Rep2.insertions.coverage.h5, 20.471 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:06:15 : Group C11._.Rep1 (21 of 26) : Creating Group Coverage File : C11._.Rep1.insertions.coverage.h5, 21.539 mins elapsed.\n",
      "\n",
      "Number of Cells = 285\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:07:22 : Group C11._.Rep2 (22 of 26) : Creating Group Coverage File : C11._.Rep2.insertions.coverage.h5, 22.646 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:08:24 : Group C12._.Rep1 (23 of 26) : Creating Group Coverage File : C12._.Rep1.insertions.coverage.h5, 23.691 mins elapsed.\n",
      "\n",
      "Number of Cells = 140\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:09:24 : Group C12._.Rep2 (24 of 26) : Creating Group Coverage File : C12._.Rep2.insertions.coverage.h5, 24.678 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:10:14 : Group C13._.Rep1 (25 of 26) : Creating Group Coverage File : C13._.Rep1.insertions.coverage.h5, 25.518 mins elapsed.\n",
      "\n",
      "Number of Cells = 185\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:11:04 : Group C13._.Rep2 (26 of 26) : Creating Group Coverage File : C13._.Rep2.insertions.coverage.h5, 26.344 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-12 00:11:51 : Adding Kmer Bias to Coverage Files!, 27.142 mins elapsed.\n",
      "\n",
      "Kmer Bias chr1 (1 of 24)\n",
      "\n",
      "chr1 \n",
      "Coverage File chr1 (1 of 26)\n",
      "\n",
      "Coverage File chr1 (2 of 26)\n",
      "\n",
      "Coverage File chr1 (3 of 26)\n",
      "\n",
      "Coverage File chr1 (4 of 26)\n",
      "\n",
      "Coverage File chr1 (5 of 26)\n",
      "\n",
      "Coverage File chr1 (6 of 26)\n",
      "\n",
      "Coverage File chr1 (7 of 26)\n",
      "\n",
      "Coverage File chr1 (8 of 26)\n",
      "\n",
      "Coverage File chr1 (9 of 26)\n",
      "\n",
      "Coverage File chr1 (10 of 26)\n",
      "\n",
      "Coverage File chr1 (11 of 26)\n",
      "\n",
      "Coverage File chr1 (12 of 26)\n",
      "\n",
      "Coverage File chr1 (13 of 26)\n",
      "\n",
      "Coverage File chr1 (14 of 26)\n",
      "\n",
      "Coverage File chr1 (15 of 26)\n",
      "\n",
      "Coverage File chr1 (16 of 26)\n",
      "\n",
      "Coverage File chr1 (17 of 26)\n",
      "\n",
      "Coverage File chr1 (18 of 26)\n",
      "\n",
      "Coverage File chr1 (19 of 26)\n",
      "\n",
      "Coverage File chr1 (20 of 26)\n",
      "\n",
      "Coverage File chr1 (21 of 26)\n",
      "\n",
      "Coverage File chr1 (22 of 26)\n",
      "\n",
      "Coverage File chr1 (23 of 26)\n",
      "\n",
      "Coverage File chr1 (24 of 26)\n",
      "\n",
      "Coverage File chr1 (25 of 26)\n",
      "\n",
      "Coverage File chr1 (26 of 26)\n",
      "\n",
      "Kmer Bias chr10 (2 of 24)\n",
      "\n",
      "chr10 \n",
      "Coverage File chr10 (1 of 26)\n",
      "\n",
      "Coverage File chr10 (2 of 26)\n",
      "\n",
      "Coverage File chr10 (3 of 26)\n",
      "\n",
      "Coverage File chr10 (4 of 26)\n",
      "\n",
      "Coverage File chr10 (5 of 26)\n",
      "\n",
      "Coverage File chr10 (6 of 26)\n",
      "\n",
      "Coverage File chr10 (7 of 26)\n",
      "\n",
      "Coverage File chr10 (8 of 26)\n",
      "\n",
      "Coverage File chr10 (9 of 26)\n",
      "\n",
      "Coverage File chr10 (10 of 26)\n",
      "\n",
      "Coverage File chr10 (11 of 26)\n",
      "\n",
      "Coverage File chr10 (12 of 26)\n",
      "\n",
      "Coverage File chr10 (13 of 26)\n",
      "\n",
      "Coverage File chr10 (14 of 26)\n",
      "\n",
      "Coverage File chr10 (15 of 26)\n",
      "\n",
      "Coverage File chr10 (16 of 26)\n",
      "\n",
      "Coverage File chr10 (17 of 26)\n",
      "\n",
      "Coverage File chr10 (18 of 26)\n",
      "\n",
      "Coverage File chr10 (19 of 26)\n",
      "\n",
      "Coverage File chr10 (20 of 26)\n",
      "\n",
      "Coverage File chr10 (21 of 26)\n",
      "\n",
      "Coverage File chr10 (22 of 26)\n",
      "\n",
      "Coverage File chr10 (23 of 26)\n",
      "\n",
      "Coverage File chr10 (24 of 26)\n",
      "\n",
      "Coverage File chr10 (25 of 26)\n",
      "\n",
      "Coverage File chr10 (26 of 26)\n",
      "\n",
      "Kmer Bias chr11 (3 of 24)\n",
      "\n",
      "chr11 \n",
      "Coverage File chr11 (1 of 26)\n",
      "\n",
      "Coverage File chr11 (2 of 26)\n",
      "\n",
      "Coverage File chr11 (3 of 26)\n",
      "\n",
      "Coverage File chr11 (4 of 26)\n",
      "\n",
      "Coverage File chr11 (5 of 26)\n",
      "\n",
      "Coverage File chr11 (6 of 26)\n",
      "\n",
      "Coverage File chr11 (7 of 26)\n",
      "\n",
      "Coverage File chr11 (8 of 26)\n",
      "\n",
      "Coverage File chr11 (9 of 26)\n",
      "\n",
      "Coverage File chr11 (10 of 26)\n",
      "\n",
      "Coverage File chr11 (11 of 26)\n",
      "\n",
      "Coverage File chr11 (12 of 26)\n",
      "\n",
      "Coverage File chr11 (13 of 26)\n",
      "\n",
      "Coverage File chr11 (14 of 26)\n",
      "\n",
      "Coverage File chr11 (15 of 26)\n",
      "\n",
      "Coverage File chr11 (16 of 26)\n",
      "\n",
      "Coverage File chr11 (17 of 26)\n",
      "\n",
      "Coverage File chr11 (18 of 26)\n",
      "\n",
      "Coverage File chr11 (19 of 26)\n",
      "\n",
      "Coverage File chr11 (20 of 26)\n",
      "\n",
      "Coverage File chr11 (21 of 26)\n",
      "\n",
      "Coverage File chr11 (22 of 26)\n",
      "\n",
      "Coverage File chr11 (23 of 26)\n",
      "\n",
      "Coverage File chr11 (24 of 26)\n",
      "\n",
      "Coverage File chr11 (25 of 26)\n",
      "\n",
      "Coverage File chr11 (26 of 26)\n",
      "\n",
      "Kmer Bias chr12 (4 of 24)\n",
      "\n",
      "chr12 \n",
      "Coverage File chr12 (1 of 26)\n",
      "\n",
      "Coverage File chr12 (2 of 26)\n",
      "\n",
      "Coverage File chr12 (3 of 26)\n",
      "\n",
      "Coverage File chr12 (4 of 26)\n",
      "\n",
      "Coverage File chr12 (5 of 26)\n",
      "\n",
      "Coverage File chr12 (6 of 26)\n",
      "\n",
      "Coverage File chr12 (7 of 26)\n",
      "\n",
      "Coverage File chr12 (8 of 26)\n",
      "\n",
      "Coverage File chr12 (9 of 26)\n",
      "\n",
      "Coverage File chr12 (10 of 26)\n",
      "\n",
      "Coverage File chr12 (11 of 26)\n",
      "\n",
      "Coverage File chr12 (12 of 26)\n",
      "\n",
      "Coverage File chr12 (13 of 26)\n",
      "\n",
      "Coverage File chr12 (14 of 26)\n",
      "\n",
      "Coverage File chr12 (15 of 26)\n",
      "\n",
      "Coverage File chr12 (16 of 26)\n",
      "\n",
      "Coverage File chr12 (17 of 26)\n",
      "\n",
      "Coverage File chr12 (18 of 26)\n",
      "\n",
      "Coverage File chr12 (19 of 26)\n",
      "\n",
      "Coverage File chr12 (20 of 26)\n",
      "\n",
      "Coverage File chr12 (21 of 26)\n",
      "\n",
      "Coverage File chr12 (22 of 26)\n",
      "\n",
      "Coverage File chr12 (23 of 26)\n",
      "\n",
      "Coverage File chr12 (24 of 26)\n",
      "\n",
      "Coverage File chr12 (25 of 26)\n",
      "\n",
      "Coverage File chr12 (26 of 26)\n",
      "\n",
      "Kmer Bias chr13 (5 of 24)\n",
      "\n",
      "chr13 \n",
      "Coverage File chr13 (1 of 26)\n",
      "\n",
      "Coverage File chr13 (2 of 26)\n",
      "\n",
      "Coverage File chr13 (3 of 26)\n",
      "\n",
      "Coverage File chr13 (4 of 26)\n",
      "\n",
      "Coverage File chr13 (5 of 26)\n",
      "\n",
      "Coverage File chr13 (6 of 26)\n",
      "\n",
      "Coverage File chr13 (7 of 26)\n",
      "\n",
      "Coverage File chr13 (8 of 26)\n",
      "\n",
      "Coverage File chr13 (9 of 26)\n",
      "\n",
      "Coverage File chr13 (10 of 26)\n",
      "\n",
      "Coverage File chr13 (11 of 26)\n",
      "\n",
      "Coverage File chr13 (12 of 26)\n",
      "\n",
      "Coverage File chr13 (13 of 26)\n",
      "\n",
      "Coverage File chr13 (14 of 26)\n",
      "\n",
      "Coverage File chr13 (15 of 26)\n",
      "\n",
      "Coverage File chr13 (16 of 26)\n",
      "\n",
      "Coverage File chr13 (17 of 26)\n",
      "\n",
      "Coverage File chr13 (18 of 26)\n",
      "\n",
      "Coverage File chr13 (19 of 26)\n",
      "\n",
      "Coverage File chr13 (20 of 26)\n",
      "\n",
      "Coverage File chr13 (21 of 26)\n",
      "\n",
      "Coverage File chr13 (22 of 26)\n",
      "\n",
      "Coverage File chr13 (23 of 26)\n",
      "\n",
      "Coverage File chr13 (24 of 26)\n",
      "\n",
      "Coverage File chr13 (25 of 26)\n",
      "\n",
      "Coverage File chr13 (26 of 26)\n",
      "\n",
      "Kmer Bias chr14 (6 of 24)\n",
      "\n",
      "chr14 \n",
      "Coverage File chr14 (1 of 26)\n",
      "\n",
      "Coverage File chr14 (2 of 26)\n",
      "\n",
      "Coverage File chr14 (3 of 26)\n",
      "\n",
      "Coverage File chr14 (4 of 26)\n",
      "\n",
      "Coverage File chr14 (5 of 26)\n",
      "\n",
      "Coverage File chr14 (6 of 26)\n",
      "\n",
      "Coverage File chr14 (7 of 26)\n",
      "\n",
      "Coverage File chr14 (8 of 26)\n",
      "\n",
      "Coverage File chr14 (9 of 26)\n",
      "\n",
      "Coverage File chr14 (10 of 26)\n",
      "\n",
      "Coverage File chr14 (11 of 26)\n",
      "\n",
      "Coverage File chr14 (12 of 26)\n",
      "\n",
      "Coverage File chr14 (13 of 26)\n",
      "\n",
      "Coverage File chr14 (14 of 26)\n",
      "\n",
      "Coverage File chr14 (15 of 26)\n",
      "\n",
      "Coverage File chr14 (16 of 26)\n",
      "\n",
      "Coverage File chr14 (17 of 26)\n",
      "\n",
      "Coverage File chr14 (18 of 26)\n",
      "\n",
      "Coverage File chr14 (19 of 26)\n",
      "\n",
      "Coverage File chr14 (20 of 26)\n",
      "\n",
      "Coverage File chr14 (21 of 26)\n",
      "\n",
      "Coverage File chr14 (22 of 26)\n",
      "\n",
      "Coverage File chr14 (23 of 26)\n",
      "\n",
      "Coverage File chr14 (24 of 26)\n",
      "\n",
      "Coverage File chr14 (25 of 26)\n",
      "\n",
      "Coverage File chr14 (26 of 26)\n",
      "\n",
      "Kmer Bias chr15 (7 of 24)\n",
      "\n",
      "chr15 \n",
      "Coverage File chr15 (1 of 26)\n",
      "\n",
      "Coverage File chr15 (2 of 26)\n",
      "\n",
      "Coverage File chr15 (3 of 26)\n",
      "\n",
      "Coverage File chr15 (4 of 26)\n",
      "\n",
      "Coverage File chr15 (5 of 26)\n",
      "\n",
      "Coverage File chr15 (6 of 26)\n",
      "\n",
      "Coverage File chr15 (7 of 26)\n",
      "\n",
      "Coverage File chr15 (8 of 26)\n",
      "\n",
      "Coverage File chr15 (9 of 26)\n",
      "\n",
      "Coverage File chr15 (10 of 26)\n",
      "\n",
      "Coverage File chr15 (11 of 26)\n",
      "\n",
      "Coverage File chr15 (12 of 26)\n",
      "\n",
      "Coverage File chr15 (13 of 26)\n",
      "\n",
      "Coverage File chr15 (14 of 26)\n",
      "\n",
      "Coverage File chr15 (15 of 26)\n",
      "\n",
      "Coverage File chr15 (16 of 26)\n",
      "\n",
      "Coverage File chr15 (17 of 26)\n",
      "\n",
      "Coverage File chr15 (18 of 26)\n",
      "\n",
      "Coverage File chr15 (19 of 26)\n",
      "\n",
      "Coverage File chr15 (20 of 26)\n",
      "\n",
      "Coverage File chr15 (21 of 26)\n",
      "\n",
      "Coverage File chr15 (22 of 26)\n",
      "\n",
      "Coverage File chr15 (23 of 26)\n",
      "\n",
      "Coverage File chr15 (24 of 26)\n",
      "\n",
      "Coverage File chr15 (25 of 26)\n",
      "\n",
      "Coverage File chr15 (26 of 26)\n",
      "\n",
      "Kmer Bias chr16 (8 of 24)\n",
      "\n",
      "chr16 \n",
      "Coverage File chr16 (1 of 26)\n",
      "\n",
      "Coverage File chr16 (2 of 26)\n",
      "\n",
      "Coverage File chr16 (3 of 26)\n",
      "\n",
      "Coverage File chr16 (4 of 26)\n",
      "\n",
      "Coverage File chr16 (5 of 26)\n",
      "\n",
      "Coverage File chr16 (6 of 26)\n",
      "\n",
      "Coverage File chr16 (7 of 26)\n",
      "\n",
      "Coverage File chr16 (8 of 26)\n",
      "\n",
      "Coverage File chr16 (9 of 26)\n",
      "\n",
      "Coverage File chr16 (10 of 26)\n",
      "\n",
      "Coverage File chr16 (11 of 26)\n",
      "\n",
      "Coverage File chr16 (12 of 26)\n",
      "\n",
      "Coverage File chr16 (13 of 26)\n",
      "\n",
      "Coverage File chr16 (14 of 26)\n",
      "\n",
      "Coverage File chr16 (15 of 26)\n",
      "\n",
      "Coverage File chr16 (16 of 26)\n",
      "\n",
      "Coverage File chr16 (17 of 26)\n",
      "\n",
      "Coverage File chr16 (18 of 26)\n",
      "\n",
      "Coverage File chr16 (19 of 26)\n",
      "\n",
      "Coverage File chr16 (20 of 26)\n",
      "\n",
      "Coverage File chr16 (21 of 26)\n",
      "\n",
      "Coverage File chr16 (22 of 26)\n",
      "\n",
      "Coverage File chr16 (23 of 26)\n",
      "\n",
      "Coverage File chr16 (24 of 26)\n",
      "\n",
      "Coverage File chr16 (25 of 26)\n",
      "\n",
      "Coverage File chr16 (26 of 26)\n",
      "\n",
      "Kmer Bias chr17 (9 of 24)\n",
      "\n",
      "chr17 \n",
      "Coverage File chr17 (1 of 26)\n",
      "\n",
      "Coverage File chr17 (2 of 26)\n",
      "\n",
      "Coverage File chr17 (3 of 26)\n",
      "\n",
      "Coverage File chr17 (4 of 26)\n",
      "\n",
      "Coverage File chr17 (5 of 26)\n",
      "\n",
      "Coverage File chr17 (6 of 26)\n",
      "\n",
      "Coverage File chr17 (7 of 26)\n",
      "\n",
      "Coverage File chr17 (8 of 26)\n",
      "\n",
      "Coverage File chr17 (9 of 26)\n",
      "\n",
      "Coverage File chr17 (10 of 26)\n",
      "\n",
      "Coverage File chr17 (11 of 26)\n",
      "\n",
      "Coverage File chr17 (12 of 26)\n",
      "\n",
      "Coverage File chr17 (13 of 26)\n",
      "\n",
      "Coverage File chr17 (14 of 26)\n",
      "\n",
      "Coverage File chr17 (15 of 26)\n",
      "\n",
      "Coverage File chr17 (16 of 26)\n",
      "\n",
      "Coverage File chr17 (17 of 26)\n",
      "\n",
      "Coverage File chr17 (18 of 26)\n",
      "\n",
      "Coverage File chr17 (19 of 26)\n",
      "\n",
      "Coverage File chr17 (20 of 26)\n",
      "\n",
      "Coverage File chr17 (21 of 26)\n",
      "\n",
      "Coverage File chr17 (22 of 26)\n",
      "\n",
      "Coverage File chr17 (23 of 26)\n",
      "\n",
      "Coverage File chr17 (24 of 26)\n",
      "\n",
      "Coverage File chr17 (25 of 26)\n",
      "\n",
      "Coverage File chr17 (26 of 26)\n",
      "\n",
      "Kmer Bias chr18 (10 of 24)\n",
      "\n",
      "chr18 \n",
      "Coverage File chr18 (1 of 26)\n",
      "\n",
      "Coverage File chr18 (2 of 26)\n",
      "\n",
      "Coverage File chr18 (3 of 26)\n",
      "\n",
      "Coverage File chr18 (4 of 26)\n",
      "\n",
      "Coverage File chr18 (5 of 26)\n",
      "\n",
      "Coverage File chr18 (6 of 26)\n",
      "\n",
      "Coverage File chr18 (7 of 26)\n",
      "\n",
      "Coverage File chr18 (8 of 26)\n",
      "\n",
      "Coverage File chr18 (9 of 26)\n",
      "\n",
      "Coverage File chr18 (10 of 26)\n",
      "\n",
      "Coverage File chr18 (11 of 26)\n",
      "\n",
      "Coverage File chr18 (12 of 26)\n",
      "\n",
      "Coverage File chr18 (13 of 26)\n",
      "\n",
      "Coverage File chr18 (14 of 26)\n",
      "\n",
      "Coverage File chr18 (15 of 26)\n",
      "\n",
      "Coverage File chr18 (16 of 26)\n",
      "\n",
      "Coverage File chr18 (17 of 26)\n",
      "\n",
      "Coverage File chr18 (18 of 26)\n",
      "\n",
      "Coverage File chr18 (19 of 26)\n",
      "\n",
      "Coverage File chr18 (20 of 26)\n",
      "\n",
      "Coverage File chr18 (21 of 26)\n",
      "\n",
      "Coverage File chr18 (22 of 26)\n",
      "\n",
      "Coverage File chr18 (23 of 26)\n",
      "\n",
      "Coverage File chr18 (24 of 26)\n",
      "\n",
      "Coverage File chr18 (25 of 26)\n",
      "\n",
      "Coverage File chr18 (26 of 26)\n",
      "\n",
      "Kmer Bias chr19 (11 of 24)\n",
      "\n",
      "chr19 \n",
      "Coverage File chr19 (1 of 26)\n",
      "\n",
      "Coverage File chr19 (2 of 26)\n",
      "\n",
      "Coverage File chr19 (3 of 26)\n",
      "\n",
      "Coverage File chr19 (4 of 26)\n",
      "\n",
      "Coverage File chr19 (5 of 26)\n",
      "\n",
      "Coverage File chr19 (6 of 26)\n",
      "\n",
      "Coverage File chr19 (7 of 26)\n",
      "\n",
      "Coverage File chr19 (8 of 26)\n",
      "\n",
      "Coverage File chr19 (9 of 26)\n",
      "\n",
      "Coverage File chr19 (10 of 26)\n",
      "\n",
      "Coverage File chr19 (11 of 26)\n",
      "\n",
      "Coverage File chr19 (12 of 26)\n",
      "\n",
      "Coverage File chr19 (13 of 26)\n",
      "\n",
      "Coverage File chr19 (14 of 26)\n",
      "\n",
      "Coverage File chr19 (15 of 26)\n",
      "\n",
      "Coverage File chr19 (16 of 26)\n",
      "\n",
      "Coverage File chr19 (17 of 26)\n",
      "\n",
      "Coverage File chr19 (18 of 26)\n",
      "\n",
      "Coverage File chr19 (19 of 26)\n",
      "\n",
      "Coverage File chr19 (20 of 26)\n",
      "\n",
      "Coverage File chr19 (21 of 26)\n",
      "\n",
      "Coverage File chr19 (22 of 26)\n",
      "\n",
      "Coverage File chr19 (23 of 26)\n",
      "\n",
      "Coverage File chr19 (24 of 26)\n",
      "\n",
      "Coverage File chr19 (25 of 26)\n",
      "\n",
      "Coverage File chr19 (26 of 26)\n",
      "\n",
      "Kmer Bias chr2 (12 of 24)\n",
      "\n",
      "chr2 \n",
      "Coverage File chr2 (1 of 26)\n",
      "\n",
      "Coverage File chr2 (2 of 26)\n",
      "\n",
      "Coverage File chr2 (3 of 26)\n",
      "\n",
      "Coverage File chr2 (4 of 26)\n",
      "\n",
      "Coverage File chr2 (5 of 26)\n",
      "\n",
      "Coverage File chr2 (6 of 26)\n",
      "\n",
      "Coverage File chr2 (7 of 26)\n",
      "\n",
      "Coverage File chr2 (8 of 26)\n",
      "\n",
      "Coverage File chr2 (9 of 26)\n",
      "\n",
      "Coverage File chr2 (10 of 26)\n",
      "\n",
      "Coverage File chr2 (11 of 26)\n",
      "\n",
      "Coverage File chr2 (12 of 26)\n",
      "\n",
      "Coverage File chr2 (13 of 26)\n",
      "\n",
      "Coverage File chr2 (14 of 26)\n",
      "\n",
      "Coverage File chr2 (15 of 26)\n",
      "\n",
      "Coverage File chr2 (16 of 26)\n",
      "\n",
      "Coverage File chr2 (17 of 26)\n",
      "\n",
      "Coverage File chr2 (18 of 26)\n",
      "\n",
      "Coverage File chr2 (19 of 26)\n",
      "\n",
      "Coverage File chr2 (20 of 26)\n",
      "\n",
      "Coverage File chr2 (21 of 26)\n",
      "\n",
      "Coverage File chr2 (22 of 26)\n",
      "\n",
      "Coverage File chr2 (23 of 26)\n",
      "\n",
      "Coverage File chr2 (24 of 26)\n",
      "\n",
      "Coverage File chr2 (25 of 26)\n",
      "\n",
      "Coverage File chr2 (26 of 26)\n",
      "\n",
      "Kmer Bias chr20 (13 of 24)\n",
      "\n",
      "chr20 \n",
      "Coverage File chr20 (1 of 26)\n",
      "\n",
      "Coverage File chr20 (2 of 26)\n",
      "\n",
      "Coverage File chr20 (3 of 26)\n",
      "\n",
      "Coverage File chr20 (4 of 26)\n",
      "\n",
      "Coverage File chr20 (5 of 26)\n",
      "\n",
      "Coverage File chr20 (6 of 26)\n",
      "\n",
      "Coverage File chr20 (7 of 26)\n",
      "\n",
      "Coverage File chr20 (8 of 26)\n",
      "\n",
      "Coverage File chr20 (9 of 26)\n",
      "\n",
      "Coverage File chr20 (10 of 26)\n",
      "\n",
      "Coverage File chr20 (11 of 26)\n",
      "\n",
      "Coverage File chr20 (12 of 26)\n",
      "\n",
      "Coverage File chr20 (13 of 26)\n",
      "\n",
      "Coverage File chr20 (14 of 26)\n",
      "\n",
      "Coverage File chr20 (15 of 26)\n",
      "\n",
      "Coverage File chr20 (16 of 26)\n",
      "\n",
      "Coverage File chr20 (17 of 26)\n",
      "\n",
      "Coverage File chr20 (18 of 26)\n",
      "\n",
      "Coverage File chr20 (19 of 26)\n",
      "\n",
      "Coverage File chr20 (20 of 26)\n",
      "\n",
      "Coverage File chr20 (21 of 26)\n",
      "\n",
      "Coverage File chr20 (22 of 26)\n",
      "\n",
      "Coverage File chr20 (23 of 26)\n",
      "\n",
      "Coverage File chr20 (24 of 26)\n",
      "\n",
      "Coverage File chr20 (25 of 26)\n",
      "\n",
      "Coverage File chr20 (26 of 26)\n",
      "\n",
      "Kmer Bias chr21 (14 of 24)\n",
      "\n",
      "chr21 \n",
      "Coverage File chr21 (1 of 26)\n",
      "\n",
      "Coverage File chr21 (2 of 26)\n",
      "\n",
      "Coverage File chr21 (3 of 26)\n",
      "\n",
      "Coverage File chr21 (4 of 26)\n",
      "\n",
      "Coverage File chr21 (5 of 26)\n",
      "\n",
      "Coverage File chr21 (6 of 26)\n",
      "\n",
      "Coverage File chr21 (7 of 26)\n",
      "\n",
      "Coverage File chr21 (8 of 26)\n",
      "\n",
      "Coverage File chr21 (9 of 26)\n",
      "\n",
      "Coverage File chr21 (10 of 26)\n",
      "\n",
      "Coverage File chr21 (11 of 26)\n",
      "\n",
      "Coverage File chr21 (12 of 26)\n",
      "\n",
      "Coverage File chr21 (13 of 26)\n",
      "\n",
      "Coverage File chr21 (14 of 26)\n",
      "\n",
      "Coverage File chr21 (15 of 26)\n",
      "\n",
      "Coverage File chr21 (16 of 26)\n",
      "\n",
      "Coverage File chr21 (17 of 26)\n",
      "\n",
      "Coverage File chr21 (18 of 26)\n",
      "\n",
      "Coverage File chr21 (19 of 26)\n",
      "\n",
      "Coverage File chr21 (20 of 26)\n",
      "\n",
      "Coverage File chr21 (21 of 26)\n",
      "\n",
      "Coverage File chr21 (22 of 26)\n",
      "\n",
      "Coverage File chr21 (23 of 26)\n",
      "\n",
      "Coverage File chr21 (24 of 26)\n",
      "\n",
      "Coverage File chr21 (25 of 26)\n",
      "\n",
      "Coverage File chr21 (26 of 26)\n",
      "\n",
      "Kmer Bias chr22 (15 of 24)\n",
      "\n",
      "chr22 \n",
      "Coverage File chr22 (1 of 26)\n",
      "\n",
      "Coverage File chr22 (2 of 26)\n",
      "\n",
      "Coverage File chr22 (3 of 26)\n",
      "\n",
      "Coverage File chr22 (4 of 26)\n",
      "\n",
      "Coverage File chr22 (5 of 26)\n",
      "\n",
      "Coverage File chr22 (6 of 26)\n",
      "\n",
      "Coverage File chr22 (7 of 26)\n",
      "\n",
      "Coverage File chr22 (8 of 26)\n",
      "\n",
      "Coverage File chr22 (9 of 26)\n",
      "\n",
      "Coverage File chr22 (10 of 26)\n",
      "\n",
      "Coverage File chr22 (11 of 26)\n",
      "\n",
      "Coverage File chr22 (12 of 26)\n",
      "\n",
      "Coverage File chr22 (13 of 26)\n",
      "\n",
      "Coverage File chr22 (14 of 26)\n",
      "\n",
      "Coverage File chr22 (15 of 26)\n",
      "\n",
      "Coverage File chr22 (16 of 26)\n",
      "\n",
      "Coverage File chr22 (17 of 26)\n",
      "\n",
      "Coverage File chr22 (18 of 26)\n",
      "\n",
      "Coverage File chr22 (19 of 26)\n",
      "\n",
      "Coverage File chr22 (20 of 26)\n",
      "\n",
      "Coverage File chr22 (21 of 26)\n",
      "\n",
      "Coverage File chr22 (22 of 26)\n",
      "\n",
      "Coverage File chr22 (23 of 26)\n",
      "\n",
      "Coverage File chr22 (24 of 26)\n",
      "\n",
      "Coverage File chr22 (25 of 26)\n",
      "\n",
      "Coverage File chr22 (26 of 26)\n",
      "\n",
      "Kmer Bias chr3 (16 of 24)\n",
      "\n",
      "chr3 \n",
      "Coverage File chr3 (1 of 26)\n",
      "\n",
      "Coverage File chr3 (2 of 26)\n",
      "\n",
      "Coverage File chr3 (3 of 26)\n",
      "\n",
      "Coverage File chr3 (4 of 26)\n",
      "\n",
      "Coverage File chr3 (5 of 26)\n",
      "\n",
      "Coverage File chr3 (6 of 26)\n",
      "\n",
      "Coverage File chr3 (7 of 26)\n",
      "\n",
      "Coverage File chr3 (8 of 26)\n",
      "\n",
      "Coverage File chr3 (9 of 26)\n",
      "\n",
      "Coverage File chr3 (10 of 26)\n",
      "\n",
      "Coverage File chr3 (11 of 26)\n",
      "\n",
      "Coverage File chr3 (12 of 26)\n",
      "\n",
      "Coverage File chr3 (13 of 26)\n",
      "\n",
      "Coverage File chr3 (14 of 26)\n",
      "\n",
      "Coverage File chr3 (15 of 26)\n",
      "\n",
      "Coverage File chr3 (16 of 26)\n",
      "\n",
      "Coverage File chr3 (17 of 26)\n",
      "\n",
      "Coverage File chr3 (18 of 26)\n",
      "\n",
      "Coverage File chr3 (19 of 26)\n",
      "\n",
      "Coverage File chr3 (20 of 26)\n",
      "\n",
      "Coverage File chr3 (21 of 26)\n",
      "\n",
      "Coverage File chr3 (22 of 26)\n",
      "\n",
      "Coverage File chr3 (23 of 26)\n",
      "\n",
      "Coverage File chr3 (24 of 26)\n",
      "\n",
      "Coverage File chr3 (25 of 26)\n",
      "\n",
      "Coverage File chr3 (26 of 26)\n",
      "\n",
      "Kmer Bias chr4 (17 of 24)\n",
      "\n",
      "chr4 \n",
      "Coverage File chr4 (1 of 26)\n",
      "\n",
      "Coverage File chr4 (2 of 26)\n",
      "\n",
      "Coverage File chr4 (3 of 26)\n",
      "\n",
      "Coverage File chr4 (4 of 26)\n",
      "\n",
      "Coverage File chr4 (5 of 26)\n",
      "\n",
      "Coverage File chr4 (6 of 26)\n",
      "\n",
      "Coverage File chr4 (7 of 26)\n",
      "\n",
      "Coverage File chr4 (8 of 26)\n",
      "\n",
      "Coverage File chr4 (9 of 26)\n",
      "\n",
      "Coverage File chr4 (10 of 26)\n",
      "\n",
      "Coverage File chr4 (11 of 26)\n",
      "\n",
      "Coverage File chr4 (12 of 26)\n",
      "\n",
      "Coverage File chr4 (13 of 26)\n",
      "\n",
      "Coverage File chr4 (14 of 26)\n",
      "\n",
      "Coverage File chr4 (15 of 26)\n",
      "\n",
      "Coverage File chr4 (16 of 26)\n",
      "\n",
      "Coverage File chr4 (17 of 26)\n",
      "\n",
      "Coverage File chr4 (18 of 26)\n",
      "\n",
      "Coverage File chr4 (19 of 26)\n",
      "\n",
      "Coverage File chr4 (20 of 26)\n",
      "\n",
      "Coverage File chr4 (21 of 26)\n",
      "\n",
      "Coverage File chr4 (22 of 26)\n",
      "\n",
      "Coverage File chr4 (23 of 26)\n",
      "\n",
      "Coverage File chr4 (24 of 26)\n",
      "\n",
      "Coverage File chr4 (25 of 26)\n",
      "\n",
      "Coverage File chr4 (26 of 26)\n",
      "\n",
      "Kmer Bias chr5 (18 of 24)\n",
      "\n",
      "chr5 \n",
      "Coverage File chr5 (1 of 26)\n",
      "\n",
      "Coverage File chr5 (2 of 26)\n",
      "\n",
      "Coverage File chr5 (3 of 26)\n",
      "\n",
      "Coverage File chr5 (4 of 26)\n",
      "\n",
      "Coverage File chr5 (5 of 26)\n",
      "\n",
      "Coverage File chr5 (6 of 26)\n",
      "\n",
      "Coverage File chr5 (7 of 26)\n",
      "\n",
      "Coverage File chr5 (8 of 26)\n",
      "\n",
      "Coverage File chr5 (9 of 26)\n",
      "\n",
      "Coverage File chr5 (10 of 26)\n",
      "\n",
      "Coverage File chr5 (11 of 26)\n",
      "\n",
      "Coverage File chr5 (12 of 26)\n",
      "\n",
      "Coverage File chr5 (13 of 26)\n",
      "\n",
      "Coverage File chr5 (14 of 26)\n",
      "\n",
      "Coverage File chr5 (15 of 26)\n",
      "\n",
      "Coverage File chr5 (16 of 26)\n",
      "\n",
      "Coverage File chr5 (17 of 26)\n",
      "\n",
      "Coverage File chr5 (18 of 26)\n",
      "\n",
      "Coverage File chr5 (19 of 26)\n",
      "\n",
      "Coverage File chr5 (20 of 26)\n",
      "\n",
      "Coverage File chr5 (21 of 26)\n",
      "\n",
      "Coverage File chr5 (22 of 26)\n",
      "\n",
      "Coverage File chr5 (23 of 26)\n",
      "\n",
      "Coverage File chr5 (24 of 26)\n",
      "\n",
      "Coverage File chr5 (25 of 26)\n",
      "\n",
      "Coverage File chr5 (26 of 26)\n",
      "\n",
      "Kmer Bias chr6 (19 of 24)\n",
      "\n",
      "chr6 \n",
      "Coverage File chr6 (1 of 26)\n",
      "\n",
      "Coverage File chr6 (2 of 26)\n",
      "\n",
      "Coverage File chr6 (3 of 26)\n",
      "\n",
      "Coverage File chr6 (4 of 26)\n",
      "\n",
      "Coverage File chr6 (5 of 26)\n",
      "\n",
      "Coverage File chr6 (6 of 26)\n",
      "\n",
      "Coverage File chr6 (7 of 26)\n",
      "\n",
      "Coverage File chr6 (8 of 26)\n",
      "\n",
      "Coverage File chr6 (9 of 26)\n",
      "\n",
      "Coverage File chr6 (10 of 26)\n",
      "\n",
      "Coverage File chr6 (11 of 26)\n",
      "\n",
      "Coverage File chr6 (12 of 26)\n",
      "\n",
      "Coverage File chr6 (13 of 26)\n",
      "\n",
      "Coverage File chr6 (14 of 26)\n",
      "\n",
      "Coverage File chr6 (15 of 26)\n",
      "\n",
      "Coverage File chr6 (16 of 26)\n",
      "\n",
      "Coverage File chr6 (17 of 26)\n",
      "\n",
      "Coverage File chr6 (18 of 26)\n",
      "\n",
      "Coverage File chr6 (19 of 26)\n",
      "\n",
      "Coverage File chr6 (20 of 26)\n",
      "\n",
      "Coverage File chr6 (21 of 26)\n",
      "\n",
      "Coverage File chr6 (22 of 26)\n",
      "\n",
      "Coverage File chr6 (23 of 26)\n",
      "\n",
      "Coverage File chr6 (24 of 26)\n",
      "\n",
      "Coverage File chr6 (25 of 26)\n",
      "\n",
      "Coverage File chr6 (26 of 26)\n",
      "\n",
      "Kmer Bias chr7 (20 of 24)\n",
      "\n",
      "chr7 \n",
      "Coverage File chr7 (1 of 26)\n",
      "\n",
      "Coverage File chr7 (2 of 26)\n",
      "\n",
      "Coverage File chr7 (3 of 26)\n",
      "\n",
      "Coverage File chr7 (4 of 26)\n",
      "\n",
      "Coverage File chr7 (5 of 26)\n",
      "\n",
      "Coverage File chr7 (6 of 26)\n",
      "\n",
      "Coverage File chr7 (7 of 26)\n",
      "\n",
      "Coverage File chr7 (8 of 26)\n",
      "\n",
      "Coverage File chr7 (9 of 26)\n",
      "\n",
      "Coverage File chr7 (10 of 26)\n",
      "\n",
      "Coverage File chr7 (11 of 26)\n",
      "\n",
      "Coverage File chr7 (12 of 26)\n",
      "\n",
      "Coverage File chr7 (13 of 26)\n",
      "\n",
      "Coverage File chr7 (14 of 26)\n",
      "\n",
      "Coverage File chr7 (15 of 26)\n",
      "\n",
      "Coverage File chr7 (16 of 26)\n",
      "\n",
      "Coverage File chr7 (17 of 26)\n",
      "\n",
      "Coverage File chr7 (18 of 26)\n",
      "\n",
      "Coverage File chr7 (19 of 26)\n",
      "\n",
      "Coverage File chr7 (20 of 26)\n",
      "\n",
      "Coverage File chr7 (21 of 26)\n",
      "\n",
      "Coverage File chr7 (22 of 26)\n",
      "\n",
      "Coverage File chr7 (23 of 26)\n",
      "\n",
      "Coverage File chr7 (24 of 26)\n",
      "\n",
      "Coverage File chr7 (25 of 26)\n",
      "\n",
      "Coverage File chr7 (26 of 26)\n",
      "\n",
      "Kmer Bias chr8 (21 of 24)\n",
      "\n",
      "chr8 \n",
      "Coverage File chr8 (1 of 26)\n",
      "\n",
      "Coverage File chr8 (2 of 26)\n",
      "\n",
      "Coverage File chr8 (3 of 26)\n",
      "\n",
      "Coverage File chr8 (4 of 26)\n",
      "\n",
      "Coverage File chr8 (5 of 26)\n",
      "\n",
      "Coverage File chr8 (6 of 26)\n",
      "\n",
      "Coverage File chr8 (7 of 26)\n",
      "\n",
      "Coverage File chr8 (8 of 26)\n",
      "\n",
      "Coverage File chr8 (9 of 26)\n",
      "\n",
      "Coverage File chr8 (10 of 26)\n",
      "\n",
      "Coverage File chr8 (11 of 26)\n",
      "\n",
      "Coverage File chr8 (12 of 26)\n",
      "\n",
      "Coverage File chr8 (13 of 26)\n",
      "\n",
      "Coverage File chr8 (14 of 26)\n",
      "\n",
      "Coverage File chr8 (15 of 26)\n",
      "\n",
      "Coverage File chr8 (16 of 26)\n",
      "\n",
      "Coverage File chr8 (17 of 26)\n",
      "\n",
      "Coverage File chr8 (18 of 26)\n",
      "\n",
      "Coverage File chr8 (19 of 26)\n",
      "\n",
      "Coverage File chr8 (20 of 26)\n",
      "\n",
      "Coverage File chr8 (21 of 26)\n",
      "\n",
      "Coverage File chr8 (22 of 26)\n",
      "\n",
      "Coverage File chr8 (23 of 26)\n",
      "\n",
      "Coverage File chr8 (24 of 26)\n",
      "\n",
      "Coverage File chr8 (25 of 26)\n",
      "\n",
      "Coverage File chr8 (26 of 26)\n",
      "\n",
      "Kmer Bias chr9 (22 of 24)\n",
      "\n",
      "chr9 \n",
      "Coverage File chr9 (1 of 26)\n",
      "\n",
      "Coverage File chr9 (2 of 26)\n",
      "\n",
      "Coverage File chr9 (3 of 26)\n",
      "\n",
      "Coverage File chr9 (4 of 26)\n",
      "\n",
      "Coverage File chr9 (5 of 26)\n",
      "\n",
      "Coverage File chr9 (6 of 26)\n",
      "\n",
      "Coverage File chr9 (7 of 26)\n",
      "\n",
      "Coverage File chr9 (8 of 26)\n",
      "\n",
      "Coverage File chr9 (9 of 26)\n",
      "\n",
      "Coverage File chr9 (10 of 26)\n",
      "\n",
      "Coverage File chr9 (11 of 26)\n",
      "\n",
      "Coverage File chr9 (12 of 26)\n",
      "\n",
      "Coverage File chr9 (13 of 26)\n",
      "\n",
      "Coverage File chr9 (14 of 26)\n",
      "\n",
      "Coverage File chr9 (15 of 26)\n",
      "\n",
      "Coverage File chr9 (16 of 26)\n",
      "\n",
      "Coverage File chr9 (17 of 26)\n",
      "\n",
      "Coverage File chr9 (18 of 26)\n",
      "\n",
      "Coverage File chr9 (19 of 26)\n",
      "\n",
      "Coverage File chr9 (20 of 26)\n",
      "\n",
      "Coverage File chr9 (21 of 26)\n",
      "\n",
      "Coverage File chr9 (22 of 26)\n",
      "\n",
      "Coverage File chr9 (23 of 26)\n",
      "\n",
      "Coverage File chr9 (24 of 26)\n",
      "\n",
      "Coverage File chr9 (25 of 26)\n",
      "\n",
      "Coverage File chr9 (26 of 26)\n",
      "\n",
      "Kmer Bias chrX (23 of 24)\n",
      "\n",
      "chrX \n",
      "Coverage File chrX (1 of 26)\n",
      "\n",
      "Coverage File chrX (2 of 26)\n",
      "\n",
      "Coverage File chrX (3 of 26)\n",
      "\n",
      "Coverage File chrX (4 of 26)\n",
      "\n",
      "Coverage File chrX (5 of 26)\n",
      "\n",
      "Coverage File chrX (6 of 26)\n",
      "\n",
      "Coverage File chrX (7 of 26)\n",
      "\n",
      "Coverage File chrX (8 of 26)\n",
      "\n",
      "Coverage File chrX (9 of 26)\n",
      "\n",
      "Coverage File chrX (10 of 26)\n",
      "\n",
      "Coverage File chrX (11 of 26)\n",
      "\n",
      "Coverage File chrX (12 of 26)\n",
      "\n",
      "Coverage File chrX (13 of 26)\n",
      "\n",
      "Coverage File chrX (14 of 26)\n",
      "\n",
      "Coverage File chrX (15 of 26)\n",
      "\n",
      "Coverage File chrX (16 of 26)\n",
      "\n",
      "Coverage File chrX (17 of 26)\n",
      "\n",
      "Coverage File chrX (18 of 26)\n",
      "\n",
      "Coverage File chrX (19 of 26)\n",
      "\n",
      "Coverage File chrX (20 of 26)\n",
      "\n",
      "Coverage File chrX (21 of 26)\n",
      "\n",
      "Coverage File chrX (22 of 26)\n",
      "\n",
      "Coverage File chrX (23 of 26)\n",
      "\n",
      "Coverage File chrX (24 of 26)\n",
      "\n",
      "Coverage File chrX (25 of 26)\n",
      "\n",
      "Coverage File chrX (26 of 26)\n",
      "\n",
      "Kmer Bias chrY (24 of 24)\n",
      "\n",
      "chrY \n",
      "Coverage File chrY (1 of 26)\n",
      "\n",
      "Coverage File chrY (2 of 26)\n",
      "\n",
      "Coverage File chrY (3 of 26)\n",
      "\n",
      "Coverage File chrY (4 of 26)\n",
      "\n",
      "Coverage File chrY (5 of 26)\n",
      "\n",
      "Coverage File chrY (6 of 26)\n",
      "\n",
      "Coverage File chrY (7 of 26)\n",
      "\n",
      "Coverage File chrY (8 of 26)\n",
      "\n",
      "Coverage File chrY (9 of 26)\n",
      "\n",
      "Coverage File chrY (10 of 26)\n",
      "\n",
      "Coverage File chrY (11 of 26)\n",
      "\n",
      "Coverage File chrY (12 of 26)\n",
      "\n",
      "Coverage File chrY (13 of 26)\n",
      "\n",
      "Coverage File chrY (14 of 26)\n",
      "\n",
      "Coverage File chrY (15 of 26)\n",
      "\n",
      "Coverage File chrY (16 of 26)\n",
      "\n",
      "Coverage File chrY (17 of 26)\n",
      "\n",
      "Coverage File chrY (18 of 26)\n",
      "\n",
      "Coverage File chrY (19 of 26)\n",
      "\n",
      "Coverage File chrY (20 of 26)\n",
      "\n",
      "Coverage File chrY (21 of 26)\n",
      "\n",
      "Coverage File chrY (22 of 26)\n",
      "\n",
      "Coverage File chrY (23 of 26)\n",
      "\n",
      "Coverage File chrY (24 of 26)\n",
      "\n",
      "Coverage File chrY (25 of 26)\n",
      "\n",
      "Coverage File chrY (26 of 26)\n",
      "\n",
      "Completed Kmer Bias Calculation\n",
      "\n",
      "Adding Kmer Bias (1 of 26)\n",
      "\n",
      "Adding Kmer Bias (2 of 26)\n",
      "\n",
      "Adding Kmer Bias (3 of 26)\n",
      "\n",
      "Adding Kmer Bias (4 of 26)\n",
      "\n",
      "Adding Kmer Bias (5 of 26)\n",
      "\n",
      "Adding Kmer Bias (6 of 26)\n",
      "\n",
      "Adding Kmer Bias (7 of 26)\n",
      "\n",
      "Adding Kmer Bias (8 of 26)\n",
      "\n",
      "Adding Kmer Bias (9 of 26)\n",
      "\n",
      "Adding Kmer Bias (10 of 26)\n",
      "\n",
      "Adding Kmer Bias (11 of 26)\n",
      "\n",
      "Adding Kmer Bias (12 of 26)\n",
      "\n",
      "Adding Kmer Bias (13 of 26)\n",
      "\n",
      "Adding Kmer Bias (14 of 26)\n",
      "\n",
      "Adding Kmer Bias (15 of 26)\n",
      "\n",
      "Adding Kmer Bias (16 of 26)\n",
      "\n",
      "Adding Kmer Bias (17 of 26)\n",
      "\n",
      "Adding Kmer Bias (18 of 26)\n",
      "\n",
      "Adding Kmer Bias (19 of 26)\n",
      "\n",
      "Adding Kmer Bias (20 of 26)\n",
      "\n",
      "Adding Kmer Bias (21 of 26)\n",
      "\n",
      "Adding Kmer Bias (22 of 26)\n",
      "\n",
      "Adding Kmer Bias (23 of 26)\n",
      "\n",
      "Adding Kmer Bias (24 of 26)\n",
      "\n",
      "Adding Kmer Bias (25 of 26)\n",
      "\n",
      "Adding Kmer Bias (26 of 26)\n",
      "\n",
      "2024-05-12 00:26:19 : Finished Creation of Coverage Files!, 41.606 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addGroupCoverages-11dc32e13eac1-Date-2024-05-11_Time-23-44-43.log\n",
      "\n",
      "Searching For MACS2..\n",
      "\n",
      "Found with $path!\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-11dc33bc896a0-Date-2024-05-12_Time-00-26-25.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "Calling Peaks with Macs2\n",
      "\n",
      "2024-05-12 00:26:29 : Peak Calling Parameters!, 0.072 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Group nCells nCellsUsed nReplicates nMin nMax maxPeaks\n",
      "C1     C1    610        540           2   40  500   150000\n",
      "C2     C2    242        242           2   40  202   121000\n",
      "C3     C3    276        276           2   40  236   138000\n",
      "C4     C4    129        129           2   40   89    64500\n",
      "C5     C5    256        256           2   40  216   128000\n",
      "C6     C6     37         34           2   21   24    17000\n",
      "C7     C7    207        207           2   40  167   103500\n",
      "C8     C8    651        540           2   40  500   150000\n",
      "C9     C9    130        130           2   40   90    65000\n",
      "C10   C10    145        145           2   40  105    72500\n",
      "C11   C11    325        325           2   40  285   150000\n",
      "C12   C12    180        180           2   40  140    90000\n",
      "C13   C13    225        225           2   40  185   112500\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-12 00:26:30 : Batching Peak Calls!, 0.081 mins elapsed.\n",
      "\n",
      "2024-05-12 00:26:31 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2024-05-12 00:26:31 : Group 1 of 24, Calling Peaks with MACS2!, 0.001 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C1._.Rep1-1 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C1._.Rep1-1.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:28:04 : Group 2 of 24, Calling Peaks with MACS2!, 1.549 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C1._.Rep2-2 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C1._.Rep2-2.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:28:30 : Group 3 of 24, Calling Peaks with MACS2!, 1.979 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C2._.Rep1-3 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C2._.Rep1-3.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:29:30 : Group 4 of 24, Calling Peaks with MACS2!, 2.986 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C2._.Rep2-4 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C2._.Rep2-4.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:29:56 : Group 5 of 24, Calling Peaks with MACS2!, 3.423 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C3._.Rep1-5 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C3._.Rep1-5.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:31:02 : Group 6 of 24, Calling Peaks with MACS2!, 4.526 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C3._.Rep2-6 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C3._.Rep2-6.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:31:27 : Group 7 of 24, Calling Peaks with MACS2!, 4.94 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C4._.Rep1-7 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C4._.Rep1-7.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:32:19 : Group 8 of 24, Calling Peaks with MACS2!, 5.809 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C4._.Rep2-8 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C4._.Rep2-8.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:32:49 : Group 9 of 24, Calling Peaks with MACS2!, 6.294 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C5._.Rep1-9 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C5._.Rep1-9.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:33:59 : Group 10 of 24, Calling Peaks with MACS2!, 7.469 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C5._.Rep2-10 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C5._.Rep2-10.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:34:25 : Group 11 of 24, Calling Peaks with MACS2!, 7.906 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C7._.Rep1-11 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C7._.Rep1-11.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:35:36 : Group 12 of 24, Calling Peaks with MACS2!, 9.092 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C7._.Rep2-12 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C7._.Rep2-12.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:36:01 : Group 13 of 24, Calling Peaks with MACS2!, 9.507 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C8._.Rep1-13 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C8._.Rep1-13.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:37:30 : Group 14 of 24, Calling Peaks with MACS2!, 10.988 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C8._.Rep2-14 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C8._.Rep2-14.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:37:57 : Group 15 of 24, Calling Peaks with MACS2!, 11.441 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C9._.Rep1-15 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C9._.Rep1-15.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:39:09 : Group 16 of 24, Calling Peaks with MACS2!, 12.635 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C9._.Rep2-16 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C9._.Rep2-16.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:40:22 : Group 17 of 24, Calling Peaks with MACS2!, 13.851 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C10._.Rep1-17 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C10._.Rep1-17.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:41:05 : Group 18 of 24, Calling Peaks with MACS2!, 14.57 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C10._.Rep2-18 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C10._.Rep2-18.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:41:27 : Group 19 of 24, Calling Peaks with MACS2!, 14.932 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C11._.Rep1-19 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C11._.Rep1-19.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:42:33 : Group 20 of 24, Calling Peaks with MACS2!, 16.039 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C11._.Rep2-20 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C11._.Rep2-20.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:42:58 : Group 21 of 24, Calling Peaks with MACS2!, 16.454 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C12._.Rep1-21 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C12._.Rep1-21.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:44:46 : Group 22 of 24, Calling Peaks with MACS2!, 18.254 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C12._.Rep2-22 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C12._.Rep2-22.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:45:34 : Group 23 of 24, Calling Peaks with MACS2!, 19.049 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C13._.Rep1-23 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C13._.Rep1-23.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:46:22 : Group 24 of 24, Calling Peaks with MACS2!, 19.855 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C13._.Rep2-24 --treatment /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds/C13._.Rep2-24.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-12 00:46:48 : Identifying Reproducible Peaks!, 20.383 mins elapsed.\n",
      "\n",
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C1-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C10-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C11-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C12-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C13-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C2-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C3-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C4-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C5-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C7-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C8-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/notebooks/Fig6-Granja_PBMC/Fig6-apply_Cellcano/BMMC_D6T1/PeakCalls/C9-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-12 00:47:48 : Creating Union Peak Set!, 21.381 mins elapsed.\n",
      "\n",
      "Converged after 9 iterations!\n",
      "\n",
      "Plotting Ggplot!\n",
      "\n",
      "2024-05-12 00:47:54 : Finished Creating Union Peak Set (97474)!, 21.495 mins elapsed.\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-11dc335022f1f-Date-2024-05-12_Time-00-47-54.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-12 00:47:55 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2024-05-12 00:47:57 : Adding BMMC_D6T1 to PeakMatrix for Chr (1 of 23)!, 0.021 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:03 : Adding BMMC_D6T1 to PeakMatrix for Chr (2 of 23)!, 0.125 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:09 : Adding BMMC_D6T1 to PeakMatrix for Chr (3 of 23)!, 0.221 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:14 : Adding BMMC_D6T1 to PeakMatrix for Chr (4 of 23)!, 0.309 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:19 : Adding BMMC_D6T1 to PeakMatrix for Chr (5 of 23)!, 0.387 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:24 : Adding BMMC_D6T1 to PeakMatrix for Chr (6 of 23)!, 0.471 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:29 : Adding BMMC_D6T1 to PeakMatrix for Chr (7 of 23)!, 0.556 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:34 : Adding BMMC_D6T1 to PeakMatrix for Chr (8 of 23)!, 0.64 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:39 : Adding BMMC_D6T1 to PeakMatrix for Chr (9 of 23)!, 0.719 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:43 : Adding BMMC_D6T1 to PeakMatrix for Chr (10 of 23)!, 0.798 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:50 : Adding BMMC_D6T1 to PeakMatrix for Chr (11 of 23)!, 0.899 mins elapsed.\n",
      "\n",
      "2024-05-12 00:48:55 : Adding BMMC_D6T1 to PeakMatrix for Chr (12 of 23)!, 0.985 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:00 : Adding BMMC_D6T1 to PeakMatrix for Chr (13 of 23)!, 1.067 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:04 : Adding BMMC_D6T1 to PeakMatrix for Chr (14 of 23)!, 1.137 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:09 : Adding BMMC_D6T1 to PeakMatrix for Chr (15 of 23)!, 1.215 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:13 : Adding BMMC_D6T1 to PeakMatrix for Chr (16 of 23)!, 1.289 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:18 : Adding BMMC_D6T1 to PeakMatrix for Chr (17 of 23)!, 1.368 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:23 : Adding BMMC_D6T1 to PeakMatrix for Chr (18 of 23)!, 1.453 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:27 : Adding BMMC_D6T1 to PeakMatrix for Chr (19 of 23)!, 1.522 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:32 : Adding BMMC_D6T1 to PeakMatrix for Chr (20 of 23)!, 1.614 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:37 : Adding BMMC_D6T1 to PeakMatrix for Chr (21 of 23)!, 1.687 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:41 : Adding BMMC_D6T1 to PeakMatrix for Chr (22 of 23)!, 1.753 mins elapsed.\n",
      "\n",
      "2024-05-12 00:49:45 : Adding BMMC_D6T1 to PeakMatrix for Chr (23 of 23)!, 1.823 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-11dc335022f1f-Date-2024-05-12_Time-00-47-54.log\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-11dc36f8eacae-Date-2024-05-12_Time-00-49-50.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-12 00:50:05 : Organizing colData, 0.265 mins elapsed.\n",
      "\n",
      "2024-05-12 00:50:05 : Organizing rowData, 0.265 mins elapsed.\n",
      "\n",
      "2024-05-12 00:50:05 : Organizing rowRanges, 0.266 mins elapsed.\n",
      "\n",
      "2024-05-12 00:50:06 : Organizing Assays (1 of 1), 0.267 mins elapsed.\n",
      "\n",
      "2024-05-12 00:50:06 : Constructing SummarizedExperiment, 0.269 mins elapsed.\n",
      "\n",
      "2024-05-12 00:50:07 : Finished Matrix Creation, 0.288 mins elapsed.\n",
      "\n",
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 2.5 GiB”\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "out_path = \"/project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/05_peaks_and_genesScore_matrix_ArchR/input_for_EpiAnno/\" # where to save resulting matrices | with / at the end\n",
    "threads_=1 # how many threads to use\n",
    "\n",
    "proj1 <- addIterativeLSI(\n",
    "    ArchRProj = proj1,\n",
    "    useMatrix = \"TileMatrix\", \n",
    "    name = \"IterativeLSI\", \n",
    "    iterations = 2, \n",
    "    clusterParams = list( #See Seurat::FindClusters\n",
    "        resolution = c(0.2), \n",
    "        sampleCells = 10000, \n",
    "        n.start = 10\n",
    "    ), \n",
    "    varFeatures = 25000, \n",
    "    dimsToUse = 1:30\n",
    ")\n",
    "\n",
    "proj1 <- addClusters(\n",
    "    input = proj1,\n",
    "    reducedDims = \"IterativeLSI\",\n",
    "    method = \"Seurat\",\n",
    "    name = \"Clusters\",\n",
    "    resolution = 0.8\n",
    ")\n",
    "\n",
    "proj1 <- addUMAP(\n",
    "    ArchRProj = proj1, \n",
    "    reducedDims = \"IterativeLSI\", \n",
    "    name = \"UMAP\", \n",
    "    nNeighbors = 30, \n",
    "    minDist = 0.5, \n",
    "    metric = \"cosine\"\n",
    ")\n",
    "\n",
    "HDF5_USE_FILE_LOCKING=FALSE\n",
    "RHDF5_USE_FILE_LOCKING=FALSE\n",
    "\n",
    "proj2 <- addGroupCoverages(ArchRProj = proj1, groupBy = \"Clusters\")\n",
    "\n",
    "pathToMacs2 <- findMacs2()\n",
    "\n",
    "proj2 <- addReproduciblePeakSet(\n",
    "    ArchRProj = proj2, \n",
    "    groupBy = \"Clusters\", \n",
    "    pathToMacs2 = pathToMacs2\n",
    ")\n",
    "\n",
    "\n",
    "#saveArchRProject(ArchRProj = proj2, outputDirectory = \"Save-Proj2_afterPeakCalling\", load = FALSE)\n",
    "\n",
    "proj3 <- addPeakMatrix(proj2)\n",
    "\n",
    "peakMatrix <- getMatrixFromProject(proj3, useMatrix='PeakMatrix')\n",
    "\n",
    "PeakMatrix_proj3_dense =  as.matrix(peakMatrix@assays@data@listData$PeakMatrix)\n",
    "\n",
    "write.csv(PeakMatrix_proj3_dense, file = paste(out_path,\"PeakMatrix_dense.csv\",sep=\"\" )) \n",
    "\n",
    "# get the peak regions\n",
    "proj3_peak_metadata  = (getPeakSet(proj3))\n",
    "proj3_peak_regions = data.frame(seqnames = seqnames(proj3_peak_metadata),\n",
    "           start= start(proj3_peak_metadata),\n",
    "            end = end(proj3_peak_metadata))\n",
    "\n",
    "write.csv(proj3_peak_metadata, file =paste(out_path,\"peak_metadata.csv\",sep=\"\" ))\n",
    "write.csv(proj3_peak_regions, file =paste(out_path,\"peak_regions.csv\",sep=\"\" ))\n",
    "write.table(proj3_peak_regions, file =paste(out_path,\"peak_regions.tsv\",sep=\"\" ),col.names = F,row.names=FALSE,sep=\"\\t\", quote = FALSE)\n",
    "\n",
    "# save the peak matrix as sparse mtx\n",
    "\n",
    "\n",
    "# save sparse matrix\n",
    "sparse_matrix <- Matrix(PeakMatrix_proj3_dense , sparse = T )\n",
    "writeMM(obj = sparse_matrix, file = paste(out_path,\"ArchR_peakMatrix.mtx\",sep=\"\" ))\n",
    "\n",
    "# save genes and cells names\n",
    "combined_peak_features =  paste(proj3_peak_regions$seqnames,\"_\",proj3_peak_regions$start,\"_\",proj3_peak_regions$end, sep=\"\")\n",
    "\n",
    "write.table(as.data.frame(combined_peak_features), file = paste(out_path,\"ArchR_peaks.tsv\",sep=\"\" ) , row.names = FALSE, quote = FALSE, col.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9c7d00c-078a-4a19-a4dd-70e00a8db9f0",
   "metadata": {},
   "outputs": [],
   "source": [
    "peakMatrix <- getMatrixFromProject(proj3, useMatrix='PeakMatrix')\n",
    "\n",
    "PeakMatrix_proj3_dense =  as.matrix(peakMatrix@assays@data@listData$PeakMatrix)\n",
    "\n",
    "write.csv(PeakMatrix_proj3_dense, file = paste(out_path,\"PeakMatrix_dense.csv\",sep=\"\" )) \n",
    "\n",
    "# get the peak regions\n",
    "proj3_peak_metadata  = (getPeakSet(proj3))\n",
    "proj3_peak_regions = data.frame(seqnames = seqnames(proj3_peak_metadata),\n",
    "           start= start(proj3_peak_metadata),\n",
    "            end = end(proj3_peak_metadata))\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "ec641abe-7a74-4eff-8955-7dd0fb133c68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneScoreMatrix'</li><li>'PeakMatrix'</li><li>'TileMatrix'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneScoreMatrix'\n",
       "\\item 'PeakMatrix'\n",
       "\\item 'TileMatrix'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneScoreMatrix'\n",
       "2. 'PeakMatrix'\n",
       "3. 'TileMatrix'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GeneScoreMatrix\" \"PeakMatrix\"      \"TileMatrix\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getAvailableMatrices(proj3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "634cea34-1618-4af5-a338-b8d1129febc5",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-11dc348026e88-Date-2024-05-12_Time-10-40-27.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-12 10:40:40 : Organizing colData, 0.212 mins elapsed.\n",
      "\n",
      "2024-05-12 10:40:40 : Organizing rowData, 0.213 mins elapsed.\n",
      "\n",
      "2024-05-12 10:40:40 : Organizing rowRanges, 0.213 mins elapsed.\n",
      "\n",
      "2024-05-12 10:40:40 : Organizing Assays (1 of 1), 0.214 mins elapsed.\n",
      "\n",
      "2024-05-12 10:40:40 : Constructing SummarizedExperiment, 0.214 mins elapsed.\n",
      "\n",
      "2024-05-12 10:40:41 : Finished Matrix Creation, 0.234 mins elapsed.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# get the gene activity matrix \n",
    "\n",
    "geneScoreMatrix <- getMatrixFromProject(proj3, useMatrix='GeneScoreMatrix')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "796184fd-9434-4a48-bb24-65013c0b1580",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "DataFrame with 24919 rows and 6 columns\n",
       "      seqnames     start       end  strand         name     idx\n",
       "         <Rle>   <array>   <array> <array>      <array> <array>\n",
       "1         chr1     69091     70008       1        OR4F5       1\n",
       "2         chr1    140339    139790       2    LOC729737       2\n",
       "3         chr1    629010    585990       2 LOC101928626       3\n",
       "4         chr1    817371    819837       1       FAM87B       4\n",
       "5         chr1    825138    859446       1    LINC01128       5\n",
       "...        ...       ...       ...     ...          ...     ...\n",
       "24915     chrX 155264589 155258241       2       RAB39B     952\n",
       "24916     chrX 155334657 155276211       2        CLIC2     953\n",
       "24917     chrX 155380787 155381134       1       H2AFB1     954\n",
       "24918     chrX 155382115 155383230       1         F8A2     955\n",
       "24919     chrX 155669944 155490115       2        TMLHE     956"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "rowData(geneScoreMatrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "0e137642-cad3-4bea-9545-45607fac5cbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>24919</li><li>3413</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 24919\n",
       "\\item 3413\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 24919\n",
       "2. 3413\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] 24919  3413"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dim(geneScoreMatrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "67599673-dcde-44df-9050-ec01bd7afe57",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneScore_outDir = \"/project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/05_peaks_and_genesScore_matrix_ArchR/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "496d2733-3552-4ae8-b1db-adf8f6e9cb1c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "write.table(rowData(geneScoreMatrix)$name, paste(geneScore_outDir, \"ArchR_genescore_genes.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "id": "4c9cc218-b028-4c12-a674-81ada3922ff6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "write.table(colnames(geneScoreMatrix), paste(geneScore_outDir, \"ArchR_genescore_barcodes.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "d6731a5a-e4a1-4705-b0bc-d7b4bb555241",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "geneScoreMatrix_sparse = geneScoreMatrix@assays@data@listData$GeneScoreMatrix\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "cbfdd6a6-c095-49d2-8fd7-c99b7db87d96",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "NULL"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "writeMM(obj = geneScoreMatrix_sparse, file=paste(geneScore_outDir, \"ArchR_genescore.mtx\", sep=\"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "3bbb363a-a304-46eb-ba54-880d0eae5b93",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Copying ArchRProject to new outputDirectory : /project/scATAC_analysis/scATAcat_review/data/Granja2019/BMMC_D6T1/05_peaks_and_genesScore_matrix_ArchR/Save-Proj\n",
      "\n",
      "Copying Arrow Files...\n",
      "\n",
      "Copying Arrow Files (1 of 1)\n",
      "\n",
      "Getting ImputeWeights\n",
      "\n",
      "No imputeWeights found, returning NULL\n",
      "\n",
      "Copying Other Files...\n",
      "\n",
      "Copying Other Files (1 of 5): Embeddings\n",
      "\n",
      "Copying Other Files (2 of 5): GroupCoverages\n",
      "\n",
      "Copying Other Files (3 of 5): IterativeLSI\n",
      "\n",
      "Copying Other Files (4 of 5): PeakCalls\n",
      "\n",
      "Copying Other Files (5 of 5): Plots\n",
      "\n",
      "Saving ArchRProject...\n",
      "\n"
     ]
    }
   ],
   "source": [
    "saveArchRProject(ArchRProj = proj3, outputDirectory = paste(geneScore_outDir,\"Save-Proj\", sep=\"\"), load = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "66a37710-4da9-4dc8-8982-2be6bbf0d1d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(stringr)\n",
    "\n",
    "write.table(as.data.frame(str_split_fixed(colnames(geneScoreMatrix),'#',2))$V2, paste(geneScore_outDir, \"cell_IDs.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db834ae7-f775-48d4-b0ba-5b161fc7a39d",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R 4.1.0",
   "language": "R",
   "name": "ir4.1.0"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
