{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "099c50f6-93aa-4c54-988d-90cfadde27b7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n",
      "ArchR : Version 1.0.2\n",
      "For more information see our website : www.ArchRProject.com\n",
      "If you encounter a bug please report : https://github.com/GreenleafLab/ArchR/issues\n",
      "\n",
      "Loading Required Packages...\n",
      "\n",
      "\tLoading Package : grid v4.3.0\n",
      "\n",
      "\tLoading Package : gridExtra v2.3\n",
      "\n",
      "\tLoading Package : gtools v3.9.4\n",
      "\n",
      "\tLoading Package : gtable v0.3.5\n",
      "\n",
      "\tLoading Package : ggplot2 v3.5.1\n",
      "\n",
      "\tLoading Package : magrittr v2.0.3\n",
      "\n",
      "\tLoading Package : plyr v1.8.8\n",
      "\n",
      "\tLoading Package : stringr v1.5.1\n",
      "\n",
      "\tLoading Package : data.table v1.15.4\n",
      "\n",
      "\tLoading Package : matrixStats v0.63.0\n",
      "\n",
      "\tLoading Package : S4Vectors v0.38.1\n",
      "\n",
      "\tLoading Package : GenomicRanges v1.52.0\n",
      "\n",
      "\tLoading Package : BiocGenerics v0.46.0\n",
      "\n",
      "\tLoading Package : Matrix v1.6.4\n",
      "\n",
      "\tLoading Package : Rcpp v1.0.12\n",
      "\n",
      "\tLoading Package : SummarizedExperiment v1.30.1\n",
      "\n",
      "\tLoading Package : rhdf5 v2.44.0\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(ArchR)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "303e9ed0-50df-4cc3-af8f-0601909c3c4d",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "'/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR'"
      ],
      "text/latex": [
       "'/project/scATAC\\_analysis/scATAcat\\_review/preprocessing/Corces2020\\_brain\\_scATAC/04\\_get\\_peaks\\_and\\_genesScore\\_matrix\\_ArchR'"
      ],
      "text/markdown": [
       "'/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR'"
      ],
      "text/plain": [
       "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getwd()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "caf5245a-27ea-48a9-838c-9b73fbd1c995",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Setting default number of Parallel threads to 1.\n",
      "\n",
      "Setting default genome to Hg38.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "set.seed(2022)\n",
    "addArchRThreads(threads = 1) \n",
    "addArchRGenome(\"hg38\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d87f9fb3-ffb4-442b-a0d0-5a2ca4367402",
   "metadata": {},
   "outputs": [],
   "source": [
    "inputFiles = \"/project/scATAC_analysis/10X_scATAC_HumanBrain_Corces2020/analysis/get_ENCODE_cCRE_by_cell_coverage/outs/human_brain_Corces2020_atac_fragments_only_stdChr.tsv.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "84df57d8-0603-45d5-b47d-8fe24cbc4e13",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-createArrows-18ed61112e72e-Date-2024-05-27_Time-18-22-13.555721.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "Cleaning Temporary Files\n",
      "\n",
      "2024-05-27 18:22:16.550348 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "(Corces2020_brain : 1 of 1) Determining Arrow Method to use!\n",
      "\n",
      "Attempting to index /project/scATAC_analysis/10X_scATAC_HumanBrain_Corces2020/analysis/get_ENCODE_cCRE_by_cell_coverage/outs/human_brain_Corces2020_atac_fragments_only_stdChr.tsv.gz as tabix..\n",
      "\n",
      "2024-05-27 18:23:08.374785 : (Corces2020_brain : 1 of 1) Reading In Fragments from inputFiles (readMethod = tabix), 0.864 mins elapsed.\n",
      "\n",
      "2024-05-27 18:23:08.487443 : (Corces2020_brain : 1 of 1) Tabix Bed To Temporary File, 0.866 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:26:07.983155 : (Corces2020_brain : 1 of 1) Reading TabixFile 8 Percent, 3.857 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:28:18.213758 : (Corces2020_brain : 1 of 1) Reading TabixFile 17 Percent, 6.028 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:30:11.882851 : (Corces2020_brain : 1 of 1) Reading TabixFile 25 Percent, 7.922 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:31:47.171072 : (Corces2020_brain : 1 of 1) Reading TabixFile 33 Percent, 9.51 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:33:16.85569 : (Corces2020_brain : 1 of 1) Reading TabixFile 42 Percent, 11.005 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:34:52.944662 : (Corces2020_brain : 1 of 1) Reading TabixFile 50 Percent, 12.607 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:36:00.284822 : (Corces2020_brain : 1 of 1) Reading TabixFile 58 Percent, 13.729 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:37:08.615857 : (Corces2020_brain : 1 of 1) Reading TabixFile 67 Percent, 14.868 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:38:15.7564 : (Corces2020_brain : 1 of 1) Reading TabixFile 75 Percent, 15.987 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:39:17.170582 : (Corces2020_brain : 1 of 1) Reading TabixFile 83 Percent, 17.01 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:39:55.459355 : (Corces2020_brain : 1 of 1) Reading TabixFile 92 Percent, 17.648 mins elapsed.\n",
      "\n",
      "Warning message in sprintf(\"%s Reading TabixFile %s Percent\", prefix, round(100 * :\n",
      "“one argument not used by format '%s Reading TabixFile %s Percent'”\n",
      "2024-05-27 18:40:33.859537 : (Corces2020_brain : 1 of 1) Reading TabixFile 100 Percent, 18.288 mins elapsed.\n",
      "\n",
      "2024-05-27 18:40:35.861336 : (Corces2020_brain : 1 of 1) Successful creation of Temporary File, 18.322 mins elapsed.\n",
      "\n",
      "2024-05-27 18:40:35.93993 : (Corces2020_brain : 1 of 1) Creating ArrowFile From Temporary File, 18.323 mins elapsed.\n",
      "\n",
      "2024-05-27 18:45:22.339868 : (Corces2020_brain : 1 of 1) Successful creation of Arrow File, 23.096 mins elapsed.\n",
      "\n",
      "2024-05-27 18:45:59.293487 : Continuing through after error ggplot for Fragment Size Distribution, 23.712 mins elapsed.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "2024-05-27 18:48:41.702911 : (Corces2020_brain : 1 of 1) CellStats : Number of Cells Pass Filter = 11941 , 26.419 mins elapsed.\n",
      "\n",
      "2024-05-27 18:48:41.741646 : (Corces2020_brain : 1 of 1) CellStats : Median Frags = 8618 , 26.42 mins elapsed.\n",
      "\n",
      "2024-05-27 18:48:41.78325 : (Corces2020_brain : 1 of 1) CellStats : Median TSS Enrichment = 10.42 , 26.421 mins elapsed.\n",
      "\n",
      "2024-05-27 18:48:43.11966 : Continuing through after error ggplot for TSS by Frags, 26.443 mins elapsed.\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "2024-05-27 18:48:45.49043 : (Corces2020_brain : 1 of 1) Adding Additional Feature Counts!, 26.482 mins elapsed.\n",
      "\n",
      "2024-05-27 18:51:19.722883 : (Corces2020_brain : 1 of 1) Removing Fragments from Filtered Cells, 29.053 mins elapsed.\n",
      "\n",
      "2024-05-27 18:51:19.762133 : (Corces2020_brain : 1 of 1) Creating Filtered Arrow File, 29.054 mins elapsed.\n",
      "\n",
      "2024-05-27 18:52:45.262613 : (Corces2020_brain : 1 of 1) Finished Constructing Filtered Arrow File!, 30.479 mins elapsed.\n",
      "\n",
      "2024-05-27 18:52:46.316478 : (Corces2020_brain : 1 of 1) Adding TileMatrix!, 30.496 mins elapsed.\n",
      "\n",
      "2024-05-27 18:59:35.2156 : (Corces2020_brain : 1 of 1) Adding GeneScoreMatrix!, 37.311 mins elapsed.\n",
      "\n",
      "2024-05-27 19:04:57.362307 : (Corces2020_brain : 1 of 1) Finished Creating Arrow File, 42.68 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-createArrows-18ed61112e72e-Date-2024-05-27_Time-18-22-13.555721.log\n",
      "\n"
     ]
    }
   ],
   "source": [
    "ArrowFiles <- createArrowFiles(\n",
    "  inputFiles = inputFiles,\n",
    "  sampleNames = \"Corces2020_brain\",\n",
    "  minTSS = 4, #Dont set this too high because you can always increase later\n",
    "  minFrags = 1000, \n",
    "  addTileMat = TRUE,\n",
    "  addGeneScoreMat = TRUE\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "d04cef22-5dc5-4a6d-9f5a-2c8582ef8fbb",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Using GeneAnnotation set by addArchRGenome(Hg38)!\n",
      "\n",
      "Validating Arrows...\n",
      "\n",
      "Getting SampleNames...\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Copying ArrowFiles to Ouptut Directory! If you want to save disk space set copyArrows = FALSE\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Getting Cell Metadata...\n",
      "\n",
      "1 \n",
      "\n",
      "\n",
      "Merging Cell Metadata...\n",
      "\n",
      "Initializing ArchRProject...\n",
      "\n",
      "\n",
      "                                                   / |\n",
      "                                                 /    \\\n",
      "            .                                  /      |.\n",
      "            \\\\\\                              /        |.\n",
      "              \\\\\\                          /           `|.\n",
      "                \\\\\\                      /              |.\n",
      "                  \\                    /                |\\\n",
      "                  \\\\#####\\           /                  ||\n",
      "                ==###########>      /                   ||\n",
      "                 \\\\##==......\\    /                     ||\n",
      "            ______ =       =|__ /__                     ||      \\\\\\\n",
      "        ,--' ,----`-,__ ___/'  --,-`-===================##========>\n",
      "       \\               '        ##_______ _____ ,--,__,=##,__   ///\n",
      "        ,    __==    ___,-,__,--'#'  ==='      `-'    | ##,-/\n",
      "        -,____,---'       \\\\####\\\\________________,--\\\\_##,/\n",
      "           ___      .______        ______  __    __  .______      \n",
      "          /   \\     |   _  \\      /      ||  |  |  | |   _  \\     \n",
      "         /  ^  \\    |  |_)  |    |  ,----'|  |__|  | |  |_)  |    \n",
      "        /  /_\\  \\   |      /     |  |     |   __   | |      /     \n",
      "       /  _____  \\  |  |\\  \\\\___ |  `----.|  |  |  | |  |\\  \\\\___.\n",
      "      /__/     \\__\\ | _| `._____| \\______||__|  |__| | _| `._____|\n",
      "    \n",
      "\n"
     ]
    }
   ],
   "source": [
    "proj1 <- ArchRProject(\n",
    "  ArrowFiles = ArrowFiles, \n",
    "  outputDirectory = \"Corces2020_brain\",\n",
    "  copyArrows = TRUE #This is recommened so that you maintain an unaltered copy for later usage.\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "07439ed6-b2c7-4e40-b180-687d1b5dee3e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneScoreMatrix'</li><li>'TileMatrix'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneScoreMatrix'\n",
       "\\item 'TileMatrix'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneScoreMatrix'\n",
       "2. 'TileMatrix'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GeneScoreMatrix\" \"TileMatrix\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getAvailableMatrices(proj1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "49e50b9c-94cf-4980-bf94-82c23686e57e",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Checking Inputs...\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addIterativeLSI-18ed6680a37bd-Date-2024-05-27_Time-19-05-19.489029.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-27 19:05:22.471525 : Computing Total Across All Features, 0.001 mins elapsed.\n",
      "\n",
      "2024-05-27 19:05:25.371045 : Computing Top Features, 0.049 mins elapsed.\n",
      "\n",
      "###########\n",
      "2024-05-27 19:05:27.212371 : Running LSI (1 of 2) on Top Features, 0.08 mins elapsed.\n",
      "###########\n",
      "\n",
      "2024-05-27 19:05:27.363389 : Sampling Cells (N = 10000) for Estimated LSI, 0.082 mins elapsed.\n",
      "\n",
      "2024-05-27 19:05:27.412642 : Creating Sampled Partial Matrix, 0.083 mins elapsed.\n",
      "\n",
      "2024-05-27 19:05:58.886744 : Computing Estimated LSI (projectAll = FALSE), 0.608 mins elapsed.\n",
      "\n",
      "2024-05-27 19:07:50.514752 : Identifying Clusters, 2.468 mins elapsed.\n",
      "\n",
      "2024-05-27 19:08:20.849721 : Identified 7 Clusters, 2.974 mins elapsed.\n",
      "\n",
      "2024-05-27 19:08:21.05506 : Saving LSI Iteration, 2.977 mins elapsed.\n",
      "\n",
      "Found more than one class \"dist\" in cache; using the first, from namespace 'BiocGenerics'\n",
      "\n",
      "Also defined by ‘spam’\n",
      "\n",
      "Found more than one class \"dist\" in cache; using the first, from namespace 'BiocGenerics'\n",
      "\n",
      "Also defined by ‘spam’\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "************************************************************\n",
      "2024-05-27 19:08:50.814721 : ERROR Found in .saveIteration for  \n",
      "LogFile = ArchRLogs/ArchR-addIterativeLSI-18ed6680a37bd-Date-2024-05-27_Time-19-05-19.489029.log\n",
      "\n",
      "<simpleError in g$grobs[[legend]]: no such index at level 2\n",
      ">\n",
      "\n",
      "************************************************************\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-27 19:08:50.821408 : Creating Cluster Matrix on the total Group Features, 3.473 mins elapsed.\n",
      "\n",
      "2024-05-27 19:10:06.213235 : Computing Variable Features, 4.73 mins elapsed.\n",
      "\n",
      "###########\n",
      "2024-05-27 19:10:06.539005 : Running LSI (2 of 2) on Variable Features, 4.735 mins elapsed.\n",
      "###########\n",
      "\n",
      "2024-05-27 19:10:06.640842 : Creating Partial Matrix, 4.737 mins elapsed.\n",
      "\n",
      "2024-05-27 19:10:40.457752 : Computing LSI, 5.301 mins elapsed.\n",
      "\n",
      "2024-05-27 19:12:08.40527 : Finished Running IterativeLSI, 6.766 mins elapsed.\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addClusters-18ed64e22c52a-Date-2024-05-27_Time-19-12-08.521957.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-27 19:12:11.890805 : Running Seurats FindClusters (Stuart et al. Cell 2019), 0.003 mins elapsed.\n",
      "\n",
      "Computing nearest neighbor graph\n",
      "\n",
      "Computing SNN\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Modularity Optimizer version 1.3.0 by Ludo Waltman and Nees Jan van Eck\n",
      "\n",
      "Number of nodes: 11941\n",
      "Number of edges: 504895\n",
      "\n",
      "Running Louvain algorithm...\n",
      "Maximum modularity in 10 random starts: 0.8829\n",
      "Number of communities: 16\n",
      "Elapsed time: 2 seconds\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "1 singletons identified. 15 final clusters.\n",
      "\n",
      "2024-05-27 19:12:29.76863 : Testing Biased Clusters, 0.301 mins elapsed.\n",
      "\n",
      "2024-05-27 19:12:29.948263 : Testing Outlier Clusters, 0.304 mins elapsed.\n",
      "\n",
      "2024-05-27 19:12:30.01018 : Assigning Cluster Names to 15 Clusters, 0.305 mins elapsed.\n",
      "\n",
      "2024-05-27 19:12:30.156479 : Finished addClusters, 0.307 mins elapsed.\n",
      "\n",
      "19:12:30 UMAP embedding parameters a = 0.583 b = 1.334\n",
      "\n",
      "Found more than one class \"dist\" in cache; using the first, from namespace 'BiocGenerics'\n",
      "\n",
      "Also defined by ‘spam’\n",
      "\n",
      "19:12:30 Read 11941 rows and found 30 numeric columns\n",
      "\n",
      "19:12:30 Using Annoy for neighbor search, n_neighbors = 30\n",
      "\n",
      "Found more than one class \"dist\" in cache; using the first, from namespace 'BiocGenerics'\n",
      "\n",
      "Also defined by ‘spam’\n",
      "\n",
      "19:12:30 Building Annoy index with metric = cosine, n_trees = 50\n",
      "\n",
      "0%   10   20   30   40   50   60   70   80   90   100%\n",
      "\n",
      "[----|----|----|----|----|----|----|----|----|----|\n",
      "\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "*\n",
      "|\n",
      "\n",
      "19:12:32 Writing NN index file to temp file /scratch/local/RtmpO5jTyj/file18ed64a83e44b\n",
      "\n",
      "19:12:32 Searching Annoy index using 64 threads, search_k = 3000\n",
      "\n",
      "19:12:32 Annoy recall = 100%\n",
      "\n",
      "19:12:34 Commencing smooth kNN distance calibration using 64 threads\n",
      " with target n_neighbors = 30\n",
      "\n",
      "19:12:36 Initializing from normalized Laplacian + noise (using irlba)\n",
      "\n",
      "19:12:37 Commencing optimization for 200 epochs, with 568390 positive edges\n",
      "\n",
      "19:12:44 Optimization finished\n",
      "\n",
      "19:12:44 Creating temp model dir /scratch/local/RtmpO5jTyj/dir18ed62d547887\n",
      "\n",
      "19:12:44 Creating dir /scratch/local/RtmpO5jTyj/dir18ed62d547887\n",
      "\n",
      "19:12:45 Changing to /scratch/local/RtmpO5jTyj/dir18ed62d547887\n",
      "\n",
      "19:12:45 Creating /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/Embeddings/Save-Uwot-UMAP-Params-IterativeLSI-18ed64808a729-Date-2024-05-27_Time-19-12-44.596137.tar\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addGroupCoverages-18ed634361199-Date-2024-05-27_Time-19-12-46.448318.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "C1 (1 of 15) : CellGroups N = 2\n",
      "\n",
      "C2 (2 of 15) : CellGroups N = 2\n",
      "\n",
      "C3 (3 of 15) : CellGroups N = 2\n",
      "\n",
      "C4 (4 of 15) : CellGroups N = 2\n",
      "\n",
      "C5 (5 of 15) : CellGroups N = 2\n",
      "\n",
      "C6 (6 of 15) : CellGroups N = 2\n",
      "\n",
      "C7 (7 of 15) : CellGroups N = 2\n",
      "\n",
      "C8 (8 of 15) : CellGroups N = 2\n",
      "\n",
      "C9 (9 of 15) : CellGroups N = 2\n",
      "\n",
      "C10 (10 of 15) : CellGroups N = 2\n",
      "\n",
      "C11 (11 of 15) : CellGroups N = 2\n",
      "\n",
      "C12 (12 of 15) : CellGroups N = 2\n",
      "\n",
      "C13 (13 of 15) : CellGroups N = 2\n",
      "\n",
      "C14 (14 of 15) : CellGroups N = 2\n",
      "\n",
      "C15 (15 of 15) : CellGroups N = 2\n",
      "\n",
      "2024-05-27 19:12:52.008705 : Creating Coverage Files!, 0.093 mins elapsed.\n",
      "\n",
      "2024-05-27 19:12:52.011829 : Batch Execution w/ safelapply!, 0.093 mins elapsed.\n",
      "\n",
      "2024-05-27 19:12:52.063783 : Group C1._.Rep1 (1 of 30) : Creating Group Coverage File : C1._.Rep1.insertions.coverage.h5, 0.094 mins elapsed.\n",
      "\n",
      "Number of Cells = 275\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:13:59.608126 : Group C1._.Rep2 (2 of 30) : Creating Group Coverage File : C1._.Rep2.insertions.coverage.h5, 1.219 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:14:59.634044 : Group C2._.Rep1 (3 of 30) : Creating Group Coverage File : C2._.Rep1.insertions.coverage.h5, 2.22 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:16:03.943321 : Group C2._.Rep2 (4 of 30) : Creating Group Coverage File : C2._.Rep2.insertions.coverage.h5, 3.292 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:17:02.918898 : Group C3._.Rep1 (5 of 30) : Creating Group Coverage File : C3._.Rep1.insertions.coverage.h5, 4.274 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:18:06.658447 : Group C3._.Rep2 (6 of 30) : Creating Group Coverage File : C3._.Rep2.insertions.coverage.h5, 5.337 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:19:06.869088 : Group C4._.Rep1 (7 of 30) : Creating Group Coverage File : C4._.Rep1.insertions.coverage.h5, 6.34 mins elapsed.\n",
      "\n",
      "Number of Cells = 197\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:20:13.237421 : Group C4._.Rep2 (8 of 30) : Creating Group Coverage File : C4._.Rep2.insertions.coverage.h5, 7.446 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:21:14.029672 : Group C5._.Rep1 (9 of 30) : Creating Group Coverage File : C5._.Rep1.insertions.coverage.h5, 8.46 mins elapsed.\n",
      "\n",
      "Number of Cells = 294\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:22:18.838652 : Group C5._.Rep2 (10 of 30) : Creating Group Coverage File : C5._.Rep2.insertions.coverage.h5, 9.54 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:23:19.080318 : Group C6._.Rep1 (11 of 30) : Creating Group Coverage File : C6._.Rep1.insertions.coverage.h5, 10.544 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:24:26.647379 : Group C6._.Rep2 (12 of 30) : Creating Group Coverage File : C6._.Rep2.insertions.coverage.h5, 11.67 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:25:25.878045 : Group C7._.Rep1 (13 of 30) : Creating Group Coverage File : C7._.Rep1.insertions.coverage.h5, 12.657 mins elapsed.\n",
      "\n",
      "Number of Cells = 75\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:26:26.662375 : Group C7._.Rep2 (14 of 30) : Creating Group Coverage File : C7._.Rep2.insertions.coverage.h5, 13.67 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:27:26.497064 : Group C8._.Rep1 (15 of 30) : Creating Group Coverage File : C8._.Rep1.insertions.coverage.h5, 14.667 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:28:30.695753 : Group C8._.Rep2 (16 of 30) : Creating Group Coverage File : C8._.Rep2.insertions.coverage.h5, 15.737 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:29:30.113818 : Group C9._.Rep1 (17 of 30) : Creating Group Coverage File : C9._.Rep1.insertions.coverage.h5, 16.728 mins elapsed.\n",
      "\n",
      "Number of Cells = 137\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:30:33.7815 : Group C9._.Rep2 (18 of 30) : Creating Group Coverage File : C9._.Rep2.insertions.coverage.h5, 17.789 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:31:33.21401 : Group C10._.Rep1 (19 of 30) : Creating Group Coverage File : C10._.Rep1.insertions.coverage.h5, 18.779 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:32:44.729019 : Group C10._.Rep2 (20 of 30) : Creating Group Coverage File : C10._.Rep2.insertions.coverage.h5, 19.971 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:33:44.101229 : Group C11._.Rep1 (21 of 30) : Creating Group Coverage File : C11._.Rep1.insertions.coverage.h5, 20.961 mins elapsed.\n",
      "\n",
      "Number of Cells = 226\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:34:48.790096 : Group C11._.Rep2 (22 of 30) : Creating Group Coverage File : C11._.Rep2.insertions.coverage.h5, 22.039 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:35:48.866484 : Group C12._.Rep1 (23 of 30) : Creating Group Coverage File : C12._.Rep1.insertions.coverage.h5, 23.04 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:36:59.499741 : Group C12._.Rep2 (24 of 30) : Creating Group Coverage File : C12._.Rep2.insertions.coverage.h5, 24.217 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:38:01.700874 : Group C13._.Rep1 (25 of 30) : Creating Group Coverage File : C13._.Rep1.insertions.coverage.h5, 25.254 mins elapsed.\n",
      "\n",
      "Number of Cells = 500\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:39:14.203743 : Group C13._.Rep2 (26 of 30) : Creating Group Coverage File : C13._.Rep2.insertions.coverage.h5, 26.463 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:40:14.973959 : Group C14._.Rep1 (27 of 30) : Creating Group Coverage File : C14._.Rep1.insertions.coverage.h5, 27.475 mins elapsed.\n",
      "\n",
      "Number of Cells = 351\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:41:22.843592 : Group C14._.Rep2 (28 of 30) : Creating Group Coverage File : C14._.Rep2.insertions.coverage.h5, 28.607 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:42:23.087579 : Group C15._.Rep1 (29 of 30) : Creating Group Coverage File : C15._.Rep1.insertions.coverage.h5, 29.611 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:43:22.76512 : Group C15._.Rep2 (30 of 30) : Creating Group Coverage File : C15._.Rep2.insertions.coverage.h5, 30.605 mins elapsed.\n",
      "\n",
      "Number of Cells = 40\n",
      "\n",
      "Coverage File Exists!\n",
      "\n",
      "Added Coverage Group\n",
      "\n",
      "Added Metadata Group\n",
      "\n",
      "Added ArrowCoverage Class\n",
      "\n",
      "Added Coverage/Info\n",
      "\n",
      "Added Coverage/Info/CellNames\n",
      "\n",
      "2024-05-27 19:44:22.547861 : Adding Kmer Bias to Coverage Files!, 31.602 mins elapsed.\n",
      "\n",
      "Kmer Bias chr1 (1 of 24)\n",
      "\n",
      "chr1 \n",
      "Coverage File chr1 (1 of 30)\n",
      "\n",
      "Coverage File chr1 (2 of 30)\n",
      "\n",
      "Coverage File chr1 (3 of 30)\n",
      "\n",
      "Coverage File chr1 (4 of 30)\n",
      "\n",
      "Coverage File chr1 (5 of 30)\n",
      "\n",
      "Coverage File chr1 (6 of 30)\n",
      "\n",
      "Coverage File chr1 (7 of 30)\n",
      "\n",
      "Coverage File chr1 (8 of 30)\n",
      "\n",
      "Coverage File chr1 (9 of 30)\n",
      "\n",
      "Coverage File chr1 (10 of 30)\n",
      "\n",
      "Coverage File chr1 (11 of 30)\n",
      "\n",
      "Coverage File chr1 (12 of 30)\n",
      "\n",
      "Coverage File chr1 (13 of 30)\n",
      "\n",
      "Coverage File chr1 (14 of 30)\n",
      "\n",
      "Coverage File chr1 (15 of 30)\n",
      "\n",
      "Coverage File chr1 (16 of 30)\n",
      "\n",
      "Coverage File chr1 (17 of 30)\n",
      "\n",
      "Coverage File chr1 (18 of 30)\n",
      "\n",
      "Coverage File chr1 (19 of 30)\n",
      "\n",
      "Coverage File chr1 (20 of 30)\n",
      "\n",
      "Coverage File chr1 (21 of 30)\n",
      "\n",
      "Coverage File chr1 (22 of 30)\n",
      "\n",
      "Coverage File chr1 (23 of 30)\n",
      "\n",
      "Coverage File chr1 (24 of 30)\n",
      "\n",
      "Coverage File chr1 (25 of 30)\n",
      "\n",
      "Coverage File chr1 (26 of 30)\n",
      "\n",
      "Coverage File chr1 (27 of 30)\n",
      "\n",
      "Coverage File chr1 (28 of 30)\n",
      "\n",
      "Coverage File chr1 (29 of 30)\n",
      "\n",
      "Coverage File chr1 (30 of 30)\n",
      "\n",
      "Kmer Bias chr10 (2 of 24)\n",
      "\n",
      "chr10 \n",
      "Coverage File chr10 (1 of 30)\n",
      "\n",
      "Coverage File chr10 (2 of 30)\n",
      "\n",
      "Coverage File chr10 (3 of 30)\n",
      "\n",
      "Coverage File chr10 (4 of 30)\n",
      "\n",
      "Coverage File chr10 (5 of 30)\n",
      "\n",
      "Coverage File chr10 (6 of 30)\n",
      "\n",
      "Coverage File chr10 (7 of 30)\n",
      "\n",
      "Coverage File chr10 (8 of 30)\n",
      "\n",
      "Coverage File chr10 (9 of 30)\n",
      "\n",
      "Coverage File chr10 (10 of 30)\n",
      "\n",
      "Coverage File chr10 (11 of 30)\n",
      "\n",
      "Coverage File chr10 (12 of 30)\n",
      "\n",
      "Coverage File chr10 (13 of 30)\n",
      "\n",
      "Coverage File chr10 (14 of 30)\n",
      "\n",
      "Coverage File chr10 (15 of 30)\n",
      "\n",
      "Coverage File chr10 (16 of 30)\n",
      "\n",
      "Coverage File chr10 (17 of 30)\n",
      "\n",
      "Coverage File chr10 (18 of 30)\n",
      "\n",
      "Coverage File chr10 (19 of 30)\n",
      "\n",
      "Coverage File chr10 (20 of 30)\n",
      "\n",
      "Coverage File chr10 (21 of 30)\n",
      "\n",
      "Coverage File chr10 (22 of 30)\n",
      "\n",
      "Coverage File chr10 (23 of 30)\n",
      "\n",
      "Coverage File chr10 (24 of 30)\n",
      "\n",
      "Coverage File chr10 (25 of 30)\n",
      "\n",
      "Coverage File chr10 (26 of 30)\n",
      "\n",
      "Coverage File chr10 (27 of 30)\n",
      "\n",
      "Coverage File chr10 (28 of 30)\n",
      "\n",
      "Coverage File chr10 (29 of 30)\n",
      "\n",
      "Coverage File chr10 (30 of 30)\n",
      "\n",
      "Kmer Bias chr11 (3 of 24)\n",
      "\n",
      "chr11 \n",
      "Coverage File chr11 (1 of 30)\n",
      "\n",
      "Coverage File chr11 (2 of 30)\n",
      "\n",
      "Coverage File chr11 (3 of 30)\n",
      "\n",
      "Coverage File chr11 (4 of 30)\n",
      "\n",
      "Coverage File chr11 (5 of 30)\n",
      "\n",
      "Coverage File chr11 (6 of 30)\n",
      "\n",
      "Coverage File chr11 (7 of 30)\n",
      "\n",
      "Coverage File chr11 (8 of 30)\n",
      "\n",
      "Coverage File chr11 (9 of 30)\n",
      "\n",
      "Coverage File chr11 (10 of 30)\n",
      "\n",
      "Coverage File chr11 (11 of 30)\n",
      "\n",
      "Coverage File chr11 (12 of 30)\n",
      "\n",
      "Coverage File chr11 (13 of 30)\n",
      "\n",
      "Coverage File chr11 (14 of 30)\n",
      "\n",
      "Coverage File chr11 (15 of 30)\n",
      "\n",
      "Coverage File chr11 (16 of 30)\n",
      "\n",
      "Coverage File chr11 (17 of 30)\n",
      "\n",
      "Coverage File chr11 (18 of 30)\n",
      "\n",
      "Coverage File chr11 (19 of 30)\n",
      "\n",
      "Coverage File chr11 (20 of 30)\n",
      "\n",
      "Coverage File chr11 (21 of 30)\n",
      "\n",
      "Coverage File chr11 (22 of 30)\n",
      "\n",
      "Coverage File chr11 (23 of 30)\n",
      "\n",
      "Coverage File chr11 (24 of 30)\n",
      "\n",
      "Coverage File chr11 (25 of 30)\n",
      "\n",
      "Coverage File chr11 (26 of 30)\n",
      "\n",
      "Coverage File chr11 (27 of 30)\n",
      "\n",
      "Coverage File chr11 (28 of 30)\n",
      "\n",
      "Coverage File chr11 (29 of 30)\n",
      "\n",
      "Coverage File chr11 (30 of 30)\n",
      "\n",
      "Kmer Bias chr12 (4 of 24)\n",
      "\n",
      "chr12 \n",
      "Coverage File chr12 (1 of 30)\n",
      "\n",
      "Coverage File chr12 (2 of 30)\n",
      "\n",
      "Coverage File chr12 (3 of 30)\n",
      "\n",
      "Coverage File chr12 (4 of 30)\n",
      "\n",
      "Coverage File chr12 (5 of 30)\n",
      "\n",
      "Coverage File chr12 (6 of 30)\n",
      "\n",
      "Coverage File chr12 (7 of 30)\n",
      "\n",
      "Coverage File chr12 (8 of 30)\n",
      "\n",
      "Coverage File chr12 (9 of 30)\n",
      "\n",
      "Coverage File chr12 (10 of 30)\n",
      "\n",
      "Coverage File chr12 (11 of 30)\n",
      "\n",
      "Coverage File chr12 (12 of 30)\n",
      "\n",
      "Coverage File chr12 (13 of 30)\n",
      "\n",
      "Coverage File chr12 (14 of 30)\n",
      "\n",
      "Coverage File chr12 (15 of 30)\n",
      "\n",
      "Coverage File chr12 (16 of 30)\n",
      "\n",
      "Coverage File chr12 (17 of 30)\n",
      "\n",
      "Coverage File chr12 (18 of 30)\n",
      "\n",
      "Coverage File chr12 (19 of 30)\n",
      "\n",
      "Coverage File chr12 (20 of 30)\n",
      "\n",
      "Coverage File chr12 (21 of 30)\n",
      "\n",
      "Coverage File chr12 (22 of 30)\n",
      "\n",
      "Coverage File chr12 (23 of 30)\n",
      "\n",
      "Coverage File chr12 (24 of 30)\n",
      "\n",
      "Coverage File chr12 (25 of 30)\n",
      "\n",
      "Coverage File chr12 (26 of 30)\n",
      "\n",
      "Coverage File chr12 (27 of 30)\n",
      "\n",
      "Coverage File chr12 (28 of 30)\n",
      "\n",
      "Coverage File chr12 (29 of 30)\n",
      "\n",
      "Coverage File chr12 (30 of 30)\n",
      "\n",
      "Kmer Bias chr13 (5 of 24)\n",
      "\n",
      "chr13 \n",
      "Coverage File chr13 (1 of 30)\n",
      "\n",
      "Coverage File chr13 (2 of 30)\n",
      "\n",
      "Coverage File chr13 (3 of 30)\n",
      "\n",
      "Coverage File chr13 (4 of 30)\n",
      "\n",
      "Coverage File chr13 (5 of 30)\n",
      "\n",
      "Coverage File chr13 (6 of 30)\n",
      "\n",
      "Coverage File chr13 (7 of 30)\n",
      "\n",
      "Coverage File chr13 (8 of 30)\n",
      "\n",
      "Coverage File chr13 (9 of 30)\n",
      "\n",
      "Coverage File chr13 (10 of 30)\n",
      "\n",
      "Coverage File chr13 (11 of 30)\n",
      "\n",
      "Coverage File chr13 (12 of 30)\n",
      "\n",
      "Coverage File chr13 (13 of 30)\n",
      "\n",
      "Coverage File chr13 (14 of 30)\n",
      "\n",
      "Coverage File chr13 (15 of 30)\n",
      "\n",
      "Coverage File chr13 (16 of 30)\n",
      "\n",
      "Coverage File chr13 (17 of 30)\n",
      "\n",
      "Coverage File chr13 (18 of 30)\n",
      "\n",
      "Coverage File chr13 (19 of 30)\n",
      "\n",
      "Coverage File chr13 (20 of 30)\n",
      "\n",
      "Coverage File chr13 (21 of 30)\n",
      "\n",
      "Coverage File chr13 (22 of 30)\n",
      "\n",
      "Coverage File chr13 (23 of 30)\n",
      "\n",
      "Coverage File chr13 (24 of 30)\n",
      "\n",
      "Coverage File chr13 (25 of 30)\n",
      "\n",
      "Coverage File chr13 (26 of 30)\n",
      "\n",
      "Coverage File chr13 (27 of 30)\n",
      "\n",
      "Coverage File chr13 (28 of 30)\n",
      "\n",
      "Coverage File chr13 (29 of 30)\n",
      "\n",
      "Coverage File chr13 (30 of 30)\n",
      "\n",
      "Kmer Bias chr14 (6 of 24)\n",
      "\n",
      "chr14 \n",
      "Coverage File chr14 (1 of 30)\n",
      "\n",
      "Coverage File chr14 (2 of 30)\n",
      "\n",
      "Coverage File chr14 (3 of 30)\n",
      "\n",
      "Coverage File chr14 (4 of 30)\n",
      "\n",
      "Coverage File chr14 (5 of 30)\n",
      "\n",
      "Coverage File chr14 (6 of 30)\n",
      "\n",
      "Coverage File chr14 (7 of 30)\n",
      "\n",
      "Coverage File chr14 (8 of 30)\n",
      "\n",
      "Coverage File chr14 (9 of 30)\n",
      "\n",
      "Coverage File chr14 (10 of 30)\n",
      "\n",
      "Coverage File chr14 (11 of 30)\n",
      "\n",
      "Coverage File chr14 (12 of 30)\n",
      "\n",
      "Coverage File chr14 (13 of 30)\n",
      "\n",
      "Coverage File chr14 (14 of 30)\n",
      "\n",
      "Coverage File chr14 (15 of 30)\n",
      "\n",
      "Coverage File chr14 (16 of 30)\n",
      "\n",
      "Coverage File chr14 (17 of 30)\n",
      "\n",
      "Coverage File chr14 (18 of 30)\n",
      "\n",
      "Coverage File chr14 (19 of 30)\n",
      "\n",
      "Coverage File chr14 (20 of 30)\n",
      "\n",
      "Coverage File chr14 (21 of 30)\n",
      "\n",
      "Coverage File chr14 (22 of 30)\n",
      "\n",
      "Coverage File chr14 (23 of 30)\n",
      "\n",
      "Coverage File chr14 (24 of 30)\n",
      "\n",
      "Coverage File chr14 (25 of 30)\n",
      "\n",
      "Coverage File chr14 (26 of 30)\n",
      "\n",
      "Coverage File chr14 (27 of 30)\n",
      "\n",
      "Coverage File chr14 (28 of 30)\n",
      "\n",
      "Coverage File chr14 (29 of 30)\n",
      "\n",
      "Coverage File chr14 (30 of 30)\n",
      "\n",
      "Kmer Bias chr15 (7 of 24)\n",
      "\n",
      "chr15 \n",
      "Coverage File chr15 (1 of 30)\n",
      "\n",
      "Coverage File chr15 (2 of 30)\n",
      "\n",
      "Coverage File chr15 (3 of 30)\n",
      "\n",
      "Coverage File chr15 (4 of 30)\n",
      "\n",
      "Coverage File chr15 (5 of 30)\n",
      "\n",
      "Coverage File chr15 (6 of 30)\n",
      "\n",
      "Coverage File chr15 (7 of 30)\n",
      "\n",
      "Coverage File chr15 (8 of 30)\n",
      "\n",
      "Coverage File chr15 (9 of 30)\n",
      "\n",
      "Coverage File chr15 (10 of 30)\n",
      "\n",
      "Coverage File chr15 (11 of 30)\n",
      "\n",
      "Coverage File chr15 (12 of 30)\n",
      "\n",
      "Coverage File chr15 (13 of 30)\n",
      "\n",
      "Coverage File chr15 (14 of 30)\n",
      "\n",
      "Coverage File chr15 (15 of 30)\n",
      "\n",
      "Coverage File chr15 (16 of 30)\n",
      "\n",
      "Coverage File chr15 (17 of 30)\n",
      "\n",
      "Coverage File chr15 (18 of 30)\n",
      "\n",
      "Coverage File chr15 (19 of 30)\n",
      "\n",
      "Coverage File chr15 (20 of 30)\n",
      "\n",
      "Coverage File chr15 (21 of 30)\n",
      "\n",
      "Coverage File chr15 (22 of 30)\n",
      "\n",
      "Coverage File chr15 (23 of 30)\n",
      "\n",
      "Coverage File chr15 (24 of 30)\n",
      "\n",
      "Coverage File chr15 (25 of 30)\n",
      "\n",
      "Coverage File chr15 (26 of 30)\n",
      "\n",
      "Coverage File chr15 (27 of 30)\n",
      "\n",
      "Coverage File chr15 (28 of 30)\n",
      "\n",
      "Coverage File chr15 (29 of 30)\n",
      "\n",
      "Coverage File chr15 (30 of 30)\n",
      "\n",
      "Kmer Bias chr16 (8 of 24)\n",
      "\n",
      "chr16 \n",
      "Coverage File chr16 (1 of 30)\n",
      "\n",
      "Coverage File chr16 (2 of 30)\n",
      "\n",
      "Coverage File chr16 (3 of 30)\n",
      "\n",
      "Coverage File chr16 (4 of 30)\n",
      "\n",
      "Coverage File chr16 (5 of 30)\n",
      "\n",
      "Coverage File chr16 (6 of 30)\n",
      "\n",
      "Coverage File chr16 (7 of 30)\n",
      "\n",
      "Coverage File chr16 (8 of 30)\n",
      "\n",
      "Coverage File chr16 (9 of 30)\n",
      "\n",
      "Coverage File chr16 (10 of 30)\n",
      "\n",
      "Coverage File chr16 (11 of 30)\n",
      "\n",
      "Coverage File chr16 (12 of 30)\n",
      "\n",
      "Coverage File chr16 (13 of 30)\n",
      "\n",
      "Coverage File chr16 (14 of 30)\n",
      "\n",
      "Coverage File chr16 (15 of 30)\n",
      "\n",
      "Coverage File chr16 (16 of 30)\n",
      "\n",
      "Coverage File chr16 (17 of 30)\n",
      "\n",
      "Coverage File chr16 (18 of 30)\n",
      "\n",
      "Coverage File chr16 (19 of 30)\n",
      "\n",
      "Coverage File chr16 (20 of 30)\n",
      "\n",
      "Coverage File chr16 (21 of 30)\n",
      "\n",
      "Coverage File chr16 (22 of 30)\n",
      "\n",
      "Coverage File chr16 (23 of 30)\n",
      "\n",
      "Coverage File chr16 (24 of 30)\n",
      "\n",
      "Coverage File chr16 (25 of 30)\n",
      "\n",
      "Coverage File chr16 (26 of 30)\n",
      "\n",
      "Coverage File chr16 (27 of 30)\n",
      "\n",
      "Coverage File chr16 (28 of 30)\n",
      "\n",
      "Coverage File chr16 (29 of 30)\n",
      "\n",
      "Coverage File chr16 (30 of 30)\n",
      "\n",
      "Kmer Bias chr17 (9 of 24)\n",
      "\n",
      "chr17 \n",
      "Coverage File chr17 (1 of 30)\n",
      "\n",
      "Coverage File chr17 (2 of 30)\n",
      "\n",
      "Coverage File chr17 (3 of 30)\n",
      "\n",
      "Coverage File chr17 (4 of 30)\n",
      "\n",
      "Coverage File chr17 (5 of 30)\n",
      "\n",
      "Coverage File chr17 (6 of 30)\n",
      "\n",
      "Coverage File chr17 (7 of 30)\n",
      "\n",
      "Coverage File chr17 (8 of 30)\n",
      "\n",
      "Coverage File chr17 (9 of 30)\n",
      "\n",
      "Coverage File chr17 (10 of 30)\n",
      "\n",
      "Coverage File chr17 (11 of 30)\n",
      "\n",
      "Coverage File chr17 (12 of 30)\n",
      "\n",
      "Coverage File chr17 (13 of 30)\n",
      "\n",
      "Coverage File chr17 (14 of 30)\n",
      "\n",
      "Coverage File chr17 (15 of 30)\n",
      "\n",
      "Coverage File chr17 (16 of 30)\n",
      "\n",
      "Coverage File chr17 (17 of 30)\n",
      "\n",
      "Coverage File chr17 (18 of 30)\n",
      "\n",
      "Coverage File chr17 (19 of 30)\n",
      "\n",
      "Coverage File chr17 (20 of 30)\n",
      "\n",
      "Coverage File chr17 (21 of 30)\n",
      "\n",
      "Coverage File chr17 (22 of 30)\n",
      "\n",
      "Coverage File chr17 (23 of 30)\n",
      "\n",
      "Coverage File chr17 (24 of 30)\n",
      "\n",
      "Coverage File chr17 (25 of 30)\n",
      "\n",
      "Coverage File chr17 (26 of 30)\n",
      "\n",
      "Coverage File chr17 (27 of 30)\n",
      "\n",
      "Coverage File chr17 (28 of 30)\n",
      "\n",
      "Coverage File chr17 (29 of 30)\n",
      "\n",
      "Coverage File chr17 (30 of 30)\n",
      "\n",
      "Kmer Bias chr18 (10 of 24)\n",
      "\n",
      "chr18 \n",
      "Coverage File chr18 (1 of 30)\n",
      "\n",
      "Coverage File chr18 (2 of 30)\n",
      "\n",
      "Coverage File chr18 (3 of 30)\n",
      "\n",
      "Coverage File chr18 (4 of 30)\n",
      "\n",
      "Coverage File chr18 (5 of 30)\n",
      "\n",
      "Coverage File chr18 (6 of 30)\n",
      "\n",
      "Coverage File chr18 (7 of 30)\n",
      "\n",
      "Coverage File chr18 (8 of 30)\n",
      "\n",
      "Coverage File chr18 (9 of 30)\n",
      "\n",
      "Coverage File chr18 (10 of 30)\n",
      "\n",
      "Coverage File chr18 (11 of 30)\n",
      "\n",
      "Coverage File chr18 (12 of 30)\n",
      "\n",
      "Coverage File chr18 (13 of 30)\n",
      "\n",
      "Coverage File chr18 (14 of 30)\n",
      "\n",
      "Coverage File chr18 (15 of 30)\n",
      "\n",
      "Coverage File chr18 (16 of 30)\n",
      "\n",
      "Coverage File chr18 (17 of 30)\n",
      "\n",
      "Coverage File chr18 (18 of 30)\n",
      "\n",
      "Coverage File chr18 (19 of 30)\n",
      "\n",
      "Coverage File chr18 (20 of 30)\n",
      "\n",
      "Coverage File chr18 (21 of 30)\n",
      "\n",
      "Coverage File chr18 (22 of 30)\n",
      "\n",
      "Coverage File chr18 (23 of 30)\n",
      "\n",
      "Coverage File chr18 (24 of 30)\n",
      "\n",
      "Coverage File chr18 (25 of 30)\n",
      "\n",
      "Coverage File chr18 (26 of 30)\n",
      "\n",
      "Coverage File chr18 (27 of 30)\n",
      "\n",
      "Coverage File chr18 (28 of 30)\n",
      "\n",
      "Coverage File chr18 (29 of 30)\n",
      "\n",
      "Coverage File chr18 (30 of 30)\n",
      "\n",
      "Kmer Bias chr19 (11 of 24)\n",
      "\n",
      "chr19 \n",
      "Coverage File chr19 (1 of 30)\n",
      "\n",
      "Coverage File chr19 (2 of 30)\n",
      "\n",
      "Coverage File chr19 (3 of 30)\n",
      "\n",
      "Coverage File chr19 (4 of 30)\n",
      "\n",
      "Coverage File chr19 (5 of 30)\n",
      "\n",
      "Coverage File chr19 (6 of 30)\n",
      "\n",
      "Coverage File chr19 (7 of 30)\n",
      "\n",
      "Coverage File chr19 (8 of 30)\n",
      "\n",
      "Coverage File chr19 (9 of 30)\n",
      "\n",
      "Coverage File chr19 (10 of 30)\n",
      "\n",
      "Coverage File chr19 (11 of 30)\n",
      "\n",
      "Coverage File chr19 (12 of 30)\n",
      "\n",
      "Coverage File chr19 (13 of 30)\n",
      "\n",
      "Coverage File chr19 (14 of 30)\n",
      "\n",
      "Coverage File chr19 (15 of 30)\n",
      "\n",
      "Coverage File chr19 (16 of 30)\n",
      "\n",
      "Coverage File chr19 (17 of 30)\n",
      "\n",
      "Coverage File chr19 (18 of 30)\n",
      "\n",
      "Coverage File chr19 (19 of 30)\n",
      "\n",
      "Coverage File chr19 (20 of 30)\n",
      "\n",
      "Coverage File chr19 (21 of 30)\n",
      "\n",
      "Coverage File chr19 (22 of 30)\n",
      "\n",
      "Coverage File chr19 (23 of 30)\n",
      "\n",
      "Coverage File chr19 (24 of 30)\n",
      "\n",
      "Coverage File chr19 (25 of 30)\n",
      "\n",
      "Coverage File chr19 (26 of 30)\n",
      "\n",
      "Coverage File chr19 (27 of 30)\n",
      "\n",
      "Coverage File chr19 (28 of 30)\n",
      "\n",
      "Coverage File chr19 (29 of 30)\n",
      "\n",
      "Coverage File chr19 (30 of 30)\n",
      "\n",
      "Kmer Bias chr2 (12 of 24)\n",
      "\n",
      "chr2 \n",
      "Coverage File chr2 (1 of 30)\n",
      "\n",
      "Coverage File chr2 (2 of 30)\n",
      "\n",
      "Coverage File chr2 (3 of 30)\n",
      "\n",
      "Coverage File chr2 (4 of 30)\n",
      "\n",
      "Coverage File chr2 (5 of 30)\n",
      "\n",
      "Coverage File chr2 (6 of 30)\n",
      "\n",
      "Coverage File chr2 (7 of 30)\n",
      "\n",
      "Coverage File chr2 (8 of 30)\n",
      "\n",
      "Coverage File chr2 (9 of 30)\n",
      "\n",
      "Coverage File chr2 (10 of 30)\n",
      "\n",
      "Coverage File chr2 (11 of 30)\n",
      "\n",
      "Coverage File chr2 (12 of 30)\n",
      "\n",
      "Coverage File chr2 (13 of 30)\n",
      "\n",
      "Coverage File chr2 (14 of 30)\n",
      "\n",
      "Coverage File chr2 (15 of 30)\n",
      "\n",
      "Coverage File chr2 (16 of 30)\n",
      "\n",
      "Coverage File chr2 (17 of 30)\n",
      "\n",
      "Coverage File chr2 (18 of 30)\n",
      "\n",
      "Coverage File chr2 (19 of 30)\n",
      "\n",
      "Coverage File chr2 (20 of 30)\n",
      "\n",
      "Coverage File chr2 (21 of 30)\n",
      "\n",
      "Coverage File chr2 (22 of 30)\n",
      "\n",
      "Coverage File chr2 (23 of 30)\n",
      "\n",
      "Coverage File chr2 (24 of 30)\n",
      "\n",
      "Coverage File chr2 (25 of 30)\n",
      "\n",
      "Coverage File chr2 (26 of 30)\n",
      "\n",
      "Coverage File chr2 (27 of 30)\n",
      "\n",
      "Coverage File chr2 (28 of 30)\n",
      "\n",
      "Coverage File chr2 (29 of 30)\n",
      "\n",
      "Coverage File chr2 (30 of 30)\n",
      "\n",
      "Kmer Bias chr20 (13 of 24)\n",
      "\n",
      "chr20 \n",
      "Coverage File chr20 (1 of 30)\n",
      "\n",
      "Coverage File chr20 (2 of 30)\n",
      "\n",
      "Coverage File chr20 (3 of 30)\n",
      "\n",
      "Coverage File chr20 (4 of 30)\n",
      "\n",
      "Coverage File chr20 (5 of 30)\n",
      "\n",
      "Coverage File chr20 (6 of 30)\n",
      "\n",
      "Coverage File chr20 (7 of 30)\n",
      "\n",
      "Coverage File chr20 (8 of 30)\n",
      "\n",
      "Coverage File chr20 (9 of 30)\n",
      "\n",
      "Coverage File chr20 (10 of 30)\n",
      "\n",
      "Coverage File chr20 (11 of 30)\n",
      "\n",
      "Coverage File chr20 (12 of 30)\n",
      "\n",
      "Coverage File chr20 (13 of 30)\n",
      "\n",
      "Coverage File chr20 (14 of 30)\n",
      "\n",
      "Coverage File chr20 (15 of 30)\n",
      "\n",
      "Coverage File chr20 (16 of 30)\n",
      "\n",
      "Coverage File chr20 (17 of 30)\n",
      "\n",
      "Coverage File chr20 (18 of 30)\n",
      "\n",
      "Coverage File chr20 (19 of 30)\n",
      "\n",
      "Coverage File chr20 (20 of 30)\n",
      "\n",
      "Coverage File chr20 (21 of 30)\n",
      "\n",
      "Coverage File chr20 (22 of 30)\n",
      "\n",
      "Coverage File chr20 (23 of 30)\n",
      "\n",
      "Coverage File chr20 (24 of 30)\n",
      "\n",
      "Coverage File chr20 (25 of 30)\n",
      "\n",
      "Coverage File chr20 (26 of 30)\n",
      "\n",
      "Coverage File chr20 (27 of 30)\n",
      "\n",
      "Coverage File chr20 (28 of 30)\n",
      "\n",
      "Coverage File chr20 (29 of 30)\n",
      "\n",
      "Coverage File chr20 (30 of 30)\n",
      "\n",
      "Kmer Bias chr21 (14 of 24)\n",
      "\n",
      "chr21 \n",
      "Coverage File chr21 (1 of 30)\n",
      "\n",
      "Coverage File chr21 (2 of 30)\n",
      "\n",
      "Coverage File chr21 (3 of 30)\n",
      "\n",
      "Coverage File chr21 (4 of 30)\n",
      "\n",
      "Coverage File chr21 (5 of 30)\n",
      "\n",
      "Coverage File chr21 (6 of 30)\n",
      "\n",
      "Coverage File chr21 (7 of 30)\n",
      "\n",
      "Coverage File chr21 (8 of 30)\n",
      "\n",
      "Coverage File chr21 (9 of 30)\n",
      "\n",
      "Coverage File chr21 (10 of 30)\n",
      "\n",
      "Coverage File chr21 (11 of 30)\n",
      "\n",
      "Coverage File chr21 (12 of 30)\n",
      "\n",
      "Coverage File chr21 (13 of 30)\n",
      "\n",
      "Coverage File chr21 (14 of 30)\n",
      "\n",
      "Coverage File chr21 (15 of 30)\n",
      "\n",
      "Coverage File chr21 (16 of 30)\n",
      "\n",
      "Coverage File chr21 (17 of 30)\n",
      "\n",
      "Coverage File chr21 (18 of 30)\n",
      "\n",
      "Coverage File chr21 (19 of 30)\n",
      "\n",
      "Coverage File chr21 (20 of 30)\n",
      "\n",
      "Coverage File chr21 (21 of 30)\n",
      "\n",
      "Coverage File chr21 (22 of 30)\n",
      "\n",
      "Coverage File chr21 (23 of 30)\n",
      "\n",
      "Coverage File chr21 (24 of 30)\n",
      "\n",
      "Coverage File chr21 (25 of 30)\n",
      "\n",
      "Coverage File chr21 (26 of 30)\n",
      "\n",
      "Coverage File chr21 (27 of 30)\n",
      "\n",
      "Coverage File chr21 (28 of 30)\n",
      "\n",
      "Coverage File chr21 (29 of 30)\n",
      "\n",
      "Coverage File chr21 (30 of 30)\n",
      "\n",
      "Kmer Bias chr22 (15 of 24)\n",
      "\n",
      "chr22 \n",
      "Coverage File chr22 (1 of 30)\n",
      "\n",
      "Coverage File chr22 (2 of 30)\n",
      "\n",
      "Coverage File chr22 (3 of 30)\n",
      "\n",
      "Coverage File chr22 (4 of 30)\n",
      "\n",
      "Coverage File chr22 (5 of 30)\n",
      "\n",
      "Coverage File chr22 (6 of 30)\n",
      "\n",
      "Coverage File chr22 (7 of 30)\n",
      "\n",
      "Coverage File chr22 (8 of 30)\n",
      "\n",
      "Coverage File chr22 (9 of 30)\n",
      "\n",
      "Coverage File chr22 (10 of 30)\n",
      "\n",
      "Coverage File chr22 (11 of 30)\n",
      "\n",
      "Coverage File chr22 (12 of 30)\n",
      "\n",
      "Coverage File chr22 (13 of 30)\n",
      "\n",
      "Coverage File chr22 (14 of 30)\n",
      "\n",
      "Coverage File chr22 (15 of 30)\n",
      "\n",
      "Coverage File chr22 (16 of 30)\n",
      "\n",
      "Coverage File chr22 (17 of 30)\n",
      "\n",
      "Coverage File chr22 (18 of 30)\n",
      "\n",
      "Coverage File chr22 (19 of 30)\n",
      "\n",
      "Coverage File chr22 (20 of 30)\n",
      "\n",
      "Coverage File chr22 (21 of 30)\n",
      "\n",
      "Coverage File chr22 (22 of 30)\n",
      "\n",
      "Coverage File chr22 (23 of 30)\n",
      "\n",
      "Coverage File chr22 (24 of 30)\n",
      "\n",
      "Coverage File chr22 (25 of 30)\n",
      "\n",
      "Coverage File chr22 (26 of 30)\n",
      "\n",
      "Coverage File chr22 (27 of 30)\n",
      "\n",
      "Coverage File chr22 (28 of 30)\n",
      "\n",
      "Coverage File chr22 (29 of 30)\n",
      "\n",
      "Coverage File chr22 (30 of 30)\n",
      "\n",
      "Kmer Bias chr3 (16 of 24)\n",
      "\n",
      "chr3 \n",
      "Coverage File chr3 (1 of 30)\n",
      "\n",
      "Coverage File chr3 (2 of 30)\n",
      "\n",
      "Coverage File chr3 (3 of 30)\n",
      "\n",
      "Coverage File chr3 (4 of 30)\n",
      "\n",
      "Coverage File chr3 (5 of 30)\n",
      "\n",
      "Coverage File chr3 (6 of 30)\n",
      "\n",
      "Coverage File chr3 (7 of 30)\n",
      "\n",
      "Coverage File chr3 (8 of 30)\n",
      "\n",
      "Coverage File chr3 (9 of 30)\n",
      "\n",
      "Coverage File chr3 (10 of 30)\n",
      "\n",
      "Coverage File chr3 (11 of 30)\n",
      "\n",
      "Coverage File chr3 (12 of 30)\n",
      "\n",
      "Coverage File chr3 (13 of 30)\n",
      "\n",
      "Coverage File chr3 (14 of 30)\n",
      "\n",
      "Coverage File chr3 (15 of 30)\n",
      "\n",
      "Coverage File chr3 (16 of 30)\n",
      "\n",
      "Coverage File chr3 (17 of 30)\n",
      "\n",
      "Coverage File chr3 (18 of 30)\n",
      "\n",
      "Coverage File chr3 (19 of 30)\n",
      "\n",
      "Coverage File chr3 (20 of 30)\n",
      "\n",
      "Coverage File chr3 (21 of 30)\n",
      "\n",
      "Coverage File chr3 (22 of 30)\n",
      "\n",
      "Coverage File chr3 (23 of 30)\n",
      "\n",
      "Coverage File chr3 (24 of 30)\n",
      "\n",
      "Coverage File chr3 (25 of 30)\n",
      "\n",
      "Coverage File chr3 (26 of 30)\n",
      "\n",
      "Coverage File chr3 (27 of 30)\n",
      "\n",
      "Coverage File chr3 (28 of 30)\n",
      "\n",
      "Coverage File chr3 (29 of 30)\n",
      "\n",
      "Coverage File chr3 (30 of 30)\n",
      "\n",
      "Kmer Bias chr4 (17 of 24)\n",
      "\n",
      "chr4 \n",
      "Coverage File chr4 (1 of 30)\n",
      "\n",
      "Coverage File chr4 (2 of 30)\n",
      "\n",
      "Coverage File chr4 (3 of 30)\n",
      "\n",
      "Coverage File chr4 (4 of 30)\n",
      "\n",
      "Coverage File chr4 (5 of 30)\n",
      "\n",
      "Coverage File chr4 (6 of 30)\n",
      "\n",
      "Coverage File chr4 (7 of 30)\n",
      "\n",
      "Coverage File chr4 (8 of 30)\n",
      "\n",
      "Coverage File chr4 (9 of 30)\n",
      "\n",
      "Coverage File chr4 (10 of 30)\n",
      "\n",
      "Coverage File chr4 (11 of 30)\n",
      "\n",
      "Coverage File chr4 (12 of 30)\n",
      "\n",
      "Coverage File chr4 (13 of 30)\n",
      "\n",
      "Coverage File chr4 (14 of 30)\n",
      "\n",
      "Coverage File chr4 (15 of 30)\n",
      "\n",
      "Coverage File chr4 (16 of 30)\n",
      "\n",
      "Coverage File chr4 (17 of 30)\n",
      "\n",
      "Coverage File chr4 (18 of 30)\n",
      "\n",
      "Coverage File chr4 (19 of 30)\n",
      "\n",
      "Coverage File chr4 (20 of 30)\n",
      "\n",
      "Coverage File chr4 (21 of 30)\n",
      "\n",
      "Coverage File chr4 (22 of 30)\n",
      "\n",
      "Coverage File chr4 (23 of 30)\n",
      "\n",
      "Coverage File chr4 (24 of 30)\n",
      "\n",
      "Coverage File chr4 (25 of 30)\n",
      "\n",
      "Coverage File chr4 (26 of 30)\n",
      "\n",
      "Coverage File chr4 (27 of 30)\n",
      "\n",
      "Coverage File chr4 (28 of 30)\n",
      "\n",
      "Coverage File chr4 (29 of 30)\n",
      "\n",
      "Coverage File chr4 (30 of 30)\n",
      "\n",
      "Kmer Bias chr5 (18 of 24)\n",
      "\n",
      "chr5 \n",
      "Coverage File chr5 (1 of 30)\n",
      "\n",
      "Coverage File chr5 (2 of 30)\n",
      "\n",
      "Coverage File chr5 (3 of 30)\n",
      "\n",
      "Coverage File chr5 (4 of 30)\n",
      "\n",
      "Coverage File chr5 (5 of 30)\n",
      "\n",
      "Coverage File chr5 (6 of 30)\n",
      "\n",
      "Coverage File chr5 (7 of 30)\n",
      "\n",
      "Coverage File chr5 (8 of 30)\n",
      "\n",
      "Coverage File chr5 (9 of 30)\n",
      "\n",
      "Coverage File chr5 (10 of 30)\n",
      "\n",
      "Coverage File chr5 (11 of 30)\n",
      "\n",
      "Coverage File chr5 (12 of 30)\n",
      "\n",
      "Coverage File chr5 (13 of 30)\n",
      "\n",
      "Coverage File chr5 (14 of 30)\n",
      "\n",
      "Coverage File chr5 (15 of 30)\n",
      "\n",
      "Coverage File chr5 (16 of 30)\n",
      "\n",
      "Coverage File chr5 (17 of 30)\n",
      "\n",
      "Coverage File chr5 (18 of 30)\n",
      "\n",
      "Coverage File chr5 (19 of 30)\n",
      "\n",
      "Coverage File chr5 (20 of 30)\n",
      "\n",
      "Coverage File chr5 (21 of 30)\n",
      "\n",
      "Coverage File chr5 (22 of 30)\n",
      "\n",
      "Coverage File chr5 (23 of 30)\n",
      "\n",
      "Coverage File chr5 (24 of 30)\n",
      "\n",
      "Coverage File chr5 (25 of 30)\n",
      "\n",
      "Coverage File chr5 (26 of 30)\n",
      "\n",
      "Coverage File chr5 (27 of 30)\n",
      "\n",
      "Coverage File chr5 (28 of 30)\n",
      "\n",
      "Coverage File chr5 (29 of 30)\n",
      "\n",
      "Coverage File chr5 (30 of 30)\n",
      "\n",
      "Kmer Bias chr6 (19 of 24)\n",
      "\n",
      "chr6 \n",
      "Coverage File chr6 (1 of 30)\n",
      "\n",
      "Coverage File chr6 (2 of 30)\n",
      "\n",
      "Coverage File chr6 (3 of 30)\n",
      "\n",
      "Coverage File chr6 (4 of 30)\n",
      "\n",
      "Coverage File chr6 (5 of 30)\n",
      "\n",
      "Coverage File chr6 (6 of 30)\n",
      "\n",
      "Coverage File chr6 (7 of 30)\n",
      "\n",
      "Coverage File chr6 (8 of 30)\n",
      "\n",
      "Coverage File chr6 (9 of 30)\n",
      "\n",
      "Coverage File chr6 (10 of 30)\n",
      "\n",
      "Coverage File chr6 (11 of 30)\n",
      "\n",
      "Coverage File chr6 (12 of 30)\n",
      "\n",
      "Coverage File chr6 (13 of 30)\n",
      "\n",
      "Coverage File chr6 (14 of 30)\n",
      "\n",
      "Coverage File chr6 (15 of 30)\n",
      "\n",
      "Coverage File chr6 (16 of 30)\n",
      "\n",
      "Coverage File chr6 (17 of 30)\n",
      "\n",
      "Coverage File chr6 (18 of 30)\n",
      "\n",
      "Coverage File chr6 (19 of 30)\n",
      "\n",
      "Coverage File chr6 (20 of 30)\n",
      "\n",
      "Coverage File chr6 (21 of 30)\n",
      "\n",
      "Coverage File chr6 (22 of 30)\n",
      "\n",
      "Coverage File chr6 (23 of 30)\n",
      "\n",
      "Coverage File chr6 (24 of 30)\n",
      "\n",
      "Coverage File chr6 (25 of 30)\n",
      "\n",
      "Coverage File chr6 (26 of 30)\n",
      "\n",
      "Coverage File chr6 (27 of 30)\n",
      "\n",
      "Coverage File chr6 (28 of 30)\n",
      "\n",
      "Coverage File chr6 (29 of 30)\n",
      "\n",
      "Coverage File chr6 (30 of 30)\n",
      "\n",
      "Kmer Bias chr7 (20 of 24)\n",
      "\n",
      "chr7 \n",
      "Coverage File chr7 (1 of 30)\n",
      "\n",
      "Coverage File chr7 (2 of 30)\n",
      "\n",
      "Coverage File chr7 (3 of 30)\n",
      "\n",
      "Coverage File chr7 (4 of 30)\n",
      "\n",
      "Coverage File chr7 (5 of 30)\n",
      "\n",
      "Coverage File chr7 (6 of 30)\n",
      "\n",
      "Coverage File chr7 (7 of 30)\n",
      "\n",
      "Coverage File chr7 (8 of 30)\n",
      "\n",
      "Coverage File chr7 (9 of 30)\n",
      "\n",
      "Coverage File chr7 (10 of 30)\n",
      "\n",
      "Coverage File chr7 (11 of 30)\n",
      "\n",
      "Coverage File chr7 (12 of 30)\n",
      "\n",
      "Coverage File chr7 (13 of 30)\n",
      "\n",
      "Coverage File chr7 (14 of 30)\n",
      "\n",
      "Coverage File chr7 (15 of 30)\n",
      "\n",
      "Coverage File chr7 (16 of 30)\n",
      "\n",
      "Coverage File chr7 (17 of 30)\n",
      "\n",
      "Coverage File chr7 (18 of 30)\n",
      "\n",
      "Coverage File chr7 (19 of 30)\n",
      "\n",
      "Coverage File chr7 (20 of 30)\n",
      "\n",
      "Coverage File chr7 (21 of 30)\n",
      "\n",
      "Coverage File chr7 (22 of 30)\n",
      "\n",
      "Coverage File chr7 (23 of 30)\n",
      "\n",
      "Coverage File chr7 (24 of 30)\n",
      "\n",
      "Coverage File chr7 (25 of 30)\n",
      "\n",
      "Coverage File chr7 (26 of 30)\n",
      "\n",
      "Coverage File chr7 (27 of 30)\n",
      "\n",
      "Coverage File chr7 (28 of 30)\n",
      "\n",
      "Coverage File chr7 (29 of 30)\n",
      "\n",
      "Coverage File chr7 (30 of 30)\n",
      "\n",
      "Kmer Bias chr8 (21 of 24)\n",
      "\n",
      "chr8 \n",
      "Coverage File chr8 (1 of 30)\n",
      "\n",
      "Coverage File chr8 (2 of 30)\n",
      "\n",
      "Coverage File chr8 (3 of 30)\n",
      "\n",
      "Coverage File chr8 (4 of 30)\n",
      "\n",
      "Coverage File chr8 (5 of 30)\n",
      "\n",
      "Coverage File chr8 (6 of 30)\n",
      "\n",
      "Coverage File chr8 (7 of 30)\n",
      "\n",
      "Coverage File chr8 (8 of 30)\n",
      "\n",
      "Coverage File chr8 (9 of 30)\n",
      "\n",
      "Coverage File chr8 (10 of 30)\n",
      "\n",
      "Coverage File chr8 (11 of 30)\n",
      "\n",
      "Coverage File chr8 (12 of 30)\n",
      "\n",
      "Coverage File chr8 (13 of 30)\n",
      "\n",
      "Coverage File chr8 (14 of 30)\n",
      "\n",
      "Coverage File chr8 (15 of 30)\n",
      "\n",
      "Coverage File chr8 (16 of 30)\n",
      "\n",
      "Coverage File chr8 (17 of 30)\n",
      "\n",
      "Coverage File chr8 (18 of 30)\n",
      "\n",
      "Coverage File chr8 (19 of 30)\n",
      "\n",
      "Coverage File chr8 (20 of 30)\n",
      "\n",
      "Coverage File chr8 (21 of 30)\n",
      "\n",
      "Coverage File chr8 (22 of 30)\n",
      "\n",
      "Coverage File chr8 (23 of 30)\n",
      "\n",
      "Coverage File chr8 (24 of 30)\n",
      "\n",
      "Coverage File chr8 (25 of 30)\n",
      "\n",
      "Coverage File chr8 (26 of 30)\n",
      "\n",
      "Coverage File chr8 (27 of 30)\n",
      "\n",
      "Coverage File chr8 (28 of 30)\n",
      "\n",
      "Coverage File chr8 (29 of 30)\n",
      "\n",
      "Coverage File chr8 (30 of 30)\n",
      "\n",
      "Kmer Bias chr9 (22 of 24)\n",
      "\n",
      "chr9 \n",
      "Coverage File chr9 (1 of 30)\n",
      "\n",
      "Coverage File chr9 (2 of 30)\n",
      "\n",
      "Coverage File chr9 (3 of 30)\n",
      "\n",
      "Coverage File chr9 (4 of 30)\n",
      "\n",
      "Coverage File chr9 (5 of 30)\n",
      "\n",
      "Coverage File chr9 (6 of 30)\n",
      "\n",
      "Coverage File chr9 (7 of 30)\n",
      "\n",
      "Coverage File chr9 (8 of 30)\n",
      "\n",
      "Coverage File chr9 (9 of 30)\n",
      "\n",
      "Coverage File chr9 (10 of 30)\n",
      "\n",
      "Coverage File chr9 (11 of 30)\n",
      "\n",
      "Coverage File chr9 (12 of 30)\n",
      "\n",
      "Coverage File chr9 (13 of 30)\n",
      "\n",
      "Coverage File chr9 (14 of 30)\n",
      "\n",
      "Coverage File chr9 (15 of 30)\n",
      "\n",
      "Coverage File chr9 (16 of 30)\n",
      "\n",
      "Coverage File chr9 (17 of 30)\n",
      "\n",
      "Coverage File chr9 (18 of 30)\n",
      "\n",
      "Coverage File chr9 (19 of 30)\n",
      "\n",
      "Coverage File chr9 (20 of 30)\n",
      "\n",
      "Coverage File chr9 (21 of 30)\n",
      "\n",
      "Coverage File chr9 (22 of 30)\n",
      "\n",
      "Coverage File chr9 (23 of 30)\n",
      "\n",
      "Coverage File chr9 (24 of 30)\n",
      "\n",
      "Coverage File chr9 (25 of 30)\n",
      "\n",
      "Coverage File chr9 (26 of 30)\n",
      "\n",
      "Coverage File chr9 (27 of 30)\n",
      "\n",
      "Coverage File chr9 (28 of 30)\n",
      "\n",
      "Coverage File chr9 (29 of 30)\n",
      "\n",
      "Coverage File chr9 (30 of 30)\n",
      "\n",
      "Kmer Bias chrX (23 of 24)\n",
      "\n",
      "chrX \n",
      "Coverage File chrX (1 of 30)\n",
      "\n",
      "Coverage File chrX (2 of 30)\n",
      "\n",
      "Coverage File chrX (3 of 30)\n",
      "\n",
      "Coverage File chrX (4 of 30)\n",
      "\n",
      "Coverage File chrX (5 of 30)\n",
      "\n",
      "Coverage File chrX (6 of 30)\n",
      "\n",
      "Coverage File chrX (7 of 30)\n",
      "\n",
      "Coverage File chrX (8 of 30)\n",
      "\n",
      "Coverage File chrX (9 of 30)\n",
      "\n",
      "Coverage File chrX (10 of 30)\n",
      "\n",
      "Coverage File chrX (11 of 30)\n",
      "\n",
      "Coverage File chrX (12 of 30)\n",
      "\n",
      "Coverage File chrX (13 of 30)\n",
      "\n",
      "Coverage File chrX (14 of 30)\n",
      "\n",
      "Coverage File chrX (15 of 30)\n",
      "\n",
      "Coverage File chrX (16 of 30)\n",
      "\n",
      "Coverage File chrX (17 of 30)\n",
      "\n",
      "Coverage File chrX (18 of 30)\n",
      "\n",
      "Coverage File chrX (19 of 30)\n",
      "\n",
      "Coverage File chrX (20 of 30)\n",
      "\n",
      "Coverage File chrX (21 of 30)\n",
      "\n",
      "Coverage File chrX (22 of 30)\n",
      "\n",
      "Coverage File chrX (23 of 30)\n",
      "\n",
      "Coverage File chrX (24 of 30)\n",
      "\n",
      "Coverage File chrX (25 of 30)\n",
      "\n",
      "Coverage File chrX (26 of 30)\n",
      "\n",
      "Coverage File chrX (27 of 30)\n",
      "\n",
      "Coverage File chrX (28 of 30)\n",
      "\n",
      "Coverage File chrX (29 of 30)\n",
      "\n",
      "Coverage File chrX (30 of 30)\n",
      "\n",
      "Kmer Bias chrY (24 of 24)\n",
      "\n",
      "chrY \n",
      "Coverage File chrY (1 of 30)\n",
      "\n",
      "Coverage File chrY (2 of 30)\n",
      "\n",
      "Coverage File chrY (3 of 30)\n",
      "\n",
      "Coverage File chrY (4 of 30)\n",
      "\n",
      "Coverage File chrY (5 of 30)\n",
      "\n",
      "Coverage File chrY (6 of 30)\n",
      "\n",
      "Coverage File chrY (7 of 30)\n",
      "\n",
      "Coverage File chrY (8 of 30)\n",
      "\n",
      "Coverage File chrY (9 of 30)\n",
      "\n",
      "Coverage File chrY (10 of 30)\n",
      "\n",
      "Coverage File chrY (11 of 30)\n",
      "\n",
      "Coverage File chrY (12 of 30)\n",
      "\n",
      "Coverage File chrY (13 of 30)\n",
      "\n",
      "Coverage File chrY (14 of 30)\n",
      "\n",
      "Coverage File chrY (15 of 30)\n",
      "\n",
      "Coverage File chrY (16 of 30)\n",
      "\n",
      "Coverage File chrY (17 of 30)\n",
      "\n",
      "Coverage File chrY (18 of 30)\n",
      "\n",
      "Coverage File chrY (19 of 30)\n",
      "\n",
      "Coverage File chrY (20 of 30)\n",
      "\n",
      "Coverage File chrY (21 of 30)\n",
      "\n",
      "Coverage File chrY (22 of 30)\n",
      "\n",
      "Coverage File chrY (23 of 30)\n",
      "\n",
      "Coverage File chrY (24 of 30)\n",
      "\n",
      "Coverage File chrY (25 of 30)\n",
      "\n",
      "Coverage File chrY (26 of 30)\n",
      "\n",
      "Coverage File chrY (27 of 30)\n",
      "\n",
      "Coverage File chrY (28 of 30)\n",
      "\n",
      "Coverage File chrY (29 of 30)\n",
      "\n",
      "Coverage File chrY (30 of 30)\n",
      "\n",
      "Completed Kmer Bias Calculation\n",
      "\n",
      "Adding Kmer Bias (1 of 30)\n",
      "\n",
      "Adding Kmer Bias (2 of 30)\n",
      "\n",
      "Adding Kmer Bias (3 of 30)\n",
      "\n",
      "Adding Kmer Bias (4 of 30)\n",
      "\n",
      "Adding Kmer Bias (5 of 30)\n",
      "\n",
      "Adding Kmer Bias (6 of 30)\n",
      "\n",
      "Adding Kmer Bias (7 of 30)\n",
      "\n",
      "Adding Kmer Bias (8 of 30)\n",
      "\n",
      "Adding Kmer Bias (9 of 30)\n",
      "\n",
      "Adding Kmer Bias (10 of 30)\n",
      "\n",
      "Adding Kmer Bias (11 of 30)\n",
      "\n",
      "Adding Kmer Bias (12 of 30)\n",
      "\n",
      "Adding Kmer Bias (13 of 30)\n",
      "\n",
      "Adding Kmer Bias (14 of 30)\n",
      "\n",
      "Adding Kmer Bias (15 of 30)\n",
      "\n",
      "Adding Kmer Bias (16 of 30)\n",
      "\n",
      "Adding Kmer Bias (17 of 30)\n",
      "\n",
      "Adding Kmer Bias (18 of 30)\n",
      "\n",
      "Adding Kmer Bias (19 of 30)\n",
      "\n",
      "Adding Kmer Bias (20 of 30)\n",
      "\n",
      "Adding Kmer Bias (21 of 30)\n",
      "\n",
      "Adding Kmer Bias (22 of 30)\n",
      "\n",
      "Adding Kmer Bias (23 of 30)\n",
      "\n",
      "Adding Kmer Bias (24 of 30)\n",
      "\n",
      "Adding Kmer Bias (25 of 30)\n",
      "\n",
      "Adding Kmer Bias (26 of 30)\n",
      "\n",
      "Adding Kmer Bias (27 of 30)\n",
      "\n",
      "Adding Kmer Bias (28 of 30)\n",
      "\n",
      "Adding Kmer Bias (29 of 30)\n",
      "\n",
      "Adding Kmer Bias (30 of 30)\n",
      "\n",
      "2024-05-27 20:07:46.302871 : Finished Creation of Coverage Files!, 54.998 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addGroupCoverages-18ed634361199-Date-2024-05-27_Time-19-12-46.448318.log\n",
      "\n",
      "Searching For MACS2..\n",
      "\n",
      "Found with $path!\n",
      "\n",
      "ArchR logging to : ArchRLogs/ArchR-addReproduciblePeakSet-18ed67c9f963d-Date-2024-05-27_Time-20-07-49.627067.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "Calling Peaks with Macs2\n",
      "\n",
      "2024-05-27 20:07:53.036817 : Peak Calling Parameters!, 0.057 mins elapsed.\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    Group nCells nCellsUsed nReplicates nMin nMax maxPeaks\n",
      "C1     C1    315        315           2   40  275   150000\n",
      "C2     C2   2666        540           2   40  500   150000\n",
      "C3     C3   2066        540           2   40  500   150000\n",
      "C4     C4    237        237           2   40  197   118500\n",
      "C5     C5    334        334           2   40  294   150000\n",
      "C6     C6    576        540           2   40  500   150000\n",
      "C7     C7    115        115           2   40   75    57500\n",
      "C8     C8    820        540           2   40  500   150000\n",
      "C9     C9    177        177           2   40  137    88500\n",
      "C10   C10   2413        540           2   40  500   150000\n",
      "C11   C11    266        266           2   40  226   133000\n",
      "C12   C12    699        540           2   40  500   150000\n",
      "C13   C13    795        540           2   40  500   150000\n",
      "C14   C14    391        391           2   40  351   150000\n",
      "C15   C15     71         62           2   40   40    31000\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-27 20:07:53.278801 : Batching Peak Calls!, 0.061 mins elapsed.\n",
      "\n",
      "2024-05-27 20:07:54.494335 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2024-05-27 20:07:54.536014 : Group 1 of 30, Calling Peaks with MACS2!, 0.001 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C1._.Rep1-1 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C1._.Rep1-1.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:09:22.023995 : Group 2 of 30, Calling Peaks with MACS2!, 1.459 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C1._.Rep2-2 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C1._.Rep2-2.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:09:52.630144 : Group 3 of 30, Calling Peaks with MACS2!, 1.969 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C2._.Rep1-3 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C2._.Rep1-3.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:11:24.554738 : Group 4 of 30, Calling Peaks with MACS2!, 3.501 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C2._.Rep2-4 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C2._.Rep2-4.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:11:44.273752 : Group 5 of 30, Calling Peaks with MACS2!, 3.83 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C3._.Rep1-5 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C3._.Rep1-5.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:12:58.621786 : Group 6 of 30, Calling Peaks with MACS2!, 5.069 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C3._.Rep2-6 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C3._.Rep2-6.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:14:25.920361 : Group 7 of 30, Calling Peaks with MACS2!, 6.524 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C4._.Rep1-7 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C4._.Rep1-7.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:16:05.648251 : Group 8 of 30, Calling Peaks with MACS2!, 8.186 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C4._.Rep2-8 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C4._.Rep2-8.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:16:37.691273 : Group 9 of 30, Calling Peaks with MACS2!, 8.72 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C5._.Rep1-9 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C5._.Rep1-9.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:18:06.814272 : Group 10 of 30, Calling Peaks with MACS2!, 10.205 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C5._.Rep2-10 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C5._.Rep2-10.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:18:33.147291 : Group 11 of 30, Calling Peaks with MACS2!, 10.644 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C6._.Rep1-11 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C6._.Rep1-11.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:20:22.747922 : Group 12 of 30, Calling Peaks with MACS2!, 12.471 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C6._.Rep2-12 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C6._.Rep2-12.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:20:46.15344 : Group 13 of 30, Calling Peaks with MACS2!, 12.861 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C7._.Rep1-13 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C7._.Rep1-13.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:21:45.392151 : Group 14 of 30, Calling Peaks with MACS2!, 13.848 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C7._.Rep2-14 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C7._.Rep2-14.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:22:21.234141 : Group 15 of 30, Calling Peaks with MACS2!, 14.446 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C8._.Rep1-15 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C8._.Rep1-15.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:23:54.06034 : Group 16 of 30, Calling Peaks with MACS2!, 15.993 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C8._.Rep2-16 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C8._.Rep2-16.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:25:37.505073 : Group 17 of 30, Calling Peaks with MACS2!, 17.717 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C9._.Rep1-17 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C9._.Rep1-17.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:27:07.85728 : Group 18 of 30, Calling Peaks with MACS2!, 19.223 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C9._.Rep2-18 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C9._.Rep2-18.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:27:57.37852 : Group 19 of 30, Calling Peaks with MACS2!, 20.048 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C10._.Rep1-19 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C10._.Rep1-19.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:31:07.986036 : Group 20 of 30, Calling Peaks with MACS2!, 23.225 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C10._.Rep2-20 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C10._.Rep2-20.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:31:47.680067 : Group 21 of 30, Calling Peaks with MACS2!, 23.886 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C11._.Rep1-21 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C11._.Rep1-21.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:33:46.065561 : Group 22 of 30, Calling Peaks with MACS2!, 25.86 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C11._.Rep2-22 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C11._.Rep2-22.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:34:26.694153 : Group 23 of 30, Calling Peaks with MACS2!, 26.537 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C12._.Rep1-23 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C12._.Rep1-23.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:38:28.686727 : Group 24 of 30, Calling Peaks with MACS2!, 30.57 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C12._.Rep2-24 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C12._.Rep2-24.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:39:24.834882 : Group 25 of 30, Calling Peaks with MACS2!, 31.506 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C13._.Rep1-25 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C13._.Rep1-25.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:42:36.851297 : Group 26 of 30, Calling Peaks with MACS2!, 34.706 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C13._.Rep2-26 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C13._.Rep2-26.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:43:33.27319 : Group 27 of 30, Calling Peaks with MACS2!, 35.646 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C14._.Rep1-27 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C14._.Rep1-27.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:46:06.452779 : Group 28 of 30, Calling Peaks with MACS2!, 38.199 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C14._.Rep2-28 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C14._.Rep2-28.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:46:41.704263 : Group 29 of 30, Calling Peaks with MACS2!, 38.787 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C15._.Rep1-29 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C15._.Rep1-29.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:47:16.370552 : Group 30 of 30, Calling Peaks with MACS2!, 39.365 mins elapsed.\n",
      "\n",
      "Running Macs2 with Params : macs2 callpeak -g 2.7e+09 --name C15._.Rep2-30 --treatment /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds/C15._.Rep2-30.insertions.bed --outdir /project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/InsertionBeds --format BED --call-summits --keep-dup all --nomodel --nolambda --shift -75 --extsize 150 -q 0.1\n",
      "\n",
      "2024-05-27 20:47:57.345968 : Identifying Reproducible Peaks!, 40.129 mins elapsed.\n",
      "\n",
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C1-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C10-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C11-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C12-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C13-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C14-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C15-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C2-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C3-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C4-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C5-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C6-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C7-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C8-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Annotating Peaks : Nearest Gene\n",
      "\n",
      "Annotating Peaks : Gene\n",
      "\n",
      "Annotating Peaks : TSS\n",
      "\n",
      "Annotating Peaks : GC\n",
      "\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"/project/scATAC_analysis/scATAcat_review/preprocessing/Corces2020_brain_scATAC/04_get_peaks_and_genesScore_matrix_ArchR/Corces2020_brain/PeakCalls/C9-reproduciblePeaks.gr.rds\"\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "2024-05-27 20:49:21.126271 : Creating Union Peak Set!, 41.525 mins elapsed.\n",
      "\n",
      "Converged after 7 iterations!\n",
      "\n",
      "2024-05-27 20:49:29.485389 : Finished Creating Union Peak Set (224481)!, 41.664 mins elapsed.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "out_path = \"../../../data/Corces2020_brain_scATAC/04_peaks_and_genesScore_matrix_ArchR/input_for_EpiAnno/\" # where to save resulting matrices | with / at the end\n",
    "threads_=1 # how many threads to use\n",
    "\n",
    "proj1 <- addIterativeLSI(\n",
    "    ArchRProj = proj1,\n",
    "    useMatrix = \"TileMatrix\", \n",
    "    name = \"IterativeLSI\", \n",
    "    iterations = 2, \n",
    "    clusterParams = list( #See Seurat::FindClusters\n",
    "        resolution = c(0.2), \n",
    "        sampleCells = 10000, \n",
    "        n.start = 10\n",
    "    ), \n",
    "    varFeatures = 25000, \n",
    "    dimsToUse = 1:30\n",
    ")\n",
    "\n",
    "proj1 <- addClusters(\n",
    "    input = proj1,\n",
    "    reducedDims = \"IterativeLSI\",\n",
    "    method = \"Seurat\",\n",
    "    name = \"Clusters\",\n",
    "    resolution = 0.8\n",
    ")\n",
    "\n",
    "proj1 <- addUMAP(\n",
    "    ArchRProj = proj1, \n",
    "    reducedDims = \"IterativeLSI\", \n",
    "    name = \"UMAP\", \n",
    "    nNeighbors = 30, \n",
    "    minDist = 0.5, \n",
    "    metric = \"cosine\"\n",
    ")\n",
    "\n",
    "HDF5_USE_FILE_LOCKING=FALSE\n",
    "RHDF5_USE_FILE_LOCKING=FALSE\n",
    "\n",
    "proj2 <- addGroupCoverages(ArchRProj = proj1, groupBy = \"Clusters\")\n",
    "\n",
    "pathToMacs2 <- findMacs2()\n",
    "\n",
    "proj2 <- addReproduciblePeakSet(\n",
    "    ArchRProj = proj2, \n",
    "    groupBy = \"Clusters\", \n",
    "    pathToMacs2 = pathToMacs2,\n",
    "    plot = FALSE\n",
    ")\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a505ac1d-5bcc-40fa-8765-5a5eb2460121",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-addPeakMatrix-18ed612bd9d88-Date-2024-05-27_Time-20-50-54.15089.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-27 20:50:54.993906 : Batch Execution w/ safelapply!, 0 mins elapsed.\n",
      "\n",
      "2024-05-27 20:50:57.012919 : Adding Corces2020_brain to PeakMatrix for Chr (1 of 23)!, 0.028 mins elapsed.\n",
      "\n",
      "2024-05-27 20:51:14.600883 : Adding Corces2020_brain to PeakMatrix for Chr (2 of 23)!, 0.321 mins elapsed.\n",
      "\n",
      "2024-05-27 20:51:25.35719 : Adding Corces2020_brain to PeakMatrix for Chr (3 of 23)!, 0.501 mins elapsed.\n",
      "\n",
      "2024-05-27 20:51:35.00565 : Adding Corces2020_brain to PeakMatrix for Chr (4 of 23)!, 0.661 mins elapsed.\n",
      "\n",
      "2024-05-27 20:51:42.742808 : Adding Corces2020_brain to PeakMatrix for Chr (5 of 23)!, 0.79 mins elapsed.\n",
      "\n",
      "2024-05-27 20:51:51.036091 : Adding Corces2020_brain to PeakMatrix for Chr (6 of 23)!, 0.929 mins elapsed.\n",
      "\n",
      "2024-05-27 20:52:00.827502 : Adding Corces2020_brain to PeakMatrix for Chr (7 of 23)!, 1.092 mins elapsed.\n",
      "\n",
      "2024-05-27 20:52:09.122067 : Adding Corces2020_brain to PeakMatrix for Chr (8 of 23)!, 1.23 mins elapsed.\n",
      "\n",
      "2024-05-27 20:52:16.825547 : Adding Corces2020_brain to PeakMatrix for Chr (9 of 23)!, 1.358 mins elapsed.\n",
      "\n",
      "2024-05-27 20:52:24.008604 : Adding Corces2020_brain to PeakMatrix for Chr (10 of 23)!, 1.478 mins elapsed.\n",
      "\n",
      "2024-05-27 20:52:31.576632 : Adding Corces2020_brain to PeakMatrix for Chr (11 of 23)!, 1.604 mins elapsed.\n",
      "\n",
      "2024-05-27 20:52:40.030919 : Adding Corces2020_brain to PeakMatrix for Chr (12 of 23)!, 1.745 mins elapsed.\n",
      "\n",
      "2024-05-27 20:52:49.342682 : Adding Corces2020_brain to PeakMatrix for Chr (13 of 23)!, 1.9 mins elapsed.\n",
      "\n",
      "2024-05-27 20:52:55.155044 : Adding Corces2020_brain to PeakMatrix for Chr (14 of 23)!, 1.997 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:01.643403 : Adding Corces2020_brain to PeakMatrix for Chr (15 of 23)!, 2.105 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:07.724082 : Adding Corces2020_brain to PeakMatrix for Chr (16 of 23)!, 2.207 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:14.399734 : Adding Corces2020_brain to PeakMatrix for Chr (17 of 23)!, 2.318 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:21.878366 : Adding Corces2020_brain to PeakMatrix for Chr (18 of 23)!, 2.443 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:27.244697 : Adding Corces2020_brain to PeakMatrix for Chr (19 of 23)!, 2.532 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:34.461318 : Adding Corces2020_brain to PeakMatrix for Chr (20 of 23)!, 2.652 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:40.166928 : Adding Corces2020_brain to PeakMatrix for Chr (21 of 23)!, 2.747 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:44.848098 : Adding Corces2020_brain to PeakMatrix for Chr (22 of 23)!, 2.825 mins elapsed.\n",
      "\n",
      "2024-05-27 20:53:50.103309 : Adding Corces2020_brain to PeakMatrix for Chr (23 of 23)!, 2.913 mins elapsed.\n",
      "\n",
      "ArchR logging successful to : ArchRLogs/ArchR-addPeakMatrix-18ed612bd9d88-Date-2024-05-27_Time-20-50-54.15089.log\n",
      "\n"
     ]
    }
   ],
   "source": [
    "proj3 <- addPeakMatrix(proj2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "5442692a-d119-4a45-bf33-5030d3a1fe5b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "ArchR logging to : ArchRLogs/ArchR-getMatrixFromProject-18ed65ded4514-Date-2024-05-27_Time-20-53-55.855521.log\n",
      "If there is an issue, please report to github with logFile!\n",
      "\n",
      "2024-05-27 20:54:37.672151 : Organizing colData, 0.697 mins elapsed.\n",
      "\n",
      "2024-05-27 20:54:37.698124 : Organizing rowData, 0.697 mins elapsed.\n",
      "\n",
      "2024-05-27 20:54:37.728092 : Organizing rowRanges, 0.698 mins elapsed.\n",
      "\n",
      "2024-05-27 20:54:37.770519 : Organizing Assays (1 of 1), 0.699 mins elapsed.\n",
      "\n",
      "2024-05-27 20:54:37.825795 : Constructing SummarizedExperiment, 0.7 mins elapsed.\n",
      "\n",
      "2024-05-27 20:54:43.768018 : Finished Matrix Creation, 0.799 mins elapsed.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "peakMatrix_ = getMatrixFromProject(proj3, useMatrix='PeakMatrix')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ce2b0f9b-1017-4968-8b01-1f67b9548c4b",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Warning message in asMethod(object):\n",
      "“sparse->dense coercion: allocating vector of size 20.0 GiB”\n"
     ]
    }
   ],
   "source": [
    "\n",
    "PeakMatrix_proj3_dense =  as.matrix(peakMatrix_@assays@data$PeakMatrix)\n",
    "\n",
    "write.csv(PeakMatrix_proj3_dense, file = paste(out_path,\"PeakMatrix_dense.csv\",sep=\"\" )) \n",
    "\n",
    "# get the peak regions\n",
    "proj3_peak_metadata  = (getPeakSet(proj3))\n",
    "proj3_peak_regions = data.frame(seqnames = seqnames(proj3_peak_metadata),\n",
    "           start= start(proj3_peak_metadata),\n",
    "            end = end(proj3_peak_metadata))\n",
    "\n",
    "write.csv(proj3_peak_metadata, file =paste(out_path,\"peak_metadata.csv\",sep=\"\" ))\n",
    "write.csv(proj3_peak_regions, file =paste(out_path,\"peak_regions.csv\",sep=\"\" ))\n",
    "write.table(proj3_peak_regions, file =paste(out_path,\"peak_regions.tsv\",sep=\"\" ),col.names = F,row.names=FALSE,sep=\"\\t\", quote = FALSE)\n",
    "\n",
    "# save the peak matrix as sparse mtx\n",
    "\n",
    "\n",
    "# save sparse matrix\n",
    "sparse_matrix <- Matrix(PeakMatrix_proj3_dense , sparse = T )\n",
    "writeMM(obj = sparse_matrix, file = paste(out_path,\"ArchR_peakMatrix.mtx\",sep=\"\" ))\n",
    "\n",
    "# save genes and cells names\n",
    "combined_peak_features =  paste(proj3_peak_regions$seqnames,\"_\",proj3_peak_regions$start,\"_\",proj3_peak_regions$end, sep=\"\")\n",
    "\n",
    "write.table(as.data.frame(combined_peak_features), file = paste(out_path,\"ArchR_peaks.tsv\",sep=\"\" ) , row.names = FALSE, quote = FALSE, col.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "ec641abe-7a74-4eff-8955-7dd0fb133c68",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'GeneScoreMatrix'</li><li>'PeakMatrix'</li><li>'TileMatrix'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'GeneScoreMatrix'\n",
       "\\item 'PeakMatrix'\n",
       "\\item 'TileMatrix'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'GeneScoreMatrix'\n",
       "2. 'PeakMatrix'\n",
       "3. 'TileMatrix'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"GeneScoreMatrix\" \"PeakMatrix\"      \"TileMatrix\"     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "getAvailableMatrices(proj3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "634cea34-1618-4af5-a338-b8d1129febc5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# get the gene activity matrix \n",
    "\n",
    "geneScoreMatrix <- getMatrixFromProject(proj3, useMatrix='GeneScoreMatrix')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0ff29c89-7d7f-4e73-8bf0-248222debbe6",
   "metadata": {},
   "outputs": [],
   "source": [
    "dim(geneScoreMatrix)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67599673-dcde-44df-9050-ec01bd7afe57",
   "metadata": {},
   "outputs": [],
   "source": [
    "geneScore_outDir = \"../../../data/Corces2020_brain_scATAC/04_peaks_and_genesScore_matrix_ArchR/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "496d2733-3552-4ae8-b1db-adf8f6e9cb1c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "write.table(rowData(geneScoreMatrix)$name, paste(geneScore_outDir, \"ArchR_genescore_genes.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4c9cc218-b028-4c12-a674-81ada3922ff6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "write.table(colnames(geneScoreMatrix), paste(geneScore_outDir, \"ArchR_genescore_barcodes.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d6731a5a-e4a1-4705-b0bc-d7b4bb555241",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "geneScoreMatrix_sparse = geneScoreMatrix@assays@data@listData$GeneScoreMatrix\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbfdd6a6-c095-49d2-8fd7-c99b7db87d96",
   "metadata": {},
   "outputs": [],
   "source": [
    "writeMM(obj = geneScoreMatrix_sparse, file=paste(geneScore_outDir, \"ArchR_genescore.mtx\", sep=\"\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3bbb363a-a304-46eb-ba54-880d0eae5b93",
   "metadata": {},
   "outputs": [],
   "source": [
    "saveArchRProject(ArchRProj = proj3, outputDirectory = paste(geneScore_outDir,\"Save-Proj\", sep=\"\"), load = FALSE)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "66a37710-4da9-4dc8-8982-2be6bbf0d1d3",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(stringr)\n",
    "\n",
    "write.table(as.data.frame(str_split_fixed(colnames(geneScoreMatrix),'#',2))$V2, paste(geneScore_outDir, \"cell_IDs.tsv\", sep=\"\"),quote = FALSE, row.names=FALSE, col.names = FALSE)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "479b2250-2372-43b5-969d-b0a3cfce8032",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R-4.3.0",
   "language": "R",
   "name": "r-4.3.0"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.3.0"
  },
  "widgets": {
   "application/vnd.jupyter.widget-state+json": {
    "state": {},
    "version_major": 2,
    "version_minor": 0
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
